{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import numpy as np\n",
    "import string as s\n",
    "import seaborn as sns\n",
    "import math\n",
    "import tables\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import style, colors\n",
    "from itertools import groupby\n",
    "import matplotlib.patches as patches\n",
    "from matplotlib import gridspec\n",
    "import scipy.stats as stats\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage, fcluster, fclusterdata, leaves_list\n",
    "from scipy.spatial.distance import pdist\n",
    "from scipy.ndimage.interpolation import shift\n",
    "# matplotlib.style.use('ggplot')\n",
    "# %matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (12, 6)\n",
    "# matplotlib.rcParams['figure.facecolor'] = 'white'\n",
    "import mpl_toolkits.mplot3d.axes3d as p3\n",
    "from sklearn.cluster import AgglomerativeClustering\n",
    "from sklearn.datasets.samples_generator import make_swiss_roll\n",
    "matplotlib.style.use('ggplot')\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "questions for Misha:\n",
    "repressive marks: H3K9me3 and H3K27me3, \n",
    "broad marks: H3K4me1, H3K9me3, H3K27me3, H3K36me3\n",
    "1. for 6 marks: marks = \"H3K4me1\",\"H3K4me3\",\"H3K9me3\",\"H3K27me3\",\"H3K36me3\",\"H3K27ac\", what regions am i supposed to look at. for example, for H3K4me3 and H3K4me1 and H3K27ac and H3K27me3(negative mark means gene not expressed), , I can look at putative promoter regions.\n",
    "H3K9me3(repressive mark ), complex repetative  regions. leave for later, mark for repetative regions and genes important for development, 1 million regions, bin genome to 10kb bins remove 0 values and cluster on the whole genome   \n",
    "H3K36me3, use gene body. ask file from Misha, \n",
    "H3K4me1 and H3K27ac() can cluster on enhancer and txn binding regions.\n",
    "\n",
    "2. confirm my workflow are correct:\n",
    "1) intersect bdg file with hg19v69_genes.TSS_2000.pc.bed\n",
    "2) sum bdg values for each putative promoters and divide by region length (4000) to get average signal values. i should have integrated on the various bins, (end-start +1)*coverage. see if 0 or 1 based. also Alessia requested using jaccard value, basically i will need a segmenation bed file telling the regions are enriched and then get the jaccard value while overlap enriched regions and putative promoter regions or gene body. \n",
    "3) cluster \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "chromatin states: 100+ histone modifictions\n",
    "example:\n",
    "H3K4me3: promoters\n",
    "H3K4me1: enhancers\n",
    "H3K9me3: repressive\n",
    "H3K27me3:repressive\n",
    "H3K36me3: transcribed\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "cluster H3k36me3, transcribed mark, cluster on gene body, use this file:/home/szong/projects/resource/hg19v69_genes_body.pc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# intersect pileup file with gene body file, need to run on gphost, insufficient memory on local machinek chr suffix is not an issue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "while read i;do /home/rcorbett/aligners/bedtools/BEDTools-Version-2.15.0/bin/bedtools intersect -a ../hg19v69_genes_body.pc -b /projects/seqdev/DA-77/new_pipeline_treatment/step2/$i -wa -wb > $i.intersected;done < ../all_H3K36me3_treat_pileup_bdg_files_names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "wkdir = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "get meta data about library ids, hiv status, histology, total reads sequenced etc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Order</th>\n",
       "      <th>Biospecimen ID</th>\n",
       "      <th>HIV status</th>\n",
       "      <th>Epigenome modifier mutation status</th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>...</th>\n",
       "      <th>H3K4me3  % aligned</th>\n",
       "      <th>H3K9me3 % aligned</th>\n",
       "      <th>H3K27me3 % aligned</th>\n",
       "      <th>H3K36me3 % aligned</th>\n",
       "      <th>H3K27ac % aligned</th>\n",
       "      <th>input DNA control   % aligned</th>\n",
       "      <th>internal sample ID</th>\n",
       "      <th>pool ID</th>\n",
       "      <th>Notes</th>\n",
       "      <th># lanes 7 PE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02007-01A</td>\n",
       "      <td>negative</td>\n",
       "      <td>mutated</td>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>...</td>\n",
       "      <td>98.13</td>\n",
       "      <td>93.1</td>\n",
       "      <td>97.93</td>\n",
       "      <td>98.57</td>\n",
       "      <td>98.10</td>\n",
       "      <td>96.26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IX5978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>HTMCP-03-06-02026-01A</td>\n",
       "      <td>negative</td>\n",
       "      <td>mutated</td>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>...</td>\n",
       "      <td>97.68</td>\n",
       "      <td>93.89</td>\n",
       "      <td>97.84</td>\n",
       "      <td>98.17</td>\n",
       "      <td>97.86</td>\n",
       "      <td>95.56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IX5981</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Order         Biospecimen ID HIV status Epigenome modifier mutation status  \\\n",
       "0      1  HTMCP-03-06-02007-01A   negative                            mutated   \n",
       "1      2  HTMCP-03-06-02026-01A   negative                            mutated   \n",
       "\n",
       "  H3K4me1 library ID H3K4me3 library ID H3K9me3 library ID  \\\n",
       "0             A84624             A84625             A84626   \n",
       "1             A84645             A84646             A84647   \n",
       "\n",
       "  H3K27me3 library ID H3K36me3 library ID H3K27ac library ID      ...       \\\n",
       "0              A84627              A84628             A84629      ...        \n",
       "1              A84648              A84649             A84650      ...        \n",
       "\n",
       "  H3K4me3  % aligned H3K9me3 % aligned H3K27me3 % aligned H3K36me3 % aligned  \\\n",
       "0              98.13              93.1              97.93              98.57   \n",
       "1              97.68             93.89              97.84              98.17   \n",
       "\n",
       "  H3K27ac % aligned input DNA control   % aligned  internal sample ID  \\\n",
       "0             98.10                         96.26                 NaN   \n",
       "1             97.86                         95.56                 NaN   \n",
       "\n",
       "   pool ID  Notes  # lanes 7 PE  \n",
       "0   IX5978    NaN             1  \n",
       "1   IX5981    NaN             1  \n",
       "\n",
       "[2 rows x 31 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = '/home/szong/projects/META_2018/Master_Sample_table_14X292_Feb27_2018_sz.xls'\n",
    "meta = pd.read_excel(f, sheetname='HTMCP CaCx epi')\n",
    "meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "meta = meta[['Biospecimen ID', 'H3K4me1 library ID',\n",
    "       'H3K4me3 library ID', 'H3K9me3 library ID', 'H3K27me3 library ID',\n",
    "       'H3K36me3 library ID', 'H3K27ac library ID',\n",
    "       'input DNA control  library ID',  'H3K4me1 total reads', 'H3K4me3  total reads',\n",
    "       'H3K9me3 total reads', 'H3K27me3 total reads', 'H3K36me3 total reads',\n",
    "       'H3K27ac total reads', 'input DNA control  total reads']].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biospecimen ID</th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>input DNA control  library ID</th>\n",
       "      <th>H3K4me1 total reads</th>\n",
       "      <th>H3K4me3  total reads</th>\n",
       "      <th>H3K9me3 total reads</th>\n",
       "      <th>H3K27me3 total reads</th>\n",
       "      <th>H3K36me3 total reads</th>\n",
       "      <th>H3K27ac total reads</th>\n",
       "      <th>input DNA control  total reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007-01A</td>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>A84630</td>\n",
       "      <td>78611726</td>\n",
       "      <td>60179836</td>\n",
       "      <td>76557498</td>\n",
       "      <td>90312752.0</td>\n",
       "      <td>106610232.0</td>\n",
       "      <td>80676254.0</td>\n",
       "      <td>86539886.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026-01A</td>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>A84651</td>\n",
       "      <td>85566812</td>\n",
       "      <td>74487212</td>\n",
       "      <td>93042922</td>\n",
       "      <td>89559600.0</td>\n",
       "      <td>87905340.0</td>\n",
       "      <td>84986034.0</td>\n",
       "      <td>88575282.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Biospecimen ID H3K4me1 library ID H3K4me3 library ID  \\\n",
       "0  HTMCP-03-06-02007-01A             A84624             A84625   \n",
       "1  HTMCP-03-06-02026-01A             A84645             A84646   \n",
       "\n",
       "  H3K9me3 library ID H3K27me3 library ID H3K36me3 library ID  \\\n",
       "0             A84626              A84627              A84628   \n",
       "1             A84647              A84648              A84649   \n",
       "\n",
       "  H3K27ac library ID input DNA control  library ID H3K4me1 total reads  \\\n",
       "0             A84629                        A84630            78611726   \n",
       "1             A84650                        A84651            85566812   \n",
       "\n",
       "  H3K4me3  total reads H3K9me3 total reads  H3K27me3 total reads  \\\n",
       "0             60179836            76557498            90312752.0   \n",
       "1             74487212            93042922            89559600.0   \n",
       "\n",
       "   H3K36me3 total reads  H3K27ac total reads  input DNA control  total reads  \n",
       "0           106610232.0           80676254.0                      86539886.0  \n",
       "1            87905340.0           84986034.0                      88575282.0  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "meta['patient'] = meta['Biospecimen ID'].apply(lambda x: '-'.join(['HTMCP-03-06', x.split('-')[3]]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "meta = meta.drop('Biospecimen ID', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>input DNA control  library ID</th>\n",
       "      <th>H3K4me1 total reads</th>\n",
       "      <th>H3K4me3  total reads</th>\n",
       "      <th>H3K9me3 total reads</th>\n",
       "      <th>H3K27me3 total reads</th>\n",
       "      <th>H3K36me3 total reads</th>\n",
       "      <th>H3K27ac total reads</th>\n",
       "      <th>input DNA control  total reads</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>A84630</td>\n",
       "      <td>78611726</td>\n",
       "      <td>60179836</td>\n",
       "      <td>76557498</td>\n",
       "      <td>90312752.0</td>\n",
       "      <td>106610232.0</td>\n",
       "      <td>80676254.0</td>\n",
       "      <td>86539886.0</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>A84651</td>\n",
       "      <td>85566812</td>\n",
       "      <td>74487212</td>\n",
       "      <td>93042922</td>\n",
       "      <td>89559600.0</td>\n",
       "      <td>87905340.0</td>\n",
       "      <td>84986034.0</td>\n",
       "      <td>88575282.0</td>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  H3K4me1 library ID H3K4me3 library ID H3K9me3 library ID  \\\n",
       "0             A84624             A84625             A84626   \n",
       "1             A84645             A84646             A84647   \n",
       "\n",
       "  H3K27me3 library ID H3K36me3 library ID H3K27ac library ID  \\\n",
       "0              A84627              A84628             A84629   \n",
       "1              A84648              A84649             A84650   \n",
       "\n",
       "  input DNA control  library ID H3K4me1 total reads H3K4me3  total reads  \\\n",
       "0                        A84630            78611726             60179836   \n",
       "1                        A84651            85566812             74487212   \n",
       "\n",
       "  H3K9me3 total reads  H3K27me3 total reads  H3K36me3 total reads  \\\n",
       "0            76557498            90312752.0           106610232.0   \n",
       "1            93042922            89559600.0            87905340.0   \n",
       "\n",
       "   H3K27ac total reads  input DNA control  total reads            patient  \n",
       "0           80676254.0                      86539886.0  HTMCP-03-06-02007  \n",
       "1           84986034.0                      88575282.0  HTMCP-03-06-02026  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>strelka_mutations</th>\n",
       "      <th>HPV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>26974</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02002</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>16661</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient HIV_status Putative_histology  strelka_mutations  \\\n",
       "0  HTMCP-03-06-02001   Positive           Squamous              26974   \n",
       "1  HTMCP-03-06-02002   Negative           Squamous              16661   \n",
       "\n",
       "  HPV_status  \n",
       "0   Positive  \n",
       "1   Positive  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add in hiv status and histology\n",
    "f22 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutation_load_clinic.txt'\n",
    "df22 = pd.read_csv(f22, sep='\\t')\n",
    "df22 = df22[['patient', 'HIV_status', 'Putative_histology', 'strelka_mutations', 'HPV_status']]\n",
    "# df22 = df22.set_index('patient')#.stack().reset_index()\n",
    "df22.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>input DNA control  library ID</th>\n",
       "      <th>H3K4me1 total reads</th>\n",
       "      <th>H3K4me3  total reads</th>\n",
       "      <th>H3K9me3 total reads</th>\n",
       "      <th>H3K27me3 total reads</th>\n",
       "      <th>H3K36me3 total reads</th>\n",
       "      <th>H3K27ac total reads</th>\n",
       "      <th>input DNA control  total reads</th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>strelka_mutations</th>\n",
       "      <th>HPV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>A84630</td>\n",
       "      <td>78611726</td>\n",
       "      <td>60179836</td>\n",
       "      <td>76557498</td>\n",
       "      <td>90312752.0</td>\n",
       "      <td>106610232.0</td>\n",
       "      <td>80676254.0</td>\n",
       "      <td>86539886.0</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>2958282</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>A84651</td>\n",
       "      <td>85566812</td>\n",
       "      <td>74487212</td>\n",
       "      <td>93042922</td>\n",
       "      <td>89559600.0</td>\n",
       "      <td>87905340.0</td>\n",
       "      <td>84986034.0</td>\n",
       "      <td>88575282.0</td>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1252217</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  H3K4me1 library ID H3K4me3 library ID H3K9me3 library ID  \\\n",
       "0             A84624             A84625             A84626   \n",
       "1             A84645             A84646             A84647   \n",
       "\n",
       "  H3K27me3 library ID H3K36me3 library ID H3K27ac library ID  \\\n",
       "0              A84627              A84628             A84629   \n",
       "1              A84648              A84649             A84650   \n",
       "\n",
       "  input DNA control  library ID H3K4me1 total reads H3K4me3  total reads  \\\n",
       "0                        A84630            78611726             60179836   \n",
       "1                        A84651            85566812             74487212   \n",
       "\n",
       "  H3K9me3 total reads  H3K27me3 total reads  H3K36me3 total reads  \\\n",
       "0            76557498            90312752.0           106610232.0   \n",
       "1            93042922            89559600.0            87905340.0   \n",
       "\n",
       "   H3K27ac total reads  input DNA control  total reads            patient  \\\n",
       "0           80676254.0                      86539886.0  HTMCP-03-06-02007   \n",
       "1           84986034.0                      88575282.0  HTMCP-03-06-02026   \n",
       "\n",
       "  HIV_status Putative_histology  strelka_mutations HPV_status  \n",
       "0   Negative              Adeno            2958282   Negative  \n",
       "1   Negative              Adeno            1252217   Negative  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta = pd.merge(meta, df22, left_on='patient', right_on='patient')\n",
    "df_meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/chipseq_meta.txt'\n",
    "df_meta.to_csv(of, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "category_track = df_meta[['patient', 'HIV_status', 'Putative_histology', 'HPV_status']]\n",
    "numeric_track = df_meta[['patient', 'strelka_mutations', ]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>HPV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient HIV_status Putative_histology HPV_status\n",
       "0  HTMCP-03-06-02007   Negative              Adeno   Negative\n",
       "1  HTMCP-03-06-02026   Negative              Adeno   Negative"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "category_track.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>strelka_mutations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>2958282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>1252217</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient  strelka_mutations\n",
       "0  HTMCP-03-06-02007            2958282\n",
       "1  HTMCP-03-06-02026            1252217"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "numeric_track.head(2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/szong/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "numeric_track['log_strelka_mutations'] = np.log10(numeric_track['strelka_mutations'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "numeric_track = numeric_track.drop('strelka_mutations', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>log_strelka_mutations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>6.47104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>6.09768</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient  log_strelka_mutations\n",
       "0  HTMCP-03-06-02007                6.47104\n",
       "1  HTMCP-03-06-02026                6.09768"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "numeric_track.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "read in Alessia's epigenetic gene categorization file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HIV neg</th>\n",
       "      <th>neg samples with mutations</th>\n",
       "      <th>HIV positive</th>\n",
       "      <th>pos samples with mutation</th>\n",
       "      <th>total</th>\n",
       "      <th>Category_1</th>\n",
       "      <th>Category_2</th>\n",
       "      <th>markes /regiones affected</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MLL3</td>\n",
       "      <td>9</td>\n",
       "      <td>MLL3</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>Writer</td>\n",
       "      <td>compass complex</td>\n",
       "      <td>H3k4me1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MLL2</td>\n",
       "      <td>7</td>\n",
       "      <td>MLL2</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>Writer</td>\n",
       "      <td>compass complex</td>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  HIV neg  neg samples with mutations HIV positive  pos samples with mutation  \\\n",
       "0    MLL3                           9         MLL3                          6   \n",
       "1    MLL2                           7         MLL2                          5   \n",
       "\n",
       "   total Category_1       Category_2 markes /regiones affected comments  \n",
       "0     15     Writer  compass complex                   H3k4me1      NaN  \n",
       "1     12     Writer  compass complex                   H3K4me3      NaN  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f4 = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/categorisation_of_epigentic_modifiers_from_Alessia_v3.xls'\n",
    "df4 = pd.read_excel(f4, sheetname='categorisation')\n",
    "df4.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category_1</th>\n",
       "      <th>Category_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIV neg</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MLL3</th>\n",
       "      <td>Writer</td>\n",
       "      <td>compass complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL2</th>\n",
       "      <td>Writer</td>\n",
       "      <td>compass complex</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Category_1       Category_2\n",
       "HIV neg                            \n",
       "MLL3        Writer  compass complex\n",
       "MLL2        Writer  compass complex"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df41 = df4[['HIV neg', 'Category_1', 'Category_2']].set_index('HIV neg')\n",
    "df41.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Writer', 'Remodeller', 'Eraser', 'Reader', nan], dtype=object)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df41.Category_1.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Polycomb complex                                                              10\n",
       "compass complex                                                                9\n",
       "INO80/SWR remodelling complexes                                                9\n",
       "ISWI remdelling complexes                                                      7\n",
       "HDAC complex                                                                   7\n",
       "GNAT family HAT complexes                                                      7\n",
       "NuRD complex                                                                   5\n",
       "MYST family HAT complexes                                                      5\n",
       "SWI/SNF remodeling complex                                                     5\n",
       "SEC (Super Elongation) Complex                                                 4\n",
       "EP300/CBP HAT complexes                                                        3\n",
       "TFIID basal transcription factor complex or PCAF histone acetylase complex     3\n",
       "chromatin remodelling                                                          2\n",
       "RNF20/40 complex                                                               2\n",
       "SWI/SNF family                                                                 2\n",
       "HAT complexes                                                                  1\n",
       "ISWI chromatin remdelling                                                      1\n",
       "REST corepressor                                                               1\n",
       "Testis-specific chromatin protein                                              1\n",
       "SWI/SNF  remodeling complex                                                    1\n",
       "Name: Category_2, dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df41.Category_2.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HIV status</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ATRX</th>\n",
       "      <th>BAZ1A</th>\n",
       "      <th>BAZ1B</th>\n",
       "      <th>BAZ2B</th>\n",
       "      <th>BRD2</th>\n",
       "      <th>BRD3</th>\n",
       "      <th>BRD4</th>\n",
       "      <th>BRD7</th>\n",
       "      <th>...</th>\n",
       "      <th>TBL1X</th>\n",
       "      <th>TFPT</th>\n",
       "      <th>TRIM33</th>\n",
       "      <th>UBE2B</th>\n",
       "      <th>TRRAP</th>\n",
       "      <th>WHSC1L1</th>\n",
       "      <th>YEATS2</th>\n",
       "      <th>YY1</th>\n",
       "      <th>ZMYND11</th>\n",
       "      <th>ZMYND8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <td>HIV negative</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <td>HIV negative</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     HIV status  ARID1A  ATRX  BAZ1A  BAZ1B  BAZ2B  BRD2  \\\n",
       "patient                                                                    \n",
       "HTMCP-03-06-02007  HIV negative       1     1      1      0      1     0   \n",
       "HTMCP-03-06-02026  HIV negative       1     1      0      1      1     0   \n",
       "\n",
       "                   BRD3  BRD4  BRD7   ...    TBL1X  TFPT  TRIM33  UBE2B  \\\n",
       "patient                               ...                                 \n",
       "HTMCP-03-06-02007     0     1     1   ...        0     0       0      0   \n",
       "HTMCP-03-06-02026     1     0     1   ...        0     0       1      0   \n",
       "\n",
       "                   TRRAP  WHSC1L1  YEATS2  YY1  ZMYND11  ZMYND8  \n",
       "patient                                                          \n",
       "HTMCP-03-06-02007      1        0       0    1        0       1  \n",
       "HTMCP-03-06-02026      0        0       0    0        1       0  \n",
       "\n",
       "[2 rows x 117 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df5 = pd.read_excel(f4, sheetname='sample_mut_details_sz', index_col='patient')\n",
    "df5.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02076</th>\n",
       "      <th>HTMCP-03-06-02113</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATRX</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient  HTMCP-03-06-02007  HTMCP-03-06-02026  HTMCP-03-06-02063  \\\n",
       "ARID1A                 1.0                1.0                0.0   \n",
       "ATRX                   1.0                1.0                1.0   \n",
       "\n",
       "patient  HTMCP-03-06-02214  HTMCP-03-06-02042  HTMCP-03-06-02332  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02185  HTMCP-03-06-02012  HTMCP-03-06-02196  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02203        ...          HTMCP-03-06-02076  \\\n",
       "ARID1A                 0.0        ...                        0.0   \n",
       "ATRX                   0.0        ...                        0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02113  HTMCP-03-06-02230  HTMCP-03-06-02170  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02152  HTMCP-03-06-02235  HTMCP-03-06-02195  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02261  HTMCP-03-06-02147  HTMCP-03-06-02267  \n",
       "ARID1A                 0.0                0.0                0.0  \n",
       "ATRX                   0.0                0.0                0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df51 = df5[df5.columns[1:]].T\n",
    "\n",
    "df51.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "      <th>Category_1</th>\n",
       "      <th>Category_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF remodeling complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATRX</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF family</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        HTMCP-03-06-02007  HTMCP-03-06-02026  HTMCP-03-06-02063  \\\n",
       "ARID1A                1.0                1.0                0.0   \n",
       "ATRX                  1.0                1.0                1.0   \n",
       "\n",
       "        HTMCP-03-06-02214  HTMCP-03-06-02042  HTMCP-03-06-02332  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02185  HTMCP-03-06-02012  HTMCP-03-06-02196  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02203             ...              HTMCP-03-06-02230  \\\n",
       "ARID1A                0.0             ...                            0.0   \n",
       "ATRX                  0.0             ...                            0.0   \n",
       "\n",
       "        HTMCP-03-06-02170  HTMCP-03-06-02152  HTMCP-03-06-02235  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02195  HTMCP-03-06-02261  HTMCP-03-06-02147  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02267  Category_1                  Category_2  \n",
       "ARID1A                0.0  Remodeller  SWI/SNF remodeling complex  \n",
       "ATRX                  0.0  Remodeller              SWI/SNF family  \n",
       "\n",
       "[2 rows x 56 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df6 = pd.merge(df51, df41, left_index=True, right_index=True)\n",
    "\n",
    "df6.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def categorize_cols(df, col):\n",
    "    col_values = df[col].unique()\n",
    "    lst = []\n",
    "    for col_value in col_values:\n",
    "        s = df.query('{0} == \"{1}\"'.format(col, col_value)).sum()\n",
    "        lst.append(s)\n",
    "    dfo = pd.DataFrame(lst)\n",
    "    dfo.index = col_values\n",
    "    dfo = dfo.drop([col], axis=1)\n",
    "    return dfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02040</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02266</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "      <th>HTMCP-03-06-02268</th>\n",
       "      <th>HTMCP-03-06-02320</th>\n",
       "      <th>HTMCP-03-06-02330</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP.03.06.02037</th>\n",
       "      <th>HTMCP.03.06.02046</th>\n",
       "      <th>HTMCP.03.06.02150</th>\n",
       "      <th>HTMCP.03.06.02174</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Remodeller</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reader</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            HTMCP-03-06-02001  HTMCP-03-06-02002  HTMCP-03-06-02006  \\\n",
       "Remodeller                0.0                0.0                2.0   \n",
       "Reader                    1.0                0.0                0.0   \n",
       "\n",
       "            HTMCP-03-06-02007  HTMCP-03-06-02012  HTMCP-03-06-02020  \\\n",
       "Remodeller               16.0                1.0                1.0   \n",
       "Reader                   21.0                1.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02026  HTMCP-03-06-02036  HTMCP-03-06-02040  \\\n",
       "Remodeller                9.0                0.0                2.0   \n",
       "Reader                    7.0                0.0                0.0   \n",
       "\n",
       "            HTMCP-03-06-02042        ...          HTMCP-03-06-02266  \\\n",
       "Remodeller                2.0        ...                        1.0   \n",
       "Reader                    1.0        ...                        0.0   \n",
       "\n",
       "            HTMCP-03-06-02267  HTMCP-03-06-02268  HTMCP-03-06-02320  \\\n",
       "Remodeller                0.0                0.0                0.0   \n",
       "Reader                    0.0                0.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02330  HTMCP-03-06-02332  HTMCP.03.06.02037  \\\n",
       "Remodeller                0.0                2.0                0.0   \n",
       "Reader                    1.0                1.0                0.0   \n",
       "\n",
       "            HTMCP.03.06.02046  HTMCP.03.06.02150  HTMCP.03.06.02174  \n",
       "Remodeller                0.0                0.0                0.0  \n",
       "Reader                    0.0                0.0                0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df8 = categorize_cols(df6, 'Category_1').drop('Category_2', axis=1)\n",
    "\n",
    "df8.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02040</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02266</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "      <th>HTMCP-03-06-02268</th>\n",
       "      <th>HTMCP-03-06-02320</th>\n",
       "      <th>HTMCP-03-06-02330</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP.03.06.02037</th>\n",
       "      <th>HTMCP.03.06.02046</th>\n",
       "      <th>HTMCP.03.06.02150</th>\n",
       "      <th>HTMCP.03.06.02174</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SWI/SNF remodeling complex</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SWI/SNF family</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            HTMCP-03-06-02001  HTMCP-03-06-02002  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02006  HTMCP-03-06-02007  \\\n",
       "SWI/SNF remodeling complex                0.0                5.0   \n",
       "SWI/SNF family                            0.0                2.0   \n",
       "\n",
       "                            HTMCP-03-06-02012  HTMCP-03-06-02020  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02026  HTMCP-03-06-02036  \\\n",
       "SWI/SNF remodeling complex                2.0                0.0   \n",
       "SWI/SNF family                            2.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02040  HTMCP-03-06-02042  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            1.0                0.0   \n",
       "\n",
       "                                  ...          HTMCP-03-06-02266  \\\n",
       "SWI/SNF remodeling complex        ...                        0.0   \n",
       "SWI/SNF family                    ...                        0.0   \n",
       "\n",
       "                            HTMCP-03-06-02267  HTMCP-03-06-02268  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02320  HTMCP-03-06-02330  \\\n",
       "SWI/SNF remodeling complex                0.0                1.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02332  HTMCP.03.06.02037  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP.03.06.02046  HTMCP.03.06.02150  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP.03.06.02174  \n",
       "SWI/SNF remodeling complex                0.0  \n",
       "SWI/SNF family                            0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df9 = categorize_cols(df6, 'Category_2').drop('Category_1', axis=1)\n",
    "\n",
    "df9.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02040</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02266</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "      <th>HTMCP-03-06-02268</th>\n",
       "      <th>HTMCP-03-06-02320</th>\n",
       "      <th>HTMCP-03-06-02330</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP.03.06.02037</th>\n",
       "      <th>HTMCP.03.06.02046</th>\n",
       "      <th>HTMCP.03.06.02150</th>\n",
       "      <th>HTMCP.03.06.02174</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Remodeller</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reader</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            HTMCP-03-06-02001  HTMCP-03-06-02002  HTMCP-03-06-02006  \\\n",
       "index                                                                 \n",
       "Remodeller                0.0                0.0                2.0   \n",
       "Reader                    1.0                0.0                0.0   \n",
       "\n",
       "            HTMCP-03-06-02007  HTMCP-03-06-02012  HTMCP-03-06-02020  \\\n",
       "index                                                                 \n",
       "Remodeller               16.0                1.0                1.0   \n",
       "Reader                   21.0                1.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02026  HTMCP-03-06-02036  HTMCP-03-06-02040  \\\n",
       "index                                                                 \n",
       "Remodeller                9.0                0.0                2.0   \n",
       "Reader                    7.0                0.0                0.0   \n",
       "\n",
       "            HTMCP-03-06-02042        ...          HTMCP-03-06-02266  \\\n",
       "index                                ...                              \n",
       "Remodeller                2.0        ...                        1.0   \n",
       "Reader                    1.0        ...                        0.0   \n",
       "\n",
       "            HTMCP-03-06-02267  HTMCP-03-06-02268  HTMCP-03-06-02320  \\\n",
       "index                                                                 \n",
       "Remodeller                0.0                0.0                0.0   \n",
       "Reader                    0.0                0.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02330  HTMCP-03-06-02332  HTMCP.03.06.02037  \\\n",
       "index                                                                 \n",
       "Remodeller                0.0                2.0                0.0   \n",
       "Reader                    1.0                1.0                0.0   \n",
       "\n",
       "            HTMCP.03.06.02046  HTMCP.03.06.02150  HTMCP.03.06.02174  \n",
       "index                                                                \n",
       "Remodeller                0.0                0.0                0.0  \n",
       "Reader                    0.0                0.0                0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df10 = pd.concat([df8, df9]).reset_index().dropna().set_index('index').astype(float)\n",
    "\n",
    "df10.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df10[df10 < 0.5] = np.nan\n",
    "df10[df10 > 0.5] = 'Positive'\n",
    "df10.fillna('Negative', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02040</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02266</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "      <th>HTMCP-03-06-02268</th>\n",
       "      <th>HTMCP-03-06-02320</th>\n",
       "      <th>HTMCP-03-06-02330</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP.03.06.02037</th>\n",
       "      <th>HTMCP.03.06.02046</th>\n",
       "      <th>HTMCP.03.06.02150</th>\n",
       "      <th>HTMCP.03.06.02174</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Remodeller</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reader</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02006  \\\n",
       "index                                                              \n",
       "Remodeller          Negative          Negative          Positive   \n",
       "Reader              Positive          Negative          Negative   \n",
       "\n",
       "           HTMCP-03-06-02007 HTMCP-03-06-02012 HTMCP-03-06-02020  \\\n",
       "index                                                              \n",
       "Remodeller          Positive          Positive          Positive   \n",
       "Reader              Positive          Positive          Positive   \n",
       "\n",
       "           HTMCP-03-06-02026 HTMCP-03-06-02036 HTMCP-03-06-02040  \\\n",
       "index                                                              \n",
       "Remodeller          Positive          Negative          Positive   \n",
       "Reader              Positive          Negative          Negative   \n",
       "\n",
       "           HTMCP-03-06-02042        ...        HTMCP-03-06-02266  \\\n",
       "index                               ...                            \n",
       "Remodeller          Positive        ...                 Positive   \n",
       "Reader              Positive        ...                 Negative   \n",
       "\n",
       "           HTMCP-03-06-02267 HTMCP-03-06-02268 HTMCP-03-06-02320  \\\n",
       "index                                                              \n",
       "Remodeller          Negative          Negative          Negative   \n",
       "Reader              Negative          Negative          Positive   \n",
       "\n",
       "           HTMCP-03-06-02330 HTMCP-03-06-02332 HTMCP.03.06.02037  \\\n",
       "index                                                              \n",
       "Remodeller          Negative          Positive          Negative   \n",
       "Reader              Positive          Positive          Negative   \n",
       "\n",
       "           HTMCP.03.06.02046 HTMCP.03.06.02150 HTMCP.03.06.02174  \n",
       "index                                                             \n",
       "Remodeller          Negative          Negative          Negative  \n",
       "Reader              Negative          Negative          Negative  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df10.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>HPV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient HIV_status Putative_histology HPV_status\n",
       "0  HTMCP-03-06-02007   Negative              Adeno   Negative\n",
       "1  HTMCP-03-06-02026   Negative              Adeno   Negative"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "category_track.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df11 = df10.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>index</th>\n",
       "      <th>Remodeller</th>\n",
       "      <th>Reader</th>\n",
       "      <th>Writer</th>\n",
       "      <th>Eraser</th>\n",
       "      <th>SWI/SNF remodeling complex</th>\n",
       "      <th>SWI/SNF family</th>\n",
       "      <th>ISWI remdelling complexes</th>\n",
       "      <th>SEC (Super Elongation) Complex</th>\n",
       "      <th>INO80/SWR remodelling complexes</th>\n",
       "      <th>Testis-specific chromatin protein</th>\n",
       "      <th>...</th>\n",
       "      <th>ISWI chromatin remdelling</th>\n",
       "      <th>HAT complexes</th>\n",
       "      <th>NuRD complex</th>\n",
       "      <th>compass complex</th>\n",
       "      <th>HDAC complex</th>\n",
       "      <th>GNAT family HAT complexes</th>\n",
       "      <th>MYST family HAT complexes</th>\n",
       "      <th>RNF20/40 complex</th>\n",
       "      <th>SWI/SNF  remodeling complex</th>\n",
       "      <th>TFIID basal transcription factor complex or PCAF histone acetylase complex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "index             Remodeller    Reader    Writer    Eraser  \\\n",
       "HTMCP-03-06-02001   Negative  Positive  Positive  Negative   \n",
       "HTMCP-03-06-02002   Negative  Negative  Positive  Negative   \n",
       "\n",
       "index             SWI/SNF remodeling complex SWI/SNF family  \\\n",
       "HTMCP-03-06-02001                   Negative       Negative   \n",
       "HTMCP-03-06-02002                   Negative       Negative   \n",
       "\n",
       "index             ISWI remdelling complexes SEC (Super Elongation) Complex  \\\n",
       "HTMCP-03-06-02001                  Negative                       Negative   \n",
       "HTMCP-03-06-02002                  Negative                       Negative   \n",
       "\n",
       "index             INO80/SWR remodelling complexes  \\\n",
       "HTMCP-03-06-02001                        Positive   \n",
       "HTMCP-03-06-02002                        Negative   \n",
       "\n",
       "index             Testis-specific chromatin protein  \\\n",
       "HTMCP-03-06-02001                          Negative   \n",
       "HTMCP-03-06-02002                          Negative   \n",
       "\n",
       "index                                                ...                                      \\\n",
       "HTMCP-03-06-02001                                    ...                                       \n",
       "HTMCP-03-06-02002                                    ...                                       \n",
       "\n",
       "index             ISWI chromatin remdelling HAT complexes NuRD complex   \\\n",
       "HTMCP-03-06-02001                  Negative      Negative      Negative   \n",
       "HTMCP-03-06-02002                  Negative      Negative      Negative   \n",
       "\n",
       "index             compass complex HDAC complex GNAT family HAT complexes  \\\n",
       "HTMCP-03-06-02001        Positive     Negative                  Positive   \n",
       "HTMCP-03-06-02002        Positive     Negative                  Negative   \n",
       "\n",
       "index             MYST family HAT complexes RNF20/40 complex   \\\n",
       "HTMCP-03-06-02001                  Negative          Negative   \n",
       "HTMCP-03-06-02002                  Negative          Negative   \n",
       "\n",
       "index             SWI/SNF  remodeling complex  \\\n",
       "HTMCP-03-06-02001                    Negative   \n",
       "HTMCP-03-06-02002                    Negative   \n",
       "\n",
       "index             TFIID basal transcription factor complex or PCAF histone acetylase complex  \n",
       "HTMCP-03-06-02001                                           Positive                          \n",
       "HTMCP-03-06-02002                                           Negative                          \n",
       "\n",
       "[2 rows x 24 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df11.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "category_track = pd.merge(category_track, df11, left_on='patient', right_index=True, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "of ='{}category_track.txt'.format(wkdir)\n",
    "category_track.to_csv(of, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "process chipseq signals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_files(f):\n",
    "    # f = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/sorted_UCSC_A84611_H3K4me3_treat_pileup.bdg.intersected'\n",
    "    df = pd.read_csv(f,header=None)\n",
    "    df.columns = ['file']\n",
    "    df.head(2)\n",
    "    return df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def integrate_signals(f):\n",
    "#     f = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/pileup_files/sorted_UCSC_A84611_H3K4me3_treat_pileup.bdg.intersected'\n",
    "    df = pd.read_csv(f, sep='\\t', header=None)\n",
    "    df.columns = ['chr', 'start', 'end', 'gene', 'chip_chr', 'chip_start', 'chip_end', 'peak_height']\n",
    "    df['area'] = df.peak_height * (df.chip_end - df.chip_start)\n",
    "    dfg = df.groupby(['chr', 'start', 'end', 'gene'])['area'].sum().reset_index()\n",
    "    of = '{0}.{1}'.format(f, 'sum')\n",
    "    dfg.to_csv(of, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "def process_signals(df1):\n",
    "#     f1 = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/treat_pileup_files.txt' \n",
    "    for idx, f in df1.iterrows():\n",
    "        f = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/H3K36me3_gb/{}'.format(f['file'])\n",
    "        integrate_signals(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# f1 = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/pileup_files/treat_intersected_files.txt'\n",
    "f1 = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/H3K36me3_gb/H3K36me3_treat_intersected_files.txt'\n",
    "df1 = get_files(f1)\n",
    "process_signals(df1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sorted_UCSC_A84614_H3K36me3_treat_pileup.bdg.i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sorted_UCSC_A84621_H3K36me3_treat_pileup.bdg.i...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                file\n",
       "0  sorted_UCSC_A84614_H3K36me3_treat_pileup.bdg.i...\n",
       "1  sorted_UCSC_A84621_H3K36me3_treat_pileup.bdg.i..."
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# # process control, sum signals\n",
    "# f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/control_files.txt'\n",
    "# df2 = get_files(f2)\n",
    "# # process_signals(f2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file_x</th>\n",
       "      <th>file_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sorted_UCSC_A84611_H3K4me3_treat_pileup.bdg</td>\n",
       "      <td>sorted_UCSC_A84611_H3K4me3_control_lambda.bdg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sorted_UCSC_A84618_H3K4me3_treat_pileup.bdg</td>\n",
       "      <td>sorted_UCSC_A84618_H3K4me3_control_lambda.bdg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        file_x  \\\n",
       "0  sorted_UCSC_A84611_H3K4me3_treat_pileup.bdg   \n",
       "1  sorted_UCSC_A84618_H3K4me3_treat_pileup.bdg   \n",
       "\n",
       "                                          file_y  \n",
       "0  sorted_UCSC_A84611_H3K4me3_control_lambda.bdg  \n",
       "1  sorted_UCSC_A84618_H3K4me3_control_lambda.bdg  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df3 = pd.merge(df1, df2, left_index=True, right_index=True)\n",
    "# df3.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def visulize_control_treat_signal():\n",
    "    # scatter plot to visualize treat vs control signal \n",
    "    num_pats = df3.shape[0]\n",
    "    plt_per_row = 8\n",
    "    nrow = math.floor(num_pats/plt_per_row)+1\n",
    "\n",
    "    fig, axes = plt.subplots(nrow, plt_per_row, sharex='col', sharey='row', figsize=(2*plt_per_row, nrow*2))\n",
    "    for idx, f in df3.iterrows():\n",
    "        f4 = '{0}{1}.intersected.sum'.format(wkdir, f['file_x'])\n",
    "        library = f['file_x'].split('_')[2]\n",
    "        f5 = '{0}{1}.intersected.sum'.format(wkdir, f['file_y'])\n",
    "        df4 = pd.read_csv(f4).set_index(['chr', 'start', 'end', 'gene'])\n",
    "        df5 = pd.read_csv(f5).set_index(['chr', 'start', 'end', 'gene'])\n",
    "        dfm = pd.merge(df4, df5, left_index=True, right_index=True)\n",
    "    #     print(dfm.head(2))\n",
    "        axrow = math.floor(idx/plt_per_row)\n",
    "        axcol = idx%plt_per_row\n",
    "        ax = axes[axrow, axcol]\n",
    "        sns.regplot(x = 'peak_height_y', y = 'peak_height_x',fit_reg=False, ax=ax, data = dfm)\n",
    "        ax.set(xlabel='control', ylabel='treat')\n",
    "        ax.set(xlim=(0, 200000), ylim=(0, 300000))\n",
    "        ax.text(50000, 250000, library)\n",
    "    plt.suptitle('H3K4me3_bdg_sum_control_vs_treat', fontsize=14)  \n",
    "    plt.tight_layout()\n",
    "    of = '{}H3K4me3_bdg_sum_control_vs_treat.png'.format(wkdir)\n",
    "    plt.savefig(of, dpi=200, box_inches='tight')\n",
    "    plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 749,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pdist?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "merge signals from all libraries into matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sorted_UCSC_A84610_H3K4me1_treat_pileup.bdg.in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sorted_UCSC_A84611_H3K4me3_treat_pileup.bdg.in...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                file\n",
       "0  sorted_UCSC_A84610_H3K4me1_treat_pileup.bdg.in...\n",
       "1  sorted_UCSC_A84611_H3K4me3_treat_pileup.bdg.in..."
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(210, 1)"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sorted_UCSC_A84621_H3K36me3_treat_pileup.bdg.intersected\n"
     ]
    }
   ],
   "source": [
    "print(df1.loc[1,:][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "marks = [\"H3K4me1\",\n",
    "\"H3K4me3\",\n",
    "\"H3K9me3\",\n",
    "\"H3K27me3\",\n",
    "\"H3K36me3\",\n",
    "\"H3K27ac\"\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# df1.file.str.contains('H3K4me3')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def merge_data(df):\n",
    "    dfm = pd.DataFrame()\n",
    "    for idx, f in df.iterrows():\n",
    "        f1 = '{0}/{1}/{2}.sum'.format(wkdir, 'pileup_files', f['file'])\n",
    "#         library = '_'.join(f['file_y'].split('_')[2:5])\n",
    "        library = f['file'].split('_')[2]\n",
    "        df1 = pd.read_csv(f1).set_index(['chr', 'start', 'end', 'gene'])\n",
    "        df1.columns = [library]\n",
    "        if dfm.empty:\n",
    "            dfm = df1.copy()\n",
    "        else:\n",
    "            dfm = pd.merge(dfm, df1, left_index=True, right_index=True)\n",
    "    return dfm\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "unsupervised clustering based on mean promoter H3K4me3 signal"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "heatmap to visualize the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# cmap = sns.light_palette('red', as_cmap=True)\n",
    "# g = sns.clustermap(dfm2, cmap = cmap)\n",
    "# plt.setp(g.ax_heatmap.yaxis.get_majorticklabels(), rotation=0)\n",
    "# pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def make_dendrogram(dfmm2, mark):\n",
    "#     dfmm2 = dfm.copy()\n",
    "    np.isnan(dfmm2.values).any()\n",
    "    np.isinf(dfmm2.values).any()\n",
    "\n",
    "    # cluster by promoter signals so the array needs to be [number of patients, number of features]\n",
    "    matrix = dfmm2.values.T\n",
    "    matrix\n",
    "\n",
    "    dist='correlation'\n",
    "    link='ward'\n",
    "    Z = linkage(pdist(matrix, dist), link)\n",
    "\n",
    "    fig, ax = plt.subplots(figsize=(12,6))\n",
    "    aa = dendrogram(Z, leaf_rotation=90.,labels = dfmm2.columns,leaf_font_size=12.,show_contracted=True)#\n",
    "#     plt.suptitle('unsupervised_clustering_{}_bdg_mean'.format, fontsize=14)  \n",
    "    plt.title('unsupervised_clustering_{}_bdg_mean'.format, fontsize=14)\n",
    "    plt.tight_layout()\n",
    "    of = '{0}unsupervised_clustering_{1}_bdg_mean.png'.format(wkdir, mark)\n",
    "    plt.savefig(of, dpi=200, box_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# df_track.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# def get_track_df(dfm2, df_meta):\n",
    "#     # get labels of the dendrogram and make track data frame\n",
    "#     # cluster by promoter signals so the array needs to be [number of patients, number of features]\n",
    "#     matrix = dfm2.values.T\n",
    "#     dist='correlation'\n",
    "#     link='ward'\n",
    "#     Z = linkage(pdist(matrix, dist), link)\n",
    "#     aa = dendrogram(Z, leaf_rotation=90.,labels = dfm2.columns,leaf_font_size=12.,show_contracted=True)\n",
    "#     den_labels = aa['ivl']\n",
    "#     adf = pd.DataFrame(den_labels)\n",
    "#     adf.columns = ['patient']\n",
    "#     df_track = pd.merge(adf, df_meta[['patient', 'HIV_status', 'Putative_histology' ]], left_on='patient', right_on='patient')#.drop(0, axis=1)\n",
    "#     df_track = df_track.set_index('patient')\n",
    "#     return (Z, df_track)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# dfm_mark.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>input DNA control  library ID</th>\n",
       "      <th>H3K4me1 total reads</th>\n",
       "      <th>H3K4me3  total reads</th>\n",
       "      <th>H3K9me3 total reads</th>\n",
       "      <th>H3K27me3 total reads</th>\n",
       "      <th>H3K36me3 total reads</th>\n",
       "      <th>H3K27ac total reads</th>\n",
       "      <th>input DNA control  total reads</th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>strelka_mutations</th>\n",
       "      <th>HPV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>A84630</td>\n",
       "      <td>78611726</td>\n",
       "      <td>60179836</td>\n",
       "      <td>76557498</td>\n",
       "      <td>90312752.0</td>\n",
       "      <td>106610232.0</td>\n",
       "      <td>80676254.0</td>\n",
       "      <td>86539886.0</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>2958282</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>A84651</td>\n",
       "      <td>85566812</td>\n",
       "      <td>74487212</td>\n",
       "      <td>93042922</td>\n",
       "      <td>89559600.0</td>\n",
       "      <td>87905340.0</td>\n",
       "      <td>84986034.0</td>\n",
       "      <td>88575282.0</td>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>1252217</td>\n",
       "      <td>Negative</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  H3K4me1 library ID H3K4me3 library ID H3K9me3 library ID  \\\n",
       "0             A84624             A84625             A84626   \n",
       "1             A84645             A84646             A84647   \n",
       "\n",
       "  H3K27me3 library ID H3K36me3 library ID H3K27ac library ID  \\\n",
       "0              A84627              A84628             A84629   \n",
       "1              A84648              A84649             A84650   \n",
       "\n",
       "  input DNA control  library ID H3K4me1 total reads H3K4me3  total reads  \\\n",
       "0                        A84630            78611726             60179836   \n",
       "1                        A84651            85566812             74487212   \n",
       "\n",
       "  H3K9me3 total reads  H3K27me3 total reads  H3K36me3 total reads  \\\n",
       "0            76557498            90312752.0           106610232.0   \n",
       "1            93042922            89559600.0            87905340.0   \n",
       "\n",
       "   H3K27ac total reads  input DNA control  total reads            patient  \\\n",
       "0           80676254.0                      86539886.0  HTMCP-03-06-02007   \n",
       "1           84986034.0                      88575282.0  HTMCP-03-06-02026   \n",
       "\n",
       "  HIV_status Putative_histology  strelka_mutations HPV_status  \n",
       "0   Negative              Adeno            2958282   Negative  \n",
       "1   Negative              Adeno            1252217   Negative  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "plot hiv and histology track"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def map_features_to_colors(df_track, map_dict1, clrs):\n",
    "#     clrs = [\"purple\", \"#E69F00\",\"#009E73\",\"#56B4E9\"]#,\"#377EB8\",\"#E41A1C\",\"#F0E442\",'black',\"#D55E00\",\"#CC79A7\"]\n",
    "#     map_dict1 =  {'Adeno': 0.15,'Squamous': 0.25, 'Negative': 0.35, 'Positive': 0.45}\n",
    "    map_dict2 = {}\n",
    "    for n, k in enumerate(map_dict1.keys()):\n",
    "        map_dict2[k] = clrs[n]\n",
    "    assert list(map_dict2.values()) == clrs\n",
    "    ndf_track = df_track.apply(lambda x: x.map(map_dict1)).fillna(0).T\n",
    "#     legends = sorted(map_dict1, key=map_dict1.get, reverse=False)\n",
    "    return (ndf_track, map_dict2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def plot_cluster_bk(numeric_track, category_track, dfm, mark, map_dict2): # df= tracks, dfm = histone mark signal df\n",
    "#     df = ncategory_track.copy() # track df\n",
    "    # dfm: histone mark signal df\n",
    "    \n",
    "    fig = plt.figure(figsize=(16, 4), facecolor='white')\n",
    "    gs = gridspec.GridSpec(3, 2, height_ratios=[0.5,1.5,8], width_ratios=[17,1], \n",
    "                           wspace=0.05, # horizontal space\n",
    "                           hspace=0.0  # vertical space\n",
    "                          )\n",
    "\n",
    "    bounds = np.arange(0.05, len(map_dict1)/10.0 + 0.1 , 0.1).tolist()\n",
    "    assert len(bounds) == len(map_dict1) + 1\n",
    "    # make a color map of fixed colors\n",
    "    cmap = colors.ListedColormap(clrs)\n",
    "    norm = colors.BoundaryNorm(bounds, cmap.N)\n",
    "    nrow = len(category_track.index)\n",
    "    ncol = len(category_track.columns)\n",
    "    print(nrow, ncol)\n",
    "\n",
    "    # add dendrogram\n",
    "    ax5 = fig.add_subplot(gs[4], facecolor='white')\n",
    "    matrix = dfm.values.T\n",
    "    dist='correlation'\n",
    "    link='ward'\n",
    "    Z = linkage(pdist(matrix, dist), link)\n",
    "    aa = dendrogram(Z, leaf_rotation=90.,labels = dfm.columns,leaf_font_size=12.,show_contracted=True)#\n",
    "    den_labels = aa['ivl']\n",
    "    adf = pd.DataFrame(den_labels)\n",
    "    adf.columns = ['patient']\n",
    "    c_track = pd.merge(adf, category_track.T, left_on='patient', right_index=True)#.drop(0, axis=1)\n",
    "    c_track = c_track.set_index('patient')\n",
    "    c_track = c_track.T\n",
    "    plt.suptitle('unsupervised_clustering_{}_bdg_mean'.format(mark), fontsize=14)  \n",
    "    print(den_labels)\n",
    "    # plt.tight_layout()\n",
    "\n",
    "    \n",
    "    # plot category track\n",
    "    ax3 = fig.add_subplot(gs[2])\n",
    "    img = ax3.imshow(c_track.values, interpolation='none', origin='upper', aspect='auto',\n",
    "                        cmap=cmap, norm=norm)\n",
    "    _ = plt.yticks(np.arange(0, nrow, 1), c_track.index, fontsize=8)\n",
    "    _ = plt.xticks(np.arange(0, ncol, 1), c_track.columns, fontsize=8, rotation=90)\n",
    "\n",
    "    _ = ax3.set_yticks(np.arange(-.5, nrow, 1), minor=True);\n",
    "    _ = ax3.set_xticks(np.arange(-.5, ncol, 1), minor=True);\n",
    "\n",
    "    # Gridlines based on minor ticks\n",
    "    _ = ax3.grid(b=True, which='minor', color='gray', linestyle='--', linewidth=0.3)\n",
    "    # hide major tick grid\n",
    "    _ = ax3.grid(b=True, which='major', color='r', linestyle='-', linewidth=0)\n",
    "\n",
    "    ax3.spines['right'].set_visible(True)\n",
    "    ax3.spines['top'].set_visible(True)\n",
    "    ax3.set_xticklabels('')\n",
    "    print(c_track.columns)\n",
    "    \n",
    "    \n",
    "    # plot numeric track\n",
    "    ax1 = fig.add_subplot(gs[0])\n",
    "    numeric_track = numeric_track.set_index('patient').astype(float)\n",
    "    n_track = pd.merge(adf, numeric_track, left_on='patient', right_index=True)#.drop(0, axis=1)\n",
    "    n_track = n_track.set_index('patient')\n",
    "    n_track = n_track.T\n",
    "    nrow = len(n_track.index)\n",
    "    ncol = len(n_track.columns)\n",
    "    print(nrow, ncol)\n",
    "    img = ax1.imshow(n_track.values, interpolation='none', origin='upper', aspect='auto', cmap='gray')\n",
    "    _ = plt.yticks(np.arange(0, nrow, 1), n_track.index, fontsize=8)\n",
    "    _ = plt.xticks(np.arange(0, ncol, 1), n_track.columns, fontsize=8, rotation=90)\n",
    "\n",
    "    _ = ax1.set_yticks(np.arange(-.5, nrow, 1), minor=True);\n",
    "    _ = ax1.set_xticks(np.arange(-.5, ncol, 1), minor=True);\n",
    "\n",
    "    # Gridlines based on minor ticks\n",
    "    _ = ax1.grid(b=True, which='minor', color='gray', linestyle='--', linewidth=0.3)\n",
    "    # hide major tick grid\n",
    "    _ = ax1.grid(b=True, which='major', color='r', linestyle='-', linewidth=0)\n",
    "\n",
    "    ax1.spines['right'].set_visible(True)\n",
    "    ax1.spines['top'].set_visible(True)\n",
    "    ax1.set_xticklabels('')\n",
    "    print(n_track.columns)\n",
    "    \n",
    "\n",
    "    # color bar for numeric_track\n",
    "    ax2 = fig.add_subplot(gs[1])\n",
    "    fig.colorbar(img)\n",
    "    \n",
    "    # make sure columns in numeric, category, and dendrogram match\n",
    "    assert den_labels == c_track.columns.tolist()\n",
    "    assert den_labels == n_track.columns.tolist()\n",
    "    \n",
    "    \n",
    "    # add track legend\n",
    "    ax6 = fig.add_subplot(gs[5], facecolor='white')\n",
    "    # plot mutation legend\n",
    "    k = 1\n",
    "    for key in map_dict2:\n",
    "        if key !=0: # don't plot key=0 which means no mutation\n",
    "            clr = map_dict2[key]\n",
    "            rect = patches.Rectangle((0,k/10), 0.3, 0.05, linewidth=0.01,edgecolor=clr,facecolor=clr)\n",
    "            _ = ax6.text(0.35, k/10+.02, key, ha='left', va='center', fontsize=8)\n",
    "            _ = ax6.add_patch(rect)\n",
    "        k+=1\n",
    "    ax6.get_xaxis().set_ticks([])\n",
    "    ax6.get_yaxis().set_ticks([])\n",
    "\n",
    "    ax6.set_title('')\n",
    "\n",
    "\n",
    "    of = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/unsupervised_clustering_{}_bdg_mean.png'.format(mark)\n",
    "    %time plt.savefig(of, bbox_inches='tight',dpi=300)\n",
    "    plt.show()\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def plot_cluster(numeric_track, category_track, dfm, mark, map_dict2): # df= tracks, dfm = histone mark signal df\n",
    "#     df = ncategory_track.copy() # track df\n",
    "    # dfm: histone mark signal df\n",
    "    \n",
    "    fig = plt.figure(figsize=(16, 8), facecolor='white')\n",
    "    gs = gridspec.GridSpec(1, 2, width_ratios=[17,1], \n",
    "                           wspace=0.05, # horizontal space\n",
    "                           hspace=0.0  # vertical space\n",
    "                          )\n",
    "\n",
    "    bounds = np.arange(0.05, len(map_dict1)/10.0 + 0.1 , 0.1).tolist()\n",
    "    assert len(bounds) == len(map_dict1) + 1\n",
    "    # make a color map of fixed colors\n",
    "    cmap = colors.ListedColormap(clrs)\n",
    "    norm = colors.BoundaryNorm(bounds, cmap.N)\n",
    "    nrow = len(category_track.index)\n",
    "    ncol = len(category_track.columns)\n",
    "    print(nrow, ncol)\n",
    "\n",
    "    # left panel 3x1 for n_track, c_trak, and dendrogram\n",
    "    gs01 = gridspec.GridSpecFromSubplotSpec(3, 1,  height_ratios=[.5, 14, 4], subplot_spec=gs[0],  hspace=0.0)\n",
    "    \n",
    "     # add dendrogram\n",
    "    ax1_3 = fig.add_subplot(gs01[2])\n",
    "    ax1_3.set_title(\"\")\n",
    "#     ax1_3 = fig.add_subplot(gs[4], facecolor='white')\n",
    "    matrix = dfm.values.T\n",
    "    dist='correlation'\n",
    "    link='ward'\n",
    "    Z = linkage(pdist(matrix, dist), link)\n",
    "    aa = dendrogram(Z, leaf_rotation=90.,labels = dfm.columns,leaf_font_size=12.,show_contracted=True)#\n",
    "    den_labels = aa['ivl']\n",
    "    adf = pd.DataFrame(den_labels)\n",
    "    adf.columns = ['patient']\n",
    "    c_track = pd.merge(adf, category_track.T, left_on='patient', right_index=True)#.drop(0, axis=1)\n",
    "    c_track = c_track.set_index('patient')\n",
    "    c_track = c_track.T\n",
    "    plt.suptitle('unsupervised_clustering_{}_bdg_area'.format(mark), fontsize=14)  \n",
    "#     print(den_labels)\n",
    "    \n",
    "\n",
    "    \n",
    "    \n",
    "    # plot category track\n",
    "    ax1_2 = fig.add_subplot(gs01[1], facecolor='white')\n",
    "#     ax1_2.set_title(\"\")\n",
    "#     print(c_track)\n",
    "    img = ax1_2.imshow(c_track.values, interpolation='none', origin='upper', aspect='auto',\n",
    "                        cmap=cmap, norm=norm)\n",
    "    _ = plt.yticks(np.arange(0, nrow, 1), c_track.index, fontsize=8)\n",
    "    _ = plt.xticks(np.arange(0, ncol, 1), c_track.columns, fontsize=8, rotation=90)\n",
    "\n",
    "    _ = ax1_2.set_yticks(np.arange(-.5, nrow, 1), minor=True);\n",
    "    _ = ax1_2.set_xticks(np.arange(-.5, ncol, 1), minor=True);\n",
    "\n",
    "    # Gridlines based on minor ticks\n",
    "    _ = ax1_2.grid(b=True, which='minor', color='gray', linestyle='--', linewidth=0.3)\n",
    "    # hide major tick grid\n",
    "    _ = ax1_2.grid(b=True, which='major', color='r', linestyle='-', linewidth=0)\n",
    "\n",
    "    ax1_2.spines['right'].set_visible(True)\n",
    "    ax1_2.spines['top'].set_visible(True)\n",
    "    ax1_2.set_xticklabels('')\n",
    "#     print(c_track.columns)\n",
    "    \n",
    "\n",
    "    # plot numeric track\n",
    "    ax1_1 = fig.add_subplot(gs01[0], facecolor='white')\n",
    "#     ax1_1.set_title(\"\")\n",
    "    numeric_track = numeric_track.set_index('patient').astype(float)\n",
    "    n_track = pd.merge(adf, numeric_track, left_on='patient', right_index=True)#.drop(0, axis=1)\n",
    "    n_track = n_track.set_index('patient')\n",
    "    n_track = n_track.T\n",
    "    nrow = len(n_track.index)\n",
    "    ncol = len(n_track.columns)\n",
    "    print(nrow, ncol)\n",
    "    img1_1 = ax1_1.imshow(n_track.values, interpolation='none', origin='upper', aspect='auto', cmap='Reds')\n",
    "    _ = plt.yticks(np.arange(0, nrow, 1), n_track.index, fontsize=8)\n",
    "    _ = plt.xticks(np.arange(0, ncol, 1), n_track.columns, fontsize=8, rotation=90)\n",
    "\n",
    "    _ = ax1_1.set_yticks(np.arange(-.5, nrow, 1), minor=True);\n",
    "    _ = ax1_1.set_xticks(np.arange(-.5, ncol, 1), minor=True);\n",
    "\n",
    "    # Gridlines based on minor ticks\n",
    "    _ = ax1_1.grid(b=True, which='minor', color='gray', linestyle='--', linewidth=0.3)\n",
    "    # hide major tick grid\n",
    "    _ = ax1_1.grid(b=True, which='major', color='r', linestyle='-', linewidth=0)\n",
    "\n",
    "    ax1_1.spines['right'].set_visible(True)\n",
    "    ax1_1.spines['top'].set_visible(True)\n",
    "    ax1_1.set_xticklabels('')\n",
    "#     print(n_track.columns)\n",
    "\n",
    "    \n",
    "\n",
    "    # right panel 2x1 for lengends, colorbar and rectangles\n",
    "    gs02 = gridspec.GridSpecFromSubplotSpec(2, 1,  height_ratios=[1, 1], subplot_spec=gs[1])\n",
    "    \n",
    "    # color bar for numeric_track \n",
    "    ax2_1 = fig.add_subplot(gs02[0], facecolor='white')\n",
    "    ax2_1.set_title(\"mutations_log10\")\n",
    "        \n",
    "    _ = plt.colorbar(img1_1, cax=ax2_1, orientation= 'vertical', label='')#, aspect=40) \n",
    "\n",
    "    \n",
    "    # add retangle legend for category track\n",
    "    ax2_2 = fig.add_subplot(gs02[1], facecolor='white')\n",
    "    ax2_2.set_title(\"rectangles\")\n",
    "        \n",
    "#     ax6 = fig.add_subplot(gs[5], facecolor='white')\n",
    "    # plot mutation legend\n",
    "    k = 1\n",
    "    for key in map_dict2:\n",
    "        if key !=0: # don't plot key=0 which means no mutation\n",
    "            clr = map_dict2[key]\n",
    "            rect = patches.Rectangle((0,k/7), 0.3, 0.1, linewidth=0.01,edgecolor=clr,facecolor=clr)\n",
    "            _ = ax2_2.text(0.35, k/7+.07, key, ha='left', va='center', fontsize=8)\n",
    "            _ = ax2_2.add_patch(rect)\n",
    "        k+=1\n",
    "    ax2_2.get_xaxis().set_ticks([])\n",
    "    ax2_2.get_yaxis().set_ticks([])\n",
    "\n",
    "    ax2_2.set_title('')\n",
    "\n",
    "  \n",
    "    \n",
    "    # make sure columns in numeric, category, and dendrogram match\n",
    "#     print(den_labels)\n",
    "#     print(c_track.columns.tolist())\n",
    "    assert den_labels == c_track.columns.tolist()\n",
    "    assert den_labels == n_track.columns.tolist()\n",
    "#     plt.tight_layout()\n",
    "    \n",
    "\n",
    "\n",
    "#     of = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/unsupervised_clustering_{}_bdg_area.png'.format(mark)\n",
    "    of = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/H3K36me3_gb/unsupervised_clustering_{}_bdg_area.png'.format(mark)\n",
    "    %time plt.savefig(of, bbox_inches='tight',dpi=300)\n",
    "    plt.show()\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>HPV_status</th>\n",
       "      <th>Remodeller</th>\n",
       "      <th>Reader</th>\n",
       "      <th>Writer</th>\n",
       "      <th>Eraser</th>\n",
       "      <th>SWI/SNF remodeling complex</th>\n",
       "      <th>SWI/SNF family</th>\n",
       "      <th>...</th>\n",
       "      <th>ISWI chromatin remdelling</th>\n",
       "      <th>HAT complexes</th>\n",
       "      <th>NuRD complex</th>\n",
       "      <th>compass complex</th>\n",
       "      <th>HDAC complex</th>\n",
       "      <th>GNAT family HAT complexes</th>\n",
       "      <th>MYST family HAT complexes</th>\n",
       "      <th>RNF20/40 complex</th>\n",
       "      <th>SWI/SNF  remodeling complex</th>\n",
       "      <th>TFIID basal transcription factor complex or PCAF histone acetylase complex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient HIV_status Putative_histology HPV_status Remodeller  \\\n",
       "0  HTMCP-03-06-02007   Negative              Adeno   Negative   Positive   \n",
       "1  HTMCP-03-06-02026   Negative              Adeno   Negative   Positive   \n",
       "\n",
       "     Reader    Writer    Eraser SWI/SNF remodeling complex SWI/SNF family  \\\n",
       "0  Positive  Positive  Positive                   Positive       Positive   \n",
       "1  Positive  Positive  Positive                   Positive       Positive   \n",
       "\n",
       "                                     ...                                      \\\n",
       "0                                    ...                                       \n",
       "1                                    ...                                       \n",
       "\n",
       "  ISWI chromatin remdelling HAT complexes NuRD complex  compass complex  \\\n",
       "0                  Positive      Positive      Positive        Positive   \n",
       "1                  Negative      Positive      Positive        Positive   \n",
       "\n",
       "  HDAC complex GNAT family HAT complexes MYST family HAT complexes  \\\n",
       "0     Positive                  Positive                  Positive   \n",
       "1     Positive                  Positive                  Positive   \n",
       "\n",
       "  RNF20/40 complex  SWI/SNF  remodeling complex  \\\n",
       "0          Positive                    Positive   \n",
       "1          Negative                    Negative   \n",
       "\n",
       "  TFIID basal transcription factor complex or PCAF histone acetylase complex  \n",
       "0                                           Positive                          \n",
       "1                                           Positive                          \n",
       "\n",
       "[2 rows x 28 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "category_track.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>log_strelka_mutations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>6.47104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "      <td>6.09768</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient  log_strelka_mutations\n",
       "0  HTMCP-03-06-02007                6.47104\n",
       "1  HTMCP-03-06-02026                6.09768"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "numeric_track.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "merging signal for mark: H3K36me3\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'df1' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-58-2aa3ff1db518>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mmark\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m\"H3K36me3\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;31m#marks:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'merging signal for mark: {}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmark\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m     \u001b[0mdf_mark\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'file'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcontains\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmark\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0mdfm_mark\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmerge_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_mark\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mdfm_mark\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdfm_mark\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0;36m4000\u001b[0m \u001b[0;31m# length of the promoter regaionx\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df1' is not defined"
     ]
    }
   ],
   "source": [
    "for mark in [\"H3K36me3\"]:#marks:\n",
    "    print('merging signal for mark: {}'.format(mark))\n",
    "    df_mark = df1[df1['file'].str.contains(mark)]\n",
    "    dfm_mark = merge_data(df_mark)\n",
    "    dfm_mark = dfm_mark/4000 # length of the promoter regaionx\n",
    "    lib_name = '{} library ID'.format(mark)\n",
    "    df_meta1 = df_meta[[lib_name, 'patient']].set_index(lib_name)\n",
    "    d = df_meta1.to_dict()\n",
    "    dfm_mark.columns = dfm_mark.columns.to_series().map(d['patient'])\n",
    "    cat_track = category_track.set_index('patient')\n",
    "    clrs = [\"purple\", \"#E69F00\",\"#009E73\",\"#56B4E9\"]#,\"#377EB8\",\"#E41A1C\",\"#F0E442\",'black',\"#D55E00\",\"#CC79A7\"]\n",
    "    map_dict1 =  {'Adeno': 0.15,'Squamous': 0.25, 'Negative': 0.35, 'Positive': 0.45}\n",
    "    ncategory_track, map_dict2 = map_features_to_colors(cat_track, map_dict1, clrs)\n",
    "    plot_cluster(numeric_track, ncategory_track, dfm_mark, mark, map_dict2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02040</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02054</th>\n",
       "      <th>HTMCP-03-06-02058</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02210</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02202</th>\n",
       "      <th>HTMCP-03-06-02182</th>\n",
       "      <th>HTMCP-03-06-02260</th>\n",
       "      <th>HTMCP-03-06-02266</th>\n",
       "      <th>HTMCP-03-06-02268</th>\n",
       "      <th>HTMCP-03-06-02320</th>\n",
       "      <th>HTMCP-03-06-02330</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">chr1</th>\n",
       "      <th>34081</th>\n",
       "      <th>38080</th>\n",
       "      <th>ENSG00000237613</th>\n",
       "      <td>2.96911</td>\n",
       "      <td>0.593800</td>\n",
       "      <td>0.517766</td>\n",
       "      <td>0.572157</td>\n",
       "      <td>1.86325</td>\n",
       "      <td>0.99825</td>\n",
       "      <td>1.511414</td>\n",
       "      <td>1.577847</td>\n",
       "      <td>1.154083</td>\n",
       "      <td>0.988018</td>\n",
       "      <td>...</td>\n",
       "      <td>1.62525</td>\n",
       "      <td>1.92225</td>\n",
       "      <td>2.814546</td>\n",
       "      <td>1.242604</td>\n",
       "      <td>1.888</td>\n",
       "      <td>1.39125</td>\n",
       "      <td>0.81125</td>\n",
       "      <td>1.0790</td>\n",
       "      <td>1.003871</td>\n",
       "      <td>1.423199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67092</th>\n",
       "      <th>71091</th>\n",
       "      <th>ENSG00000186092</th>\n",
       "      <td>3.51677</td>\n",
       "      <td>0.584748</td>\n",
       "      <td>0.690354</td>\n",
       "      <td>0.289419</td>\n",
       "      <td>2.39625</td>\n",
       "      <td>0.21825</td>\n",
       "      <td>0.951870</td>\n",
       "      <td>3.199019</td>\n",
       "      <td>0.164869</td>\n",
       "      <td>0.982336</td>\n",
       "      <td>...</td>\n",
       "      <td>2.81525</td>\n",
       "      <td>2.03700</td>\n",
       "      <td>2.040106</td>\n",
       "      <td>0.333469</td>\n",
       "      <td>1.744</td>\n",
       "      <td>0.84800</td>\n",
       "      <td>0.87200</td>\n",
       "      <td>0.4425</td>\n",
       "      <td>1.005059</td>\n",
       "      <td>2.165821</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  HTMCP-03-06-02001  HTMCP-03-06-02006  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613            2.96911           0.593800   \n",
       "     67092 71091 ENSG00000186092            3.51677           0.584748   \n",
       "\n",
       "                                  HTMCP-03-06-02007  HTMCP-03-06-02012  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613           0.517766           0.572157   \n",
       "     67092 71091 ENSG00000186092           0.690354           0.289419   \n",
       "\n",
       "                                  HTMCP-03-06-02020  HTMCP-03-06-02026  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613            1.86325            0.99825   \n",
       "     67092 71091 ENSG00000186092            2.39625            0.21825   \n",
       "\n",
       "                                  HTMCP-03-06-02040  HTMCP-03-06-02042  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613           1.511414           1.577847   \n",
       "     67092 71091 ENSG00000186092           0.951870           3.199019   \n",
       "\n",
       "                                  HTMCP-03-06-02054  HTMCP-03-06-02058  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613           1.154083           0.988018   \n",
       "     67092 71091 ENSG00000186092           0.164869           0.982336   \n",
       "\n",
       "                                        ...          HTMCP-03-06-02210  \\\n",
       "chr  start end   gene                   ...                              \n",
       "chr1 34081 38080 ENSG00000237613        ...                    1.62525   \n",
       "     67092 71091 ENSG00000186092        ...                    2.81525   \n",
       "\n",
       "                                  HTMCP-03-06-02214  HTMCP-03-06-02202  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613            1.92225           2.814546   \n",
       "     67092 71091 ENSG00000186092            2.03700           2.040106   \n",
       "\n",
       "                                  HTMCP-03-06-02182  HTMCP-03-06-02260  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613           1.242604              1.888   \n",
       "     67092 71091 ENSG00000186092           0.333469              1.744   \n",
       "\n",
       "                                  HTMCP-03-06-02266  HTMCP-03-06-02268  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613            1.39125            0.81125   \n",
       "     67092 71091 ENSG00000186092            0.84800            0.87200   \n",
       "\n",
       "                                  HTMCP-03-06-02320  HTMCP-03-06-02330  \\\n",
       "chr  start end   gene                                                    \n",
       "chr1 34081 38080 ENSG00000237613             1.0790           1.003871   \n",
       "     67092 71091 ENSG00000186092             0.4425           1.005059   \n",
       "\n",
       "                                  HTMCP-03-06-02332  \n",
       "chr  start end   gene                                \n",
       "chr1 34081 38080 ENSG00000237613           1.423199  \n",
       "     67092 71091 ENSG00000186092           2.165821  \n",
       "\n",
       "[2 rows x 35 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfm_mark.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(20059, 35)"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfm_mark.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matplotlib.is_interactive()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>A84611_H3K4me3_control</th>\n",
       "      <th>A84618_H3K4me3_control</th>\n",
       "      <th>A84625_H3K4me3_control</th>\n",
       "      <th>A84632_H3K4me3_control</th>\n",
       "      <th>A84639_H3K4me3_control</th>\n",
       "      <th>A84646_H3K4me3_control</th>\n",
       "      <th>A84653_H3K4me3_control</th>\n",
       "      <th>A84660_H3K4me3_control</th>\n",
       "      <th>A84667_H3K4me3_control</th>\n",
       "      <th>A84674_H3K4me3_control</th>\n",
       "      <th>...</th>\n",
       "      <th>A84786_H3K4me3_control</th>\n",
       "      <th>A84793_H3K4me3_control</th>\n",
       "      <th>A84800_H3K4me3_control</th>\n",
       "      <th>A84807_H3K4me3_control</th>\n",
       "      <th>A84821_H3K4me3_control</th>\n",
       "      <th>A84828_H3K4me3_control</th>\n",
       "      <th>A84835_H3K4me3_control</th>\n",
       "      <th>A84842_H3K4me3_control</th>\n",
       "      <th>A84849_H3K4me3_control</th>\n",
       "      <th>A84856_H3K4me3_control</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">chr1</th>\n",
       "      <th>34081</th>\n",
       "      <th>38080</th>\n",
       "      <th>ENSG00000237613</th>\n",
       "      <td>0.427685</td>\n",
       "      <td>0.310419</td>\n",
       "      <td>0.252474</td>\n",
       "      <td>0.204968</td>\n",
       "      <td>0.228736</td>\n",
       "      <td>0.146941</td>\n",
       "      <td>0.130224</td>\n",
       "      <td>0.237918</td>\n",
       "      <td>0.352411</td>\n",
       "      <td>0.313879</td>\n",
       "      <td>...</td>\n",
       "      <td>0.383520</td>\n",
       "      <td>0.098699</td>\n",
       "      <td>0.174823</td>\n",
       "      <td>0.342673</td>\n",
       "      <td>0.142027</td>\n",
       "      <td>0.178795</td>\n",
       "      <td>0.125970</td>\n",
       "      <td>0.091436</td>\n",
       "      <td>0.127949</td>\n",
       "      <td>0.162249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67092</th>\n",
       "      <th>71091</th>\n",
       "      <th>ENSG00000186092</th>\n",
       "      <td>0.061214</td>\n",
       "      <td>0.055682</td>\n",
       "      <td>0.113265</td>\n",
       "      <td>0.049366</td>\n",
       "      <td>0.203678</td>\n",
       "      <td>0.133603</td>\n",
       "      <td>0.105407</td>\n",
       "      <td>0.316299</td>\n",
       "      <td>0.073475</td>\n",
       "      <td>0.164595</td>\n",
       "      <td>...</td>\n",
       "      <td>0.377893</td>\n",
       "      <td>0.108213</td>\n",
       "      <td>0.135646</td>\n",
       "      <td>0.219149</td>\n",
       "      <td>0.084153</td>\n",
       "      <td>0.147442</td>\n",
       "      <td>0.077533</td>\n",
       "      <td>0.127165</td>\n",
       "      <td>0.068792</td>\n",
       "      <td>0.065099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365641</th>\n",
       "      <th>369640</th>\n",
       "      <th>ENSG00000235249</th>\n",
       "      <td>0.041954</td>\n",
       "      <td>0.009753</td>\n",
       "      <td>0.007909</td>\n",
       "      <td>0.034107</td>\n",
       "      <td>0.041958</td>\n",
       "      <td>0.016568</td>\n",
       "      <td>0.022140</td>\n",
       "      <td>0.013750</td>\n",
       "      <td>0.018340</td>\n",
       "      <td>0.025527</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018320</td>\n",
       "      <td>0.045702</td>\n",
       "      <td>0.010178</td>\n",
       "      <td>0.030097</td>\n",
       "      <td>0.026634</td>\n",
       "      <td>0.041925</td>\n",
       "      <td>0.033472</td>\n",
       "      <td>0.013969</td>\n",
       "      <td>0.030208</td>\n",
       "      <td>0.029260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620053</th>\n",
       "      <th>624052</th>\n",
       "      <th>ENSG00000185097</th>\n",
       "      <td>0.043833</td>\n",
       "      <td>0.019739</td>\n",
       "      <td>0.003780</td>\n",
       "      <td>0.044374</td>\n",
       "      <td>0.027462</td>\n",
       "      <td>0.020352</td>\n",
       "      <td>0.058604</td>\n",
       "      <td>0.044327</td>\n",
       "      <td>0.017215</td>\n",
       "      <td>0.016336</td>\n",
       "      <td>...</td>\n",
       "      <td>0.011956</td>\n",
       "      <td>0.058394</td>\n",
       "      <td>0.042427</td>\n",
       "      <td>0.034761</td>\n",
       "      <td>0.034794</td>\n",
       "      <td>0.030486</td>\n",
       "      <td>0.029694</td>\n",
       "      <td>0.011169</td>\n",
       "      <td>0.032295</td>\n",
       "      <td>0.039054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>858261</th>\n",
       "      <th>862260</th>\n",
       "      <th>ENSG00000187634</th>\n",
       "      <td>0.559055</td>\n",
       "      <td>0.287142</td>\n",
       "      <td>0.111270</td>\n",
       "      <td>0.226632</td>\n",
       "      <td>0.014630</td>\n",
       "      <td>0.094569</td>\n",
       "      <td>0.020124</td>\n",
       "      <td>0.032539</td>\n",
       "      <td>0.262516</td>\n",
       "      <td>0.133116</td>\n",
       "      <td>...</td>\n",
       "      <td>0.019815</td>\n",
       "      <td>0.025514</td>\n",
       "      <td>0.079165</td>\n",
       "      <td>0.007100</td>\n",
       "      <td>0.025245</td>\n",
       "      <td>0.025352</td>\n",
       "      <td>0.042925</td>\n",
       "      <td>0.083198</td>\n",
       "      <td>0.055739</td>\n",
       "      <td>0.110546</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    A84611_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.427685   \n",
       "     67092  71091  ENSG00000186092                0.061214   \n",
       "     365641 369640 ENSG00000235249                0.041954   \n",
       "     620053 624052 ENSG00000185097                0.043833   \n",
       "     858261 862260 ENSG00000187634                0.559055   \n",
       "\n",
       "                                    A84618_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.310419   \n",
       "     67092  71091  ENSG00000186092                0.055682   \n",
       "     365641 369640 ENSG00000235249                0.009753   \n",
       "     620053 624052 ENSG00000185097                0.019739   \n",
       "     858261 862260 ENSG00000187634                0.287142   \n",
       "\n",
       "                                    A84625_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.252474   \n",
       "     67092  71091  ENSG00000186092                0.113265   \n",
       "     365641 369640 ENSG00000235249                0.007909   \n",
       "     620053 624052 ENSG00000185097                0.003780   \n",
       "     858261 862260 ENSG00000187634                0.111270   \n",
       "\n",
       "                                    A84632_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.204968   \n",
       "     67092  71091  ENSG00000186092                0.049366   \n",
       "     365641 369640 ENSG00000235249                0.034107   \n",
       "     620053 624052 ENSG00000185097                0.044374   \n",
       "     858261 862260 ENSG00000187634                0.226632   \n",
       "\n",
       "                                    A84639_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.228736   \n",
       "     67092  71091  ENSG00000186092                0.203678   \n",
       "     365641 369640 ENSG00000235249                0.041958   \n",
       "     620053 624052 ENSG00000185097                0.027462   \n",
       "     858261 862260 ENSG00000187634                0.014630   \n",
       "\n",
       "                                    A84646_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.146941   \n",
       "     67092  71091  ENSG00000186092                0.133603   \n",
       "     365641 369640 ENSG00000235249                0.016568   \n",
       "     620053 624052 ENSG00000185097                0.020352   \n",
       "     858261 862260 ENSG00000187634                0.094569   \n",
       "\n",
       "                                    A84653_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.130224   \n",
       "     67092  71091  ENSG00000186092                0.105407   \n",
       "     365641 369640 ENSG00000235249                0.022140   \n",
       "     620053 624052 ENSG00000185097                0.058604   \n",
       "     858261 862260 ENSG00000187634                0.020124   \n",
       "\n",
       "                                    A84660_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.237918   \n",
       "     67092  71091  ENSG00000186092                0.316299   \n",
       "     365641 369640 ENSG00000235249                0.013750   \n",
       "     620053 624052 ENSG00000185097                0.044327   \n",
       "     858261 862260 ENSG00000187634                0.032539   \n",
       "\n",
       "                                    A84667_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.352411   \n",
       "     67092  71091  ENSG00000186092                0.073475   \n",
       "     365641 369640 ENSG00000235249                0.018340   \n",
       "     620053 624052 ENSG00000185097                0.017215   \n",
       "     858261 862260 ENSG00000187634                0.262516   \n",
       "\n",
       "                                    A84674_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.313879   \n",
       "     67092  71091  ENSG00000186092                0.164595   \n",
       "     365641 369640 ENSG00000235249                0.025527   \n",
       "     620053 624052 ENSG00000185097                0.016336   \n",
       "     858261 862260 ENSG00000187634                0.133116   \n",
       "\n",
       "                                             ...            \\\n",
       "chr  start  end    gene                      ...             \n",
       "chr1 34081  38080  ENSG00000237613           ...             \n",
       "     67092  71091  ENSG00000186092           ...             \n",
       "     365641 369640 ENSG00000235249           ...             \n",
       "     620053 624052 ENSG00000185097           ...             \n",
       "     858261 862260 ENSG00000187634           ...             \n",
       "\n",
       "                                    A84786_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.383520   \n",
       "     67092  71091  ENSG00000186092                0.377893   \n",
       "     365641 369640 ENSG00000235249                0.018320   \n",
       "     620053 624052 ENSG00000185097                0.011956   \n",
       "     858261 862260 ENSG00000187634                0.019815   \n",
       "\n",
       "                                    A84793_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.098699   \n",
       "     67092  71091  ENSG00000186092                0.108213   \n",
       "     365641 369640 ENSG00000235249                0.045702   \n",
       "     620053 624052 ENSG00000185097                0.058394   \n",
       "     858261 862260 ENSG00000187634                0.025514   \n",
       "\n",
       "                                    A84800_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.174823   \n",
       "     67092  71091  ENSG00000186092                0.135646   \n",
       "     365641 369640 ENSG00000235249                0.010178   \n",
       "     620053 624052 ENSG00000185097                0.042427   \n",
       "     858261 862260 ENSG00000187634                0.079165   \n",
       "\n",
       "                                    A84807_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.342673   \n",
       "     67092  71091  ENSG00000186092                0.219149   \n",
       "     365641 369640 ENSG00000235249                0.030097   \n",
       "     620053 624052 ENSG00000185097                0.034761   \n",
       "     858261 862260 ENSG00000187634                0.007100   \n",
       "\n",
       "                                    A84821_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.142027   \n",
       "     67092  71091  ENSG00000186092                0.084153   \n",
       "     365641 369640 ENSG00000235249                0.026634   \n",
       "     620053 624052 ENSG00000185097                0.034794   \n",
       "     858261 862260 ENSG00000187634                0.025245   \n",
       "\n",
       "                                    A84828_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.178795   \n",
       "     67092  71091  ENSG00000186092                0.147442   \n",
       "     365641 369640 ENSG00000235249                0.041925   \n",
       "     620053 624052 ENSG00000185097                0.030486   \n",
       "     858261 862260 ENSG00000187634                0.025352   \n",
       "\n",
       "                                    A84835_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.125970   \n",
       "     67092  71091  ENSG00000186092                0.077533   \n",
       "     365641 369640 ENSG00000235249                0.033472   \n",
       "     620053 624052 ENSG00000185097                0.029694   \n",
       "     858261 862260 ENSG00000187634                0.042925   \n",
       "\n",
       "                                    A84842_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.091436   \n",
       "     67092  71091  ENSG00000186092                0.127165   \n",
       "     365641 369640 ENSG00000235249                0.013969   \n",
       "     620053 624052 ENSG00000185097                0.011169   \n",
       "     858261 862260 ENSG00000187634                0.083198   \n",
       "\n",
       "                                    A84849_H3K4me3_control  \\\n",
       "chr  start  end    gene                                      \n",
       "chr1 34081  38080  ENSG00000237613                0.127949   \n",
       "     67092  71091  ENSG00000186092                0.068792   \n",
       "     365641 369640 ENSG00000235249                0.030208   \n",
       "     620053 624052 ENSG00000185097                0.032295   \n",
       "     858261 862260 ENSG00000187634                0.055739   \n",
       "\n",
       "                                    A84856_H3K4me3_control  \n",
       "chr  start  end    gene                                     \n",
       "chr1 34081  38080  ENSG00000237613                0.162249  \n",
       "     67092  71091  ENSG00000186092                0.065099  \n",
       "     365641 369640 ENSG00000235249                0.029260  \n",
       "     620053 624052 ENSG00000185097                0.039054  \n",
       "     858261 862260 ENSG00000187634                0.110546  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df32 = pd.DataFrame(df3.file_y)\n",
    "dfm2 = merge_data(df32)\n",
    "dfm2 = dfm2/4000 # length of the promoter regaion\n",
    "dfm2.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "make input file for ChromHMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Order</th>\n",
       "      <th>Biospecimen ID</th>\n",
       "      <th>HIV status</th>\n",
       "      <th>Epigenome modifier mutation status</th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>...</th>\n",
       "      <th>H3K4me3  % aligned</th>\n",
       "      <th>H3K9me3 % aligned</th>\n",
       "      <th>H3K27me3 % aligned</th>\n",
       "      <th>H3K36me3 % aligned</th>\n",
       "      <th>H3K27ac % aligned</th>\n",
       "      <th>input DNA control   % aligned</th>\n",
       "      <th>internal sample ID</th>\n",
       "      <th>pool ID</th>\n",
       "      <th>Notes</th>\n",
       "      <th># lanes 7 PE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>HTMCP-03-06-02007-01A</td>\n",
       "      <td>negative</td>\n",
       "      <td>mutated</td>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>...</td>\n",
       "      <td>98.13</td>\n",
       "      <td>93.1</td>\n",
       "      <td>97.93</td>\n",
       "      <td>98.57</td>\n",
       "      <td>98.10</td>\n",
       "      <td>96.26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IX5978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>HTMCP-03-06-02026-01A</td>\n",
       "      <td>negative</td>\n",
       "      <td>mutated</td>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>...</td>\n",
       "      <td>97.68</td>\n",
       "      <td>93.89</td>\n",
       "      <td>97.84</td>\n",
       "      <td>98.17</td>\n",
       "      <td>97.86</td>\n",
       "      <td>95.56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IX5981</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Order         Biospecimen ID HIV status Epigenome modifier mutation status  \\\n",
       "0      1  HTMCP-03-06-02007-01A   negative                            mutated   \n",
       "1      2  HTMCP-03-06-02026-01A   negative                            mutated   \n",
       "\n",
       "  H3K4me1 library ID H3K4me3 library ID H3K9me3 library ID  \\\n",
       "0             A84624             A84625             A84626   \n",
       "1             A84645             A84646             A84647   \n",
       "\n",
       "  H3K27me3 library ID H3K36me3 library ID H3K27ac library ID      ...       \\\n",
       "0              A84627              A84628             A84629      ...        \n",
       "1              A84648              A84649             A84650      ...        \n",
       "\n",
       "  H3K4me3  % aligned H3K9me3 % aligned H3K27me3 % aligned H3K36me3 % aligned  \\\n",
       "0              98.13              93.1              97.93              98.57   \n",
       "1              97.68             93.89              97.84              98.17   \n",
       "\n",
       "  H3K27ac % aligned input DNA control   % aligned  internal sample ID  \\\n",
       "0             98.10                         96.26                 NaN   \n",
       "1             97.86                         95.56                 NaN   \n",
       "\n",
       "   pool ID  Notes  # lanes 7 PE  \n",
       "0   IX5978    NaN             1  \n",
       "1   IX5981    NaN             1  \n",
       "\n",
       "[2 rows x 31 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = '/home/szong/projects/META_2018/Master_Sample_table_14X292_Feb27_2018_sz.xls'\n",
    "meta = pd.read_excel(f, sheetname='HTMCP CaCx epi')\n",
    "meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Order', 'Biospecimen ID', 'HIV status',\n",
       "       'Epigenome modifier mutation status', 'H3K4me1 library ID',\n",
       "       'H3K4me3 library ID', 'H3K9me3 library ID', 'H3K27me3 library ID',\n",
       "       'H3K36me3 library ID', 'H3K27ac library ID',\n",
       "       'input DNA control  library ID', ' Library construction status',\n",
       "       'Sequencing status', 'H3K4me1 total reads', 'H3K4me3  total reads',\n",
       "       'H3K9me3 total reads', 'H3K27me3 total reads', 'H3K36me3 total reads',\n",
       "       'H3K27ac total reads', 'input DNA control  total reads',\n",
       "       'H3K4me1 % aligned', 'H3K4me3  % aligned', 'H3K9me3 % aligned',\n",
       "       'H3K27me3 % aligned', 'H3K36me3 % aligned', 'H3K27ac % aligned',\n",
       "       'input DNA control   % aligned', 'internal sample ID', 'pool ID',\n",
       "       'Notes', '# lanes 7 PE'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "meta.set_index('Biospecimen ID', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df = meta[['H3K4me1 library ID',\n",
    "       'H3K4me3 library ID', 'H3K9me3 library ID', 'H3K27me3 library ID',\n",
    "       'H3K36me3 library ID', 'H3K27ac library ID']].stack().reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biospecimen ID</th>\n",
       "      <th>level_1</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007-01A</td>\n",
       "      <td>H3K4me1 library ID</td>\n",
       "      <td>A84624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02007-01A</td>\n",
       "      <td>H3K4me3 library ID</td>\n",
       "      <td>A84625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Biospecimen ID             level_1       0\n",
       "0  HTMCP-03-06-02007-01A  H3K4me1 library ID  A84624\n",
       "1  HTMCP-03-06-02007-01A  H3K4me3 library ID  A84625"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df['marks'] = df.level_1.apply(lambda x: x.replace(' library ID', ''))\n",
    "df['patient'] = df['Biospecimen ID'].apply(lambda x: x.replace('-01A', ''))\n",
    "df.drop('Biospecimen ID', axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df = df.drop('level_1', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>marks</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A84624</td>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A84625</td>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        0    marks            patient\n",
       "0  A84624  H3K4me1  HTMCP-03-06-02007\n",
       "1  A84625  H3K4me3  HTMCP-03-06-02007"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "f = '/projects/da_workspace/software/ChromHMM/Cervical/libs.txt'\n",
    "df.to_csv(f, sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df['treat'] = 'sorted_UCSC_' + df[0] + '_' + df['marks'] + '_treat_pileup.bdg'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df['control'] = 'sorted_UCSC_' + df[0] + '_' + df['marks'] + '_control_lambda.bdg'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>marks</th>\n",
       "      <th>patient</th>\n",
       "      <th>treat</th>\n",
       "      <th>control</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A84624</td>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>sorted_UCSC_A84624_H3K4me1_treat_pileup.bdg</td>\n",
       "      <td>sorted_UCSC_A84624_H3K4me1_control_lambda.bdg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A84625</td>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>sorted_UCSC_A84625_H3K4me3_treat_pileup.bdg</td>\n",
       "      <td>sorted_UCSC_A84625_H3K4me3_control_lambda.bdg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        0    marks            patient  \\\n",
       "0  A84624  H3K4me1  HTMCP-03-06-02007   \n",
       "1  A84625  H3K4me3  HTMCP-03-06-02007   \n",
       "\n",
       "                                         treat  \\\n",
       "0  sorted_UCSC_A84624_H3K4me1_treat_pileup.bdg   \n",
       "1  sorted_UCSC_A84625_H3K4me3_treat_pileup.bdg   \n",
       "\n",
       "                                         control  \n",
       "0  sorted_UCSC_A84624_H3K4me1_control_lambda.bdg  \n",
       "1  sorted_UCSC_A84625_H3K4me3_control_lambda.bdg  "
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df = df.drop(0, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>marks</th>\n",
       "      <th>patient</th>\n",
       "      <th>treat</th>\n",
       "      <th>control</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>H3K4me1</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>sorted_UCSC_A84624_H3K4me1_treat_pileup.bdg</td>\n",
       "      <td>sorted_UCSC_A84624_H3K4me1_control_lambda.bdg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>sorted_UCSC_A84625_H3K4me3_treat_pileup.bdg</td>\n",
       "      <td>sorted_UCSC_A84625_H3K4me3_control_lambda.bdg</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     marks            patient                                        treat  \\\n",
       "0  H3K4me1  HTMCP-03-06-02007  sorted_UCSC_A84624_H3K4me1_treat_pileup.bdg   \n",
       "1  H3K4me3  HTMCP-03-06-02007  sorted_UCSC_A84625_H3K4me3_treat_pileup.bdg   \n",
       "\n",
       "                                         control  \n",
       "0  sorted_UCSC_A84624_H3K4me1_control_lambda.bdg  \n",
       "1  sorted_UCSC_A84625_H3K4me3_control_lambda.bdg  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cols = ['patient', 'marks', 'treat', 'control']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df = df[cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "f = '/projects/da_workspace/software/ChromHMM/Cervical/bed_files.txt'\n",
    "df.to_csv(f, sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "df1 = df.set_index(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A84610</th>\n",
       "      <td>/projects/analysis/analysis30/IX5976/CC0TLANXX...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A84611</th>\n",
       "      <td>/projects/analysis/analysis30/IX5976/CC0TLANXX...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        1\n",
       "0                                                        \n",
       "A84610  /projects/analysis/analysis30/IX5976/CC0TLANXX...\n",
       "A84611  /projects/analysis/analysis30/IX5976/CC0TLANXX..."
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fbam = '/projects/da_workspace/software/ChromHMM/Cervical/bam_files.txt'\n",
    "dfbam = pd.read_csv(fbam, sep='\\t', header=None, index_col=0)\n",
    "dfbam.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "f = '/projects/da_workspace/software/ChromHMM/Cervical/bam_files.txt'\n",
    "pd.merge(df1, dfbam, left_index=True, right_index=True).set_index('patient', drop=True).to_csv(f, sep='\\t', header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/szong/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>H3K27ac</th>\n",
       "      <th>H3K27me3</th>\n",
       "      <th>H3K36me3</th>\n",
       "      <th>H3K4me1</th>\n",
       "      <th>H3K4me3</th>\n",
       "      <th>H3K9me3</th>\n",
       "      <th>chr</th>\n",
       "      <th>ix1</th>\n",
       "      <th>ix2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>65200</td>\n",
       "      <td>65400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>71200</td>\n",
       "      <td>71400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>71400</td>\n",
       "      <td>71600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>72400</td>\n",
       "      <td>72600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>83000</td>\n",
       "      <td>83200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  H3K27ac  H3K27me3  H3K36me3  H3K4me1  H3K4me3  H3K9me3 chr  \\\n",
       "0           0        0         0         0        0        0        1   1   \n",
       "1           1        0         1         0        0        0        0   1   \n",
       "2           2        1         0         0        0        0        0   1   \n",
       "3           3        0         0         0        0        0        1   1   \n",
       "4           4        0         1         0        0        0        0   1   \n",
       "\n",
       "     ix1    ix2  \n",
       "0  65200  65400  \n",
       "1  71200  71400  \n",
       "2  71400  71600  \n",
       "3  72400  72600  \n",
       "4  83000  83200  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "f = '/projects/da_workspace/software/ChromHMM/Cervical/signal_intervals.txt'\n",
    "df = pd.read_csv(f)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'H3K27ac', 'H3K27me3', 'H3K36me3', 'H3K4me1', 'H3K4me3',\n",
       "       'H3K9me3', 'chr', 'ix1', 'ix2'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "H3K27ac     1730476\n",
       "H3K27me3    1309242\n",
       "H3K36me3    2575782\n",
       "H3K4me1     2247894\n",
       "H3K4me3     1058464\n",
       "H3K9me3      980628\n",
       "dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
