{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import string as s\n",
    "import seaborn as sns\n",
    "import math\n",
    "# import tables\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import style, colors\n",
    "from itertools import groupby\n",
    "import matplotlib.patches as patches\n",
    "from matplotlib import gridspec\n",
    "import scipy.stats as stats\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage, fcluster, fclusterdata, leaves_list\n",
    "from scipy.spatial.distance import pdist\n",
    "from scipy.ndimage.interpolation import shift\n",
    "# matplotlib.style.use('ggplot')\n",
    "# %matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (12, 6)\n",
    "# matplotlib.rcParams['figure.facecolor'] = 'white'\n",
    "import mpl_toolkits.mplot3d.axes3d as p3\n",
    "from sklearn.cluster import AgglomerativeClustering\n",
    "from sklearn.datasets.samples_generator import make_swiss_roll\n",
    "matplotlib.style.use('ggplot')\n",
    "%matplotlib inline\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "wkdir = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "get meta data about library ids, hiv status, histology, total reads sequenced etc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>billed status</th>\n",
       "      <th>Order</th>\n",
       "      <th>Biospecimen ID</th>\n",
       "      <th>HIV status</th>\n",
       "      <th>Epigenome modifier mutation status</th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>...</th>\n",
       "      <th>H3K4me3  % aligned</th>\n",
       "      <th>H3K9me3 % aligned</th>\n",
       "      <th>H3K27me3 % aligned</th>\n",
       "      <th>H3K36me3 % aligned</th>\n",
       "      <th>H3K27ac % aligned</th>\n",
       "      <th>input DNA control   % aligned</th>\n",
       "      <th>pool ID</th>\n",
       "      <th>Notes</th>\n",
       "      <th># lanes 75 PE</th>\n",
       "      <th>Internal sample ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>billed May 2018</td>\n",
       "      <td>1.0</td>\n",
       "      <td>HTMCP-03-06-02007-01A</td>\n",
       "      <td>negative</td>\n",
       "      <td>mutated</td>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>...</td>\n",
       "      <td>98.13</td>\n",
       "      <td>92.83</td>\n",
       "      <td>97.56</td>\n",
       "      <td>98.14</td>\n",
       "      <td>98.10</td>\n",
       "      <td>95.87</td>\n",
       "      <td>IX5978; IX6429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.5</td>\n",
       "      <td>HTMCP_130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>billed May 2018</td>\n",
       "      <td>2.0</td>\n",
       "      <td>HTMCP-03-06-02026-01A</td>\n",
       "      <td>negative</td>\n",
       "      <td>mutated</td>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>...</td>\n",
       "      <td>97.68</td>\n",
       "      <td>93.76</td>\n",
       "      <td>97.66</td>\n",
       "      <td>98.09</td>\n",
       "      <td>97.86</td>\n",
       "      <td>95.25</td>\n",
       "      <td>IX5981; IX6430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.5</td>\n",
       "      <td>HTMCP_139</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     billed status  Order         Biospecimen ID HIV status  \\\n",
       "0  billed May 2018    1.0  HTMCP-03-06-02007-01A   negative   \n",
       "1  billed May 2018    2.0  HTMCP-03-06-02026-01A   negative   \n",
       "\n",
       "  Epigenome modifier mutation status H3K4me1 library ID H3K4me3 library ID  \\\n",
       "0                            mutated             A84624             A84625   \n",
       "1                            mutated             A84645             A84646   \n",
       "\n",
       "  H3K9me3 library ID H3K27me3 library ID H3K36me3 library ID  \\\n",
       "0             A84626              A84627              A84628   \n",
       "1             A84647              A84648              A84649   \n",
       "\n",
       "          ...         H3K4me3  % aligned H3K9me3 % aligned H3K27me3 % aligned  \\\n",
       "0         ...                      98.13             92.83              97.56   \n",
       "1         ...                      97.68             93.76              97.66   \n",
       "\n",
       "  H3K36me3 % aligned  H3K27ac % aligned  input DNA control   % aligned  \\\n",
       "0              98.14              98.10                          95.87   \n",
       "1              98.09              97.86                          95.25   \n",
       "\n",
       "          pool ID  Notes  # lanes 75 PE  Internal sample ID  \n",
       "0  IX5978; IX6429    NaN            1.5           HTMCP_130  \n",
       "1  IX5981; IX6430    NaN            1.5           HTMCP_139  \n",
       "\n",
       "[2 rows x 32 columns]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# f = '/home/szong/projects/META_2018/Master_Sample_table_14X292_Feb27_2018_sz.xls'\n",
    "f = '/home/szong/projects/META_2018/Master_Sample_table_14X292_Sep28_2018_sz.xls'\n",
    "meta = pd.read_excel(f, sheet_name='HTMCP CaCx epi')\n",
    "meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(57, 32)"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = meta[['Biospecimen ID', 'H3K4me1 library ID',\n",
    "       'H3K4me3 library ID', 'H3K9me3 library ID', 'H3K27me3 library ID',\n",
    "       'H3K36me3 library ID', 'H3K27ac library ID',\n",
    "       'input DNA control  library ID',  'H3K4me1 total reads', 'H3K4me3  total reads',\n",
    "       'H3K9me3 total reads', 'H3K27me3 total reads', 'H3K36me3 total reads',\n",
    "       'H3K27ac total reads', 'input DNA control  total reads']]#.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biospecimen ID</th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>input DNA control  library ID</th>\n",
       "      <th>H3K4me1 total reads</th>\n",
       "      <th>H3K4me3  total reads</th>\n",
       "      <th>H3K9me3 total reads</th>\n",
       "      <th>H3K27me3 total reads</th>\n",
       "      <th>H3K36me3 total reads</th>\n",
       "      <th>H3K27ac total reads</th>\n",
       "      <th>input DNA control  total reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02007-01A</td>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>A84630</td>\n",
       "      <td>127431532.0</td>\n",
       "      <td>60179836.0</td>\n",
       "      <td>123495836.0</td>\n",
       "      <td>149090470.0</td>\n",
       "      <td>196198296.0</td>\n",
       "      <td>80676254.0</td>\n",
       "      <td>156512468.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02026-01A</td>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>A84651</td>\n",
       "      <td>168600066.0</td>\n",
       "      <td>74487212.0</td>\n",
       "      <td>165810484.0</td>\n",
       "      <td>159223668.0</td>\n",
       "      <td>135643234.0</td>\n",
       "      <td>84986034.0</td>\n",
       "      <td>145945272.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Biospecimen ID H3K4me1 library ID H3K4me3 library ID  \\\n",
       "0  HTMCP-03-06-02007-01A             A84624             A84625   \n",
       "1  HTMCP-03-06-02026-01A             A84645             A84646   \n",
       "\n",
       "  H3K9me3 library ID H3K27me3 library ID H3K36me3 library ID  \\\n",
       "0             A84626              A84627              A84628   \n",
       "1             A84647              A84648              A84649   \n",
       "\n",
       "  H3K27ac library ID input DNA control  library ID  H3K4me1 total reads  \\\n",
       "0             A84629                        A84630          127431532.0   \n",
       "1             A84650                        A84651          168600066.0   \n",
       "\n",
       "   H3K4me3  total reads  H3K9me3 total reads  H3K27me3 total reads  \\\n",
       "0            60179836.0          123495836.0           149090470.0   \n",
       "1            74487212.0          165810484.0           159223668.0   \n",
       "\n",
       "   H3K36me3 total reads  H3K27ac total reads  input DNA control  total reads  \n",
       "0           196198296.0           80676254.0                     156512468.0  \n",
       "1           135643234.0           84986034.0                     145945272.0  "
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta['patient'] = meta['Biospecimen ID'].apply(lambda x: '-'.join(['HTMCP-03-06', x.split('-')[3]]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = meta.drop('Biospecimen ID', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H3K4me1 library ID</th>\n",
       "      <th>H3K4me3 library ID</th>\n",
       "      <th>H3K9me3 library ID</th>\n",
       "      <th>H3K27me3 library ID</th>\n",
       "      <th>H3K36me3 library ID</th>\n",
       "      <th>H3K27ac library ID</th>\n",
       "      <th>input DNA control  library ID</th>\n",
       "      <th>H3K4me1 total reads</th>\n",
       "      <th>H3K4me3  total reads</th>\n",
       "      <th>H3K9me3 total reads</th>\n",
       "      <th>H3K27me3 total reads</th>\n",
       "      <th>H3K36me3 total reads</th>\n",
       "      <th>H3K27ac total reads</th>\n",
       "      <th>input DNA control  total reads</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A84624</td>\n",
       "      <td>A84625</td>\n",
       "      <td>A84626</td>\n",
       "      <td>A84627</td>\n",
       "      <td>A84628</td>\n",
       "      <td>A84629</td>\n",
       "      <td>A84630</td>\n",
       "      <td>127431532.0</td>\n",
       "      <td>60179836.0</td>\n",
       "      <td>123495836.0</td>\n",
       "      <td>149090470.0</td>\n",
       "      <td>196198296.0</td>\n",
       "      <td>80676254.0</td>\n",
       "      <td>156512468.0</td>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A84645</td>\n",
       "      <td>A84646</td>\n",
       "      <td>A84647</td>\n",
       "      <td>A84648</td>\n",
       "      <td>A84649</td>\n",
       "      <td>A84650</td>\n",
       "      <td>A84651</td>\n",
       "      <td>168600066.0</td>\n",
       "      <td>74487212.0</td>\n",
       "      <td>165810484.0</td>\n",
       "      <td>159223668.0</td>\n",
       "      <td>135643234.0</td>\n",
       "      <td>84986034.0</td>\n",
       "      <td>145945272.0</td>\n",
       "      <td>HTMCP-03-06-02026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  H3K4me1 library ID H3K4me3 library ID H3K9me3 library ID  \\\n",
       "0             A84624             A84625             A84626   \n",
       "1             A84645             A84646             A84647   \n",
       "\n",
       "  H3K27me3 library ID H3K36me3 library ID H3K27ac library ID  \\\n",
       "0              A84627              A84628             A84629   \n",
       "1              A84648              A84649             A84650   \n",
       "\n",
       "  input DNA control  library ID  H3K4me1 total reads  H3K4me3  total reads  \\\n",
       "0                        A84630          127431532.0            60179836.0   \n",
       "1                        A84651          168600066.0            74487212.0   \n",
       "\n",
       "   H3K9me3 total reads  H3K27me3 total reads  H3K36me3 total reads  \\\n",
       "0          123495836.0           149090470.0           196198296.0   \n",
       "1          165810484.0           159223668.0           135643234.0   \n",
       "\n",
       "   H3K27ac total reads  input DNA control  total reads            patient  \n",
       "0           80676254.0                     156512468.0  HTMCP-03-06-02007  \n",
       "1           84986034.0                     145945272.0  HTMCP-03-06-02026  "
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(57, 15)"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>strelka_mutations</th>\n",
       "      <th>HPV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>26974</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02002</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>16661</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient HIV_status Putative_histology  strelka_mutations  \\\n",
       "0  HTMCP-03-06-02001   Positive           Squamous              26974   \n",
       "1  HTMCP-03-06-02002   Negative           Squamous              16661   \n",
       "\n",
       "  HPV_status  \n",
       "0   Positive  \n",
       "1   Positive  "
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add in hiv status and histology\n",
    "f22 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutation_load_clinic.txt'\n",
    "df22 = pd.read_csv(f22, sep='\\t')\n",
    "df22 = df22[['patient', 'HIV_status', 'Putative_histology', 'strelka_mutations', 'HPV_status']]\n",
    "# df22 = df22.set_index('patient')#.stack().reset_index()\n",
    "df22.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Squamous', 'Adeno'], dtype=object)"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df22.Putative_histology.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get latest clinic histology"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Adenosquamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02075</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Adenosquamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02047</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Neuroendocrine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          0               1\n",
       "HTMCP-03-06-02007  Negative   Adenosquamous\n",
       "HTMCP-03-06-02075  Negative           Adeno\n",
       "HTMCP-03-06-02203  Negative   Adenosquamous\n",
       "HTMCP-03-06-02047  Negative  Neuroendocrine\n",
       "HTMCP-03-06-02026  Negative           Adeno"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f33 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/mutations_for_oncoprint_123_patients.txt.new'\n",
    "df33 = pd.read_csv(f33)\n",
    "df44 = df33.iloc[:2,5:].T\n",
    "df44.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "df55 = pd.merge(df22, df44, left_on='patient', right_index=True, how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(123, 7)"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df55.head(2)\n",
    "df55.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>Putative_histology</th>\n",
       "      <th>strelka_mutations</th>\n",
       "      <th>HPV_status</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HTMCP-03-06-02007</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>2958282</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adenosquamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HTMCP-03-06-02047</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>8834</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Neuroendocrine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>HTMCP-03-06-02110</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>8565</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Adenosquamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>HTMCP-03-06-02176</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>25803</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adenosquamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>HTMCP-03-06-02203</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>35995</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Adenosquamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>HTMCP-03-06-02216</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>5831</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Adenosquamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>HTMCP-03-06-02235</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>7797</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Neuroendocrine</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              patient HIV_status Putative_histology  strelka_mutations  \\\n",
       "4   HTMCP-03-06-02007   Negative              Adeno            2958282   \n",
       "15  HTMCP-03-06-02047   Negative              Adeno               8834   \n",
       "36  HTMCP-03-06-02110   Positive              Adeno               8565   \n",
       "50  HTMCP-03-06-02176   Negative           Squamous              25803   \n",
       "55  HTMCP-03-06-02203   Negative              Adeno              35995   \n",
       "67  HTMCP-03-06-02216   Positive           Squamous               5831   \n",
       "76  HTMCP-03-06-02235   Positive              Adeno               7797   \n",
       "\n",
       "   HPV_status         0               1  \n",
       "4    Negative  Negative   Adenosquamous  \n",
       "15   Positive  Negative  Neuroendocrine  \n",
       "36   Positive  Positive   Adenosquamous  \n",
       "50   Positive  Negative   Adenosquamous  \n",
       "55   Positive  Negative   Adenosquamous  \n",
       "67   Positive  Positive   Adenosquamous  \n",
       "76   Positive  Positive  Neuroendocrine  "
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df55[(df55.Putative_histology != df55[1])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "df66 = df55.drop([0, 'Putative_histology'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>strelka_mutations</th>\n",
       "      <th>HPV_status</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>Positive</td>\n",
       "      <td>26974</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02002</td>\n",
       "      <td>Negative</td>\n",
       "      <td>16661</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient HIV_status  strelka_mutations HPV_status         1\n",
       "0  HTMCP-03-06-02001   Positive              26974   Positive  Squamous\n",
       "1  HTMCP-03-06-02002   Negative              16661   Positive  Squamous"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df66.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>strelka_mutations</th>\n",
       "      <th>HPV_status</th>\n",
       "      <th>clinic_histology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>Positive</td>\n",
       "      <td>26974</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HTMCP-03-06-02002</td>\n",
       "      <td>Negative</td>\n",
       "      <td>16661</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             patient HIV_status  strelka_mutations HPV_status clinic_histology\n",
       "0  HTMCP-03-06-02001   Positive              26974   Positive         Squamous\n",
       "1  HTMCP-03-06-02002   Negative              16661   Positive         Squamous"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df66.columns = df66.columns[:4].tolist() + ['clinic_histology']\n",
    "df66.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "read in Alessia's epigenetic gene categorization file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# histone modifier complex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "complexes = ['COMPASS complexes',\n",
    "'Polycomb complexes',\n",
    "'INO80/SWR remodelling complexes',\n",
    "'ISWI remdelling complexes',\n",
    "'SWI/SNF remodeling complex',\n",
    "'SWI/SNF family',\n",
    "'NuRD complex',\n",
    "'HDAC complex',\n",
    "'EP300/CBP HAT complexes',\n",
    "'GNAT family HAT complexes',\n",
    "'MYST family HAT complexes',\n",
    "'HAT complexes',\n",
    "'SEC (Super Elongation) Complex',\n",
    "'TFIID basal transcription factor complex or PCAF histone acetylase complex',\n",
    "'RNF20/40 complex']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HIV neg</th>\n",
       "      <th>neg samples with mutations</th>\n",
       "      <th>HIV positive</th>\n",
       "      <th>pos samples with mutation</th>\n",
       "      <th>total</th>\n",
       "      <th>category_1</th>\n",
       "      <th>category_2</th>\n",
       "      <th>markes /regiones affected</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>2</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF remodeling complex</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATRX</td>\n",
       "      <td>4</td>\n",
       "      <td>ATRX</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF family</td>\n",
       "      <td>gene bodies of ZFN genes</td>\n",
       "      <td>Binds to zinc-finger coding genes with atypica...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  HIV neg  neg samples with mutations HIV positive  pos samples with mutation  \\\n",
       "0  ARID1A                           2       ARID1A                          1   \n",
       "1    ATRX                           4         ATRX                          1   \n",
       "\n",
       "   total  category_1                  category_2 markes /regiones affected  \\\n",
       "0      3  Remodeller  SWI/SNF remodeling complex                       NaN   \n",
       "1      5  Remodeller              SWI/SNF family  gene bodies of ZFN genes   \n",
       "\n",
       "                                            comments  \n",
       "0                                                NaN  \n",
       "1  Binds to zinc-finger coding genes with atypica...  "
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# f4 = '/projects/trans_scratch/validations/workspace/szong/Cervical/chipseq/categorisation_of_epigentic_modifiers_from_Alessia_v3.xls'\n",
    "f4 = '{0}{1}'.format(wkdir, 'categorisation_of_epigentic_modifiers_v4.xlsx')\n",
    "df4 = pd.read_excel(f4, sheet_name='categorisation')\n",
    "df4.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HIV neg</th>\n",
       "      <th>neg samples with mutations</th>\n",
       "      <th>HIV positive</th>\n",
       "      <th>pos samples with mutation</th>\n",
       "      <th>total</th>\n",
       "      <th>category_1</th>\n",
       "      <th>category_2</th>\n",
       "      <th>markes /regiones affected</th>\n",
       "      <th>comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>HDAC10</td>\n",
       "      <td>1</td>\n",
       "      <td>HDAC10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Eraser</td>\n",
       "      <td>HDAC complex</td>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>HDAC2</td>\n",
       "      <td>1</td>\n",
       "      <td>HDAC2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>Eraser</td>\n",
       "      <td>HDAC complex</td>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>HDAC4</td>\n",
       "      <td>0</td>\n",
       "      <td>HDAC4</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>Eraser</td>\n",
       "      <td>HDAC complex</td>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>HDAC5</td>\n",
       "      <td>0</td>\n",
       "      <td>HDAC5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Eraser</td>\n",
       "      <td>HDAC complex</td>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>1</td>\n",
       "      <td>HDAC6</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>Eraser</td>\n",
       "      <td>HDAC complex</td>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>HDAC7</td>\n",
       "      <td>0</td>\n",
       "      <td>HDAC7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>Eraser</td>\n",
       "      <td>HDAC complex</td>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>HDAC9</td>\n",
       "      <td>3</td>\n",
       "      <td>HDAC9</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>Eraser</td>\n",
       "      <td>HDAC complex</td>\n",
       "      <td>H3K27ac</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   HIV neg  neg samples with mutations HIV positive  \\\n",
       "39  HDAC10                           1       HDAC10   \n",
       "40   HDAC2                           1        HDAC2   \n",
       "42   HDAC4                           0        HDAC4   \n",
       "43   HDAC5                           0        HDAC5   \n",
       "44   HDAC6                           1        HDAC6   \n",
       "45   HDAC7                           0        HDAC7   \n",
       "46   HDAC9                           3        HDAC9   \n",
       "\n",
       "    pos samples with mutation  total category_1    category_2  \\\n",
       "39                          0      1     Eraser  HDAC complex   \n",
       "40                          0      1     Eraser  HDAC complex   \n",
       "42                          3      3     Eraser  HDAC complex   \n",
       "43                          1      1     Eraser  HDAC complex   \n",
       "44                          2      3     Eraser  HDAC complex   \n",
       "45                          2      2     Eraser  HDAC complex   \n",
       "46                          0      3     Eraser  HDAC complex   \n",
       "\n",
       "   markes /regiones affected comments  \n",
       "39                   H3K27ac      NaN  \n",
       "40                   H3K27ac      NaN  \n",
       "42                   H3K27ac      NaN  \n",
       "43                   H3K27ac      NaN  \n",
       "44                   H3K27ac      NaN  \n",
       "45                   H3K27ac      NaN  \n",
       "46                   H3K27ac      NaN  "
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df4[df4.category_2 == 'HDAC complex']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>category_1</th>\n",
       "      <th>category_2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIV neg</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF remodeling complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATRX</th>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF family</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         category_1                  category_2\n",
       "HIV neg                                        \n",
       "ARID1A   Remodeller  SWI/SNF remodeling complex\n",
       "ATRX     Remodeller              SWI/SNF family"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df41 = df4[['HIV neg', 'category_1', 'category_2']].set_index('HIV neg')\n",
    "df41.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Remodeller', 'Reader', 'Writer', nan, 'Eraser'], dtype=object)"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df41.category_1.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Polycomb complex                                                              10\n",
       "INO80/SWR remodelling complexes                                                9\n",
       "compass complex                                                                9\n",
       "GNAT family HAT complexes                                                      7\n",
       "ISWI remdelling complexes                                                      7\n",
       "HDAC complex                                                                   7\n",
       "SWI/SNF remodeling complex                                                     6\n",
       "NuRD complexÂ                                                                   5\n",
       "MYST family HAT complexes                                                      5\n",
       "SEC (Super Elongation) Complex                                                 4\n",
       "EP300/CBP HAT complexes                                                        3\n",
       "TFIID basal transcription factor complex or PCAF histone acetylase complex     3\n",
       "chromatin remodelling                                                          2\n",
       "RNF20/40 complex                                                               2\n",
       "SWI/SNF family                                                                 2\n",
       "REST corepressor                                                               1\n",
       "Testis-specific chromatin protein                                              1\n",
       "HAT complexes                                                                  1\n",
       "ISWI chromatin remdelling                                                      1\n",
       "Name: category_2, dtype: int64"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df41.category_2.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HIV status</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ATRX</th>\n",
       "      <th>BAZ1A</th>\n",
       "      <th>BAZ1B</th>\n",
       "      <th>BAZ2B</th>\n",
       "      <th>BRD2</th>\n",
       "      <th>BRD3</th>\n",
       "      <th>BRD4</th>\n",
       "      <th>BRD7</th>\n",
       "      <th>...</th>\n",
       "      <th>TBL1X</th>\n",
       "      <th>TFPT</th>\n",
       "      <th>TRIM33</th>\n",
       "      <th>UBE2B</th>\n",
       "      <th>TRRAP</th>\n",
       "      <th>WHSC1L1</th>\n",
       "      <th>YEATS2</th>\n",
       "      <th>YY1</th>\n",
       "      <th>ZMYND11</th>\n",
       "      <th>ZMYND8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <td>HIV negative</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <td>HIV negative</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 117 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     HIV status  ARID1A  ATRX  BAZ1A  BAZ1B  BAZ2B  BRD2  \\\n",
       "patient                                                                    \n",
       "HTMCP-03-06-02007  HIV negative       1     1      1      0      1     0   \n",
       "HTMCP-03-06-02026  HIV negative       1     1      0      1      1     0   \n",
       "\n",
       "                   BRD3  BRD4  BRD7   ...    TBL1X  TFPT  TRIM33  UBE2B  \\\n",
       "patient                               ...                                 \n",
       "HTMCP-03-06-02007     0     1     1   ...        0     0       0      0   \n",
       "HTMCP-03-06-02026     1     0     1   ...        0     0       1      0   \n",
       "\n",
       "                   TRRAP  WHSC1L1  YEATS2  YY1  ZMYND11  ZMYND8  \n",
       "patient                                                          \n",
       "HTMCP-03-06-02007      1        0       0    1        0       1  \n",
       "HTMCP-03-06-02026      0        0       0    0        1       0  \n",
       "\n",
       "[2 rows x 117 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df5 = pd.read_excel(f4, sheet_name='sample_mut_details_sz', index_col='patient')\n",
    "df5.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02076</th>\n",
       "      <th>HTMCP-03-06-02113</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATRX</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient  HTMCP-03-06-02007  HTMCP-03-06-02026  HTMCP-03-06-02063  \\\n",
       "ARID1A                 1.0                1.0                0.0   \n",
       "ATRX                   1.0                1.0                1.0   \n",
       "\n",
       "patient  HTMCP-03-06-02214  HTMCP-03-06-02042  HTMCP-03-06-02332  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02185  HTMCP-03-06-02012  HTMCP-03-06-02196  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02203        ...          HTMCP-03-06-02076  \\\n",
       "ARID1A                 0.0        ...                        0.0   \n",
       "ATRX                   0.0        ...                        0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02113  HTMCP-03-06-02230  HTMCP-03-06-02170  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02152  HTMCP-03-06-02235  HTMCP-03-06-02195  \\\n",
       "ARID1A                 0.0                0.0                0.0   \n",
       "ATRX                   0.0                0.0                0.0   \n",
       "\n",
       "patient  HTMCP-03-06-02261  HTMCP-03-06-02147  HTMCP-03-06-02267  \n",
       "ARID1A                 0.0                0.0                0.0  \n",
       "ATRX                   0.0                0.0                0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df51 = df5[df5.columns[1:]].T\n",
    "\n",
    "df51.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "      <th>category_1</th>\n",
       "      <th>category_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARID1A</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF remodeling complex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATRX</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Remodeller</td>\n",
       "      <td>SWI/SNF family</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        HTMCP-03-06-02007  HTMCP-03-06-02026  HTMCP-03-06-02063  \\\n",
       "ARID1A                1.0                1.0                0.0   \n",
       "ATRX                  1.0                1.0                1.0   \n",
       "\n",
       "        HTMCP-03-06-02214  HTMCP-03-06-02042  HTMCP-03-06-02332  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02185  HTMCP-03-06-02012  HTMCP-03-06-02196  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02203             ...              HTMCP-03-06-02230  \\\n",
       "ARID1A                0.0             ...                            0.0   \n",
       "ATRX                  0.0             ...                            0.0   \n",
       "\n",
       "        HTMCP-03-06-02170  HTMCP-03-06-02152  HTMCP-03-06-02235  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02195  HTMCP-03-06-02261  HTMCP-03-06-02147  \\\n",
       "ARID1A                0.0                0.0                0.0   \n",
       "ATRX                  0.0                0.0                0.0   \n",
       "\n",
       "        HTMCP-03-06-02267  category_1                  category_2  \n",
       "ARID1A                0.0  Remodeller  SWI/SNF remodeling complex  \n",
       "ATRX                  0.0  Remodeller              SWI/SNF family  \n",
       "\n",
       "[2 rows x 56 columns]"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df6 = pd.merge(df51, df41, left_index=True, right_index=True)\n",
    "\n",
    "df6.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [],
   "source": [
    "def categorize_cols(df, col):\n",
    "    col_values = df[col].unique()\n",
    "    lst = []\n",
    "    for col_value in col_values:\n",
    "        s = df.query('{0} == \"{1}\"'.format(col, col_value)).sum()\n",
    "        lst.append(s)\n",
    "    dfo = pd.DataFrame(lst)\n",
    "    dfo.index = col_values\n",
    "    dfo = dfo.drop([col], axis=1)\n",
    "    return dfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02076</th>\n",
       "      <th>HTMCP-03-06-02113</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Remodeller</th>\n",
       "      <td>16.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reader</th>\n",
       "      <td>21.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            HTMCP-03-06-02007  HTMCP-03-06-02026  HTMCP-03-06-02063  \\\n",
       "Remodeller               16.0                9.0                5.0   \n",
       "Reader                   21.0                7.0                6.0   \n",
       "\n",
       "            HTMCP-03-06-02214  HTMCP-03-06-02042  HTMCP-03-06-02332  \\\n",
       "Remodeller                2.0                2.0                2.0   \n",
       "Reader                    3.0                1.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02185  HTMCP-03-06-02012  HTMCP-03-06-02196  \\\n",
       "Remodeller                0.0                1.0                0.0   \n",
       "Reader                    1.0                1.0                2.0   \n",
       "\n",
       "            HTMCP-03-06-02203        ...          HTMCP-03-06-02076  \\\n",
       "Remodeller                0.0        ...                        0.0   \n",
       "Reader                    2.0        ...                        0.0   \n",
       "\n",
       "            HTMCP-03-06-02113  HTMCP-03-06-02230  HTMCP-03-06-02170  \\\n",
       "Remodeller                0.0                0.0                0.0   \n",
       "Reader                    0.0                0.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02152  HTMCP-03-06-02235  HTMCP-03-06-02195  \\\n",
       "Remodeller                0.0                0.0                0.0   \n",
       "Reader                    0.0                0.0                0.0   \n",
       "\n",
       "            HTMCP-03-06-02261  HTMCP-03-06-02147  HTMCP-03-06-02267  \n",
       "Remodeller                0.0                0.0                0.0  \n",
       "Reader                    0.0                0.0                0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df8 = categorize_cols(df6, 'category_1').drop('category_2', axis=1)\n",
    "\n",
    "df8.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02076</th>\n",
       "      <th>HTMCP-03-06-02113</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SWI/SNF remodeling complex</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SWI/SNF family</th>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            HTMCP-03-06-02007  HTMCP-03-06-02026  \\\n",
       "SWI/SNF remodeling complex                6.0                2.0   \n",
       "SWI/SNF family                            2.0                2.0   \n",
       "\n",
       "                            HTMCP-03-06-02063  HTMCP-03-06-02214  \\\n",
       "SWI/SNF remodeling complex                0.0                2.0   \n",
       "SWI/SNF family                            1.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02042  HTMCP-03-06-02332  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02185  HTMCP-03-06-02012  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02196  HTMCP-03-06-02203  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                                  ...          HTMCP-03-06-02076  \\\n",
       "SWI/SNF remodeling complex        ...                        0.0   \n",
       "SWI/SNF family                    ...                        0.0   \n",
       "\n",
       "                            HTMCP-03-06-02113  HTMCP-03-06-02230  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02170  HTMCP-03-06-02152  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02235  HTMCP-03-06-02195  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02261  HTMCP-03-06-02147  \\\n",
       "SWI/SNF remodeling complex                0.0                0.0   \n",
       "SWI/SNF family                            0.0                0.0   \n",
       "\n",
       "                            HTMCP-03-06-02267  \n",
       "SWI/SNF remodeling complex                0.0  \n",
       "SWI/SNF family                            0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df9 = categorize_cols(df6, 'category_2').drop('category_1', axis=1)\n",
    "\n",
    "df9.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02076</th>\n",
       "      <th>HTMCP-03-06-02113</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Remodeller</th>\n",
       "      <td>16.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reader</th>\n",
       "      <td>21.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            HTMCP-03-06-02007  HTMCP-03-06-02026  HTMCP-03-06-02063  \\\n",
       "index                                                                 \n",
       "Remodeller               16.0                9.0                5.0   \n",
       "Reader                   21.0                7.0                6.0   \n",
       "\n",
       "            HTMCP-03-06-02214  HTMCP-03-06-02042  HTMCP-03-06-02332  \\\n",
       "index                                                                 \n",
       "Remodeller                2.0                2.0                2.0   \n",
       "Reader                    3.0                1.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02185  HTMCP-03-06-02012  HTMCP-03-06-02196  \\\n",
       "index                                                                 \n",
       "Remodeller                0.0                1.0                0.0   \n",
       "Reader                    1.0                1.0                2.0   \n",
       "\n",
       "            HTMCP-03-06-02203        ...          HTMCP-03-06-02076  \\\n",
       "index                                ...                              \n",
       "Remodeller                0.0        ...                        0.0   \n",
       "Reader                    2.0        ...                        0.0   \n",
       "\n",
       "            HTMCP-03-06-02113  HTMCP-03-06-02230  HTMCP-03-06-02170  \\\n",
       "index                                                                 \n",
       "Remodeller                0.0                0.0                0.0   \n",
       "Reader                    0.0                0.0                1.0   \n",
       "\n",
       "            HTMCP-03-06-02152  HTMCP-03-06-02235  HTMCP-03-06-02195  \\\n",
       "index                                                                 \n",
       "Remodeller                0.0                0.0                0.0   \n",
       "Reader                    0.0                0.0                0.0   \n",
       "\n",
       "            HTMCP-03-06-02261  HTMCP-03-06-02147  HTMCP-03-06-02267  \n",
       "index                                                                \n",
       "Remodeller                0.0                0.0                0.0  \n",
       "Reader                    0.0                0.0                0.0  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df10 = pd.concat([df8, df9]).reset_index().dropna().set_index('index').astype(float)\n",
    "\n",
    "df10.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [],
   "source": [
    "df10[df10 < 0.5] = np.nan\n",
    "df10[df10 > 0.5] = 'Mutated'\n",
    "df10.fillna('Wild Type', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <th>HTMCP-03-06-02063</th>\n",
       "      <th>HTMCP-03-06-02214</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02332</th>\n",
       "      <th>HTMCP-03-06-02185</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02196</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02076</th>\n",
       "      <th>HTMCP-03-06-02113</th>\n",
       "      <th>HTMCP-03-06-02230</th>\n",
       "      <th>HTMCP-03-06-02170</th>\n",
       "      <th>HTMCP-03-06-02152</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "      <th>HTMCP-03-06-02195</th>\n",
       "      <th>HTMCP-03-06-02261</th>\n",
       "      <th>HTMCP-03-06-02147</th>\n",
       "      <th>HTMCP-03-06-02267</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>index</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Remodeller</th>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>...</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Reader</th>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>...</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           HTMCP-03-06-02007 HTMCP-03-06-02026 HTMCP-03-06-02063  \\\n",
       "index                                                              \n",
       "Remodeller           Mutated           Mutated           Mutated   \n",
       "Reader               Mutated           Mutated           Mutated   \n",
       "\n",
       "           HTMCP-03-06-02214 HTMCP-03-06-02042 HTMCP-03-06-02332  \\\n",
       "index                                                              \n",
       "Remodeller           Mutated           Mutated           Mutated   \n",
       "Reader               Mutated           Mutated           Mutated   \n",
       "\n",
       "           HTMCP-03-06-02185 HTMCP-03-06-02012 HTMCP-03-06-02196  \\\n",
       "index                                                              \n",
       "Remodeller         Wild Type           Mutated         Wild Type   \n",
       "Reader               Mutated           Mutated           Mutated   \n",
       "\n",
       "           HTMCP-03-06-02203        ...        HTMCP-03-06-02076  \\\n",
       "index                               ...                            \n",
       "Remodeller         Wild Type        ...                Wild Type   \n",
       "Reader               Mutated        ...                Wild Type   \n",
       "\n",
       "           HTMCP-03-06-02113 HTMCP-03-06-02230 HTMCP-03-06-02170  \\\n",
       "index                                                              \n",
       "Remodeller         Wild Type         Wild Type         Wild Type   \n",
       "Reader             Wild Type         Wild Type           Mutated   \n",
       "\n",
       "           HTMCP-03-06-02152 HTMCP-03-06-02235 HTMCP-03-06-02195  \\\n",
       "index                                                              \n",
       "Remodeller         Wild Type         Wild Type         Wild Type   \n",
       "Reader             Wild Type         Wild Type         Wild Type   \n",
       "\n",
       "           HTMCP-03-06-02261 HTMCP-03-06-02147 HTMCP-03-06-02267  \n",
       "index                                                             \n",
       "Remodeller         Wild Type         Wild Type         Wild Type  \n",
       "Reader             Wild Type         Wild Type         Wild Type  \n",
       "\n",
       "[2 rows x 54 columns]"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df10.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(23, 54)"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df10.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [],
   "source": [
    "df11 = df10.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>index</th>\n",
       "      <th>Remodeller</th>\n",
       "      <th>Reader</th>\n",
       "      <th>Writer</th>\n",
       "      <th>Eraser</th>\n",
       "      <th>SWI/SNF remodeling complex</th>\n",
       "      <th>SWI/SNF family</th>\n",
       "      <th>ISWI remdelling complexes</th>\n",
       "      <th>SEC (Super Elongation) Complex</th>\n",
       "      <th>INO80/SWR remodelling complexes</th>\n",
       "      <th>Testis-specific chromatin protein</th>\n",
       "      <th>...</th>\n",
       "      <th>REST corepressor</th>\n",
       "      <th>ISWI chromatin remdelling</th>\n",
       "      <th>HAT complexes</th>\n",
       "      <th>NuRD complex</th>\n",
       "      <th>compass complex</th>\n",
       "      <th>HDAC complex</th>\n",
       "      <th>GNAT family HAT complexes</th>\n",
       "      <th>MYST family HAT complexes</th>\n",
       "      <th>RNF20/40 complex</th>\n",
       "      <th>TFIID basal transcription factor complex or PCAF histone acetylase complex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02007</th>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>...</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02026</th>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>...</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Mutated</td>\n",
       "      <td>Wild Type</td>\n",
       "      <td>Mutated</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "index             Remodeller   Reader   Writer   Eraser  \\\n",
       "HTMCP-03-06-02007    Mutated  Mutated  Mutated  Mutated   \n",
       "HTMCP-03-06-02026    Mutated  Mutated  Mutated  Mutated   \n",
       "\n",
       "index             SWI/SNF remodeling complex SWI/SNF family  \\\n",
       "HTMCP-03-06-02007                    Mutated        Mutated   \n",
       "HTMCP-03-06-02026                    Mutated        Mutated   \n",
       "\n",
       "index             ISWI remdelling complexes SEC (Super Elongation) Complex  \\\n",
       "HTMCP-03-06-02007                   Mutated                        Mutated   \n",
       "HTMCP-03-06-02026                   Mutated                      Wild Type   \n",
       "\n",
       "index             INO80/SWR remodelling complexes  \\\n",
       "HTMCP-03-06-02007                         Mutated   \n",
       "HTMCP-03-06-02026                         Mutated   \n",
       "\n",
       "index             Testis-specific chromatin protein  \\\n",
       "HTMCP-03-06-02007                           Mutated   \n",
       "HTMCP-03-06-02026                         Wild Type   \n",
       "\n",
       "index                                                ...                                      \\\n",
       "HTMCP-03-06-02007                                    ...                                       \n",
       "HTMCP-03-06-02026                                    ...                                       \n",
       "\n",
       "index             REST corepressor ISWI chromatin remdelling HAT complexes  \\\n",
       "HTMCP-03-06-02007          Mutated                   Mutated       Mutated   \n",
       "HTMCP-03-06-02026        Wild Type                 Wild Type       Mutated   \n",
       "\n",
       "index             NuRD complexÂ  compass complex HDAC complex  \\\n",
       "HTMCP-03-06-02007       Mutated         Mutated      Mutated   \n",
       "HTMCP-03-06-02026       Mutated         Mutated      Mutated   \n",
       "\n",
       "index             GNAT family HAT complexes MYST family HAT complexes  \\\n",
       "HTMCP-03-06-02007                   Mutated                   Mutated   \n",
       "HTMCP-03-06-02026                   Mutated                   Mutated   \n",
       "\n",
       "index             RNF20/40 complex  \\\n",
       "HTMCP-03-06-02007          Mutated   \n",
       "HTMCP-03-06-02026        Wild Type   \n",
       "\n",
       "index             TFIID basal transcription factor complex or PCAF histone acetylase complex  \n",
       "HTMCP-03-06-02007                                            Mutated                          \n",
       "HTMCP-03-06-02026                                            Mutated                          \n",
       "\n",
       "[2 rows x 23 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df11.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "df77 = pd.merge(df11, df66, left_index=True, right_on='patient', how='left').set_index('patient')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(54, 27)"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df77.head(2)\n",
    "df77.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = f'/projects/da_workspace/DA-167_Cervical_epi_clustering/clinic_covariate_20181009.csv'\n",
    "df77.to_csv(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
