{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Thanks very much - Veronica, Ru Chen and I went over the pedigrees this morning and reverified the identities. I have attached the sample list here.\n",
    "\n",
    "The sibship defined by B04522, B04523, B04524, B04525, B04526, B04527 and B04528 should all be compared to each other. Since B04523 and B04528 could not be collected, we are left with assuming B04522, B04524, B04525, B04526, B04527 are full sibs. Of those 5 samples, 3 are affected and 2 are unaffected. We are interested in the list of rare variants (either coding or non-coding) that are shared among the affecteds (B04522, B04524 and B04525) and not shared with the unaffecteds (B04526, B04527). \n",
    "These can be homozygous (autosomal recessive model) or heterozygous (autosomal dominant model).\n",
    "\n",
    "B04529 and B04530 are third cousins to each other, both affected. We are interested in the list of rare variants that are shared between them.\n",
    "\n",
    "The other samples are all individuals from different families, unrelated to each other. It is possible (though not likely, given what we know about the genetic architecture of nonsyndromic brain aneurysms) that one or more of these samples has a rare variant in or near a gene that is flagged by one of the above approaches.\n",
    "If there is an algorithmic way to approach that, I'd appreciate you letting me know, but as far as I am aware the onus will be on me to scroll through the variant lists for those individuals one at a time to see by eye whether any rare variants in these singletons are in the same genes as are flagged by the filtering of the multiplex sibship and/or the two third cousins."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# email correspondence to help filtering"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The set of samples IBA1863, IBA1823, IBA1801, IBA1837 as labelled here each represent individual affected members from multiplex families. For each of these we will want .vcf files that represent all rare coding variants present at a MAF of 0.01 or less in gnomAD. It is likely that these families each have a different causative allele, <font color='red'>so there is no rationale for overlapping them to exclude rare alleles from consideration.</font>\n",
    "\n",
    "It is possible that any two of them may have different causative alleles in the same gene, but that is not likely so we will analyse them in conjunction with other rare families we have.\n",
    "\n",
    "\n",
    "\n",
    "The two samples labelled \"IV-1\" and \"IV-2\" here will be provided with IBA numbers when we send them to you. As you can see, they are third cousins. Thus, we would want to analyse these samples together - the .vcf files we would want from these two can safely <font color='red'>be restricted to rare coding variants with a MAF of 0.01 or less in gnomAD that are shared between both individuals.</font>\n",
    "\n",
    "\n",
    "\n",
    "Lastly (though perhaps most importantly), the multiplex family II-1, II-2, II-3, II-4, II-5, II-6 and II-7.\n",
    "\n",
    "\n",
    "\n",
    "Here we are looking for rare coding variants that satisfy the following\n",
    "<font color='red'>\n",
    "\n",
    "criteria:\n",
    "\n",
    "\n",
    "\n",
    "1. MAF of 0.01 or less in gnomAD\n",
    "\n",
    "\n",
    "\n",
    "2. Shared among all 4 of II-1, II-2, II-4, II-5.\n",
    "\n",
    "\n",
    "\n",
    "AND\n",
    "\n",
    "\n",
    "\n",
    "3. Not shared with any of II-3, II-6 or II-7.\n",
    "</font>\n",
    "\n",
    "\n",
    "We would appreciate a quote for an SOW that satisfied the above parameters, with 40X coverage across the whole genome.\n",
    "\n",
    "\n",
    "\n",
    "Of course we are most interested in the variants that are predicted to be damaging. If CADD scores can be provided, that would be great, alongside predictions from SIFT, Polyphen, Mutation Taster and/or Align GVGD. I am not sure if there is a \"standard\" pipeline that you would use to call heterozygous dominant variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A haplotype is a group of genes within an organism that was inherited together from a single parent. The word \"haplotype\" is derived from the word \"haploid,\" which describes cells with only one set of chromosomes, and from the word \"genotype,\" which refers to the genetic makeup of an organism. A haplotype can describe a pair of genes inherited together from one parent on one chromosome, or it can describe all of the genes on a chromosome that were inherited together from a single parent. This group of genes was inherited together because of genetic linkage, or the phenomenon by which genes that are close to each other on the same chromosome are often inherited together. In addition, the term \"haplotype\" can also refer to the inheritance of a cluster of single nucleotide polymorphisms (SNPs), which are variations at single positions in the DNA sequence among individuals.\n",
    "\n",
    "By examining haplotypes, scientists can identify patterns of genetic variation that are associated with health and disease states. For instance, if a haplotype is associated with a certain disease, then scientists can examine stretches of DNA near the SNP cluster to try to identify the gene or genes responsible for causing the disease."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://macarthurlab.org/2017/02/27/the-genome-aggregation-database-gnomad/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import matplotlib.pyplot as plt\n",
    "%reload_ext autoreload\n",
    "%autoreload 2\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "wkdir = '/projects/da_workspace/DA-337/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "siblibs = ['B04529', 'B04530']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# vcf v4.2 specification"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "QUAL: −10log10prob(no variant)\n",
    "\n",
    "AC : allele count in genotypes, for each ALT allele, in the same order as listed, it is the alternativew allele count for all samples. AC=3 means 3 alt alleles found in 5 individuals in this case.\n",
    "\n",
    "AF : allele frequency for each ALT allele in the same order as listed: use this when estimated from primary\n",
    "data, not called genotypes, AF= AC/AN\n",
    "\n",
    "AN : total number of alleles in called genotypes, in this case AN=10, 5 diploid samples\n",
    "\n",
    "BQ : RMS base quality at this position\n",
    "\n",
    "DB : dbSNP membership\n",
    "\n",
    "DP : combined depth across samples, e.g. DP=154\n",
    "\n",
    "GT : genotype, encoded as allele values separated by either of / or |. The allele values are 0 for the reference\n",
    "allele (what is in the REF field), 1 for the first allele listed in ALT, 2 for the second allele list in ALT and\n",
    "so on. For diploid calls examples could be 0/1, 1 | 0, or 1/2, etc. For haploid calls, e.g. on Y, male nonpseudoautosomal X, or mitochondrion, only one allele value should be given; a triploid call might look like\n",
    "0/0/1. If a call cannot be made for a sample at a given locus, ‘.’ should be specified for each missing allele\n",
    "5\n",
    "\n",
    "in the GT field (for example ‘./.’ for a diploid genotype and ‘.’ for haploid genotype). The meanings of the\n",
    "separators are as follows (see the PS field below for more details on incorporating phasing information into the\n",
    "genotypes):\n",
    "\n",
    "◦ / : genotype unphased\n",
    "\n",
    "◦ | : genotype phased\n",
    "\n",
    "GL : genotype likelihoods comprised of comma separated floating point log10-scaled likelihoods for all possible\n",
    "genotypes given the set of alleles defined in the REF and ALT fields. In presence of the GT field the same\n",
    "ploidy is expected and the canonical order is used; without GT field, diploidy is assumed. If A is the allele in\n",
    "REF and B,C,... are the alleles as ordered in ALT, the ordering of genotypes for the likelihoods is given by:\n",
    "F(j/k) = (k*(k+1)/2)+j. In other words, for biallelic sites the ordering is: AA,AB,BB; for triallelic sites the\n",
    "ordering is: AA,AB,BB,AC,BC,CC, etc. For example: GT:GL 0/1:-323.03,-99.29,-802.53 (Floats)\n",
    "\n",
    "\n",
    "DP : read depth at this position for this sample (Integer)\n",
    "\n",
    "GQ : conditional genotype quality, encoded as a phred quality −10log10 p(genotype call is wrong, conditioned\n",
    "on the site’s being variant) (Integer)\n",
    "\n",
    "PL : the phred-scaled genotype likelihoods rounded to the closest integer (and otherwise defined precisely as\n",
    "the GL field) (Integers)\n",
    "\n",
    "Phasing:  \n",
    "If the GTs in the VCF have the pipe symbol instead of the forward slash, i.e., 1|0, 1|1, or 0|1, it means that we know which variants were called on the same paternal or maternal chromosome, i.e., the same haplotype. To get this level of information, faithfully, requires a special type of sample prep that I won't go into right now. To interpret this data, all variants reported on the left of the pipe symbol will be on the same haplotype, whilst all on the right will be on the another.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "20 14370 rs6054257 G A 29 PASS NS=3;DP=14;AF=0.5;DB;H2 GT:GQ:DP:HQ 0|0:48:1:51,51 1|0:48:8:51,51 1/1:43:5:.,.  \n",
    "20 17330 . T A 3 q10 NS=3;DP=11;AF=0.017 GT:GQ:DP:HQ 0|0:49:3:58,50 0|1:3:5:65,3 0/0:41:3  \n",
    "20 1110696 rs6040355 A G,T 67 PASS NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ 1|2:21:6:23,27 2|1:2:0:18,2 2/2:35:4  \n",
    "20 1230237 . T . 47 PASS NS=3;DP=13;AA=T GT:GQ:DP:HQ 0|0:54:7:56,60 0|0:48:4:51,51 0/0:61:2  \n",
    "20 1234567 microsat1 GTC G,GTCT 50 PASS NS=3;DP=9;AA=G GT:GQ:DP 0/1:35:4 0/2:17:2 1/1:40:3  \n",
    "This example shows (in order): a good simple SNP, a possible SNP that has been filtered out because its quality is\n",
    "below 10, a site at which two alternate alleles are called, with one of them (T) being ancestral (possibly a reference\n",
    "sequencing error), a site that is called monomorphic reference (i.e. with no alternate alleles), and a microsatellite with\n",
    "two alternative alleles, one a deletion of 2 bases (TC), and the other an insertion of one base (T). Genotype data are\n",
    "given for three samples, two of which are phased and the third unphased, with per sample genotype quality, depth\n",
    "and haplotype qualities (the latter only for the phased samples) given as well as the genotypes. The microsatellite\n",
    "calls are unphased."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# annotated vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>B04529</th>\n",
       "      <th>B04530</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>10150</td>\n",
       "      <td>.</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>34.88</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.250;AN=4;BaseQRankSum=1.59;DP=141;Ex...</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>0/0:126,0:126:75:.:.:0,75,3085</td>\n",
       "      <td>0/1:7,2:9:43:0|1:10150_CT_C:43,0,135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>10157</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>37.29</td>\n",
       "      <td>VQSRTrancheSNP99.90to99.95</td>\n",
       "      <td>AC=1;AF=0.250;AN=4;BaseQRankSum=-1.350e-01;DP=...</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>0/0:106,0:106:0:.:.:0,0,954</td>\n",
       "      <td>0/1:6,2:8:44:0|1:10150_CT_C:44,0,134</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS ID REF ALT   QUAL                      FILTER  \\\n",
       "0     1  10150  .  CT   C  34.88                        PASS   \n",
       "1     1  10157  .   T   C  37.29  VQSRTrancheSNP99.90to99.95   \n",
       "\n",
       "                                                INFO                  FORMAT  \\\n",
       "0  AC=1;AF=0.250;AN=4;BaseQRankSum=1.59;DP=141;Ex...  GT:AD:DP:GQ:PGT:PID:PL   \n",
       "1  AC=1;AF=0.250;AN=4;BaseQRankSum=-1.350e-01;DP=...  GT:AD:DP:GQ:PGT:PID:PL   \n",
       "\n",
       "                           B04529                                B04530  \n",
       "0  0/0:126,0:126:75:.:.:0,75,3085  0/1:7,2:9:43:0|1:10150_CT_C:43,0,135  \n",
       "1     0/0:106,0:106:0:.:.:0,0,954  0/1:6,2:8:44:0|1:10150_CT_C:44,0,134  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# f = '/projects/da_workspace/DA-337/haplotype/B04522_27/haplotype_caller-1.1.4/joint/B04522_B04524_B04525_B04526_B04527/379/joint_calling.exclude_no_variation.snpeff.hg19_multianno.vcf.gz'\n",
    "f ='/projects/da_workspace/DA-337/haplotype/B04529_30/haplotype_caller-1.1.4/joint/B04529_B04530/382/joint_calling.exclude_no_variation.snpeff.hg19_multianno.vcf.gz'\n",
    "df = pd.read_csv(f, sep='\\t', comment='#', compression='gzip', low_memory=False, header=None)\n",
    "df.columns = ['CHROM', 'POS','ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', \n",
    "              'FORMAT', 'B04529', 'B04530']\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6836300, 11)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter step by  step\n",
    "df = df[df.FILTER=='PASS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6210235, 11)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gnomAD_anno(info):\n",
    "    sps = [(i.split('=')) for i in info.split(';')[:-1]]\n",
    "#     sps  = [(i.split('=')) for i in items]\n",
    "    d = {i[0]: i[1] for i in sps if len(i) == 2}\n",
    "    ks = ['cosmic70',  'ExAC_ALL', 'gnomAD_genome_ALL'] #'Gene.refGene', 'AAChange.refGene',\n",
    "#     return {k: d[k] for k in ks}\n",
    "    return pd.Series([d[k] for k in ks])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep it easy for now pick HIGH, MODERATE and then LOW and the first transcript. could pull out all transcripts if needed\n",
    "def get_annotations(info):\n",
    "    effs = info.split('EFF=')[1].split(',')\n",
    "#     effs = eff_line.split(',')\n",
    "    #     extract impact, impact_type, amino acid change, and gene\n",
    "    effs = ['@'.join(list(np.array(re.split('\\(|\\|',ef))[[0,1,2,4,6,9]])) for ef in effs if ('HIGH' in  ef) or ('MODERATE' in ef) or ('LOW' in ef)]\n",
    "    effs = list(set(effs))\n",
    "    high = [ef for ef in effs if 'HIGH' in ef]\n",
    "    moderate = [ef for ef in effs if 'MODERATE' in ef]\n",
    "    low = [ef for ef in effs if 'LOW' in ef]\n",
    "#     print(high, moderate, low)\n",
    "    if high:\n",
    "#         anno = '$'.join(high)\n",
    "        anno = high[0]\n",
    "    elif moderate:\n",
    "#         anno = '$'.join(moderate)\n",
    "        anno = moderate[0]\n",
    "    elif low:\n",
    "#         anno = '$'.join(low)\n",
    "        anno = low[0]\n",
    "    else:\n",
    "        print('ERROR!')\n",
    "        pass\n",
    "#     make sure the genes have the same name\n",
    "#     genes = [ef.split('@')[2] for ef in effs]\n",
    "#     print(anno)\n",
    "    # anno= list of annotations\n",
    "    return pd.Series(anno.split('@'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "libs = [i for i in df.columns if i.startswith('B')]\n",
    "target_feat = 'GT'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_genotype(target_feat, features, x):\n",
    "    feats = features.split(':')\n",
    "    x = x.split(':')\n",
    "    d = {k:v for k, v in zip(feats, x)}\n",
    "    return d[target_feat]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0/1'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_genotype(target_feat, 'GT:AD:DP:GQ:PL',  '0/1:36,23:59:99:827,0,1516')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/da/lib/python3.7/site-packages/pandas/core/frame.py:3391: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self[k1] = value[k2]\n",
      "/projects/da_workspace/software/miniconda/envs/da/lib/python3.7/site-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/projects/da_workspace/software/miniconda/envs/da/lib/python3.7/site-packages/pandas/core/frame.py:3940: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  errors=errors)\n"
     ]
    }
   ],
   "source": [
    "dfhml = df[(df['INFO'].str.contains(\"HIGH\"))|(df['INFO'].str.contains(\"MODERATE\"))|(df['INFO'].str.contains(\"LOW\"))]\n",
    "if not dfhml.empty:\n",
    "    dfhml[['variant', 'impact','mutation_type','aa_change','gene','transcript']] = dfhml['INFO'].apply(lambda x: get_annotations(x))   \n",
    "    dfhml[['cosmic70',  'ExAC_ALL', 'gnomAD_genome_ALL']] = dfhml['INFO'].apply(lambda x: get_gnomAD_anno(x))\n",
    "    for lib in libs:\n",
    "        dfhml['_'.join([lib, 'gt'])] = dfhml[['FORMAT', lib]].apply(lambda x: get_genotype(target_feat, x['FORMAT'], x[lib]), axis=1)\n",
    "    dfhml.drop(['INFO', 'FILTER'], axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(44850, 20)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfhml.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>B04529</th>\n",
       "      <th>B04530</th>\n",
       "      <th>variant</th>\n",
       "      <th>impact</th>\n",
       "      <th>mutation_type</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>ExAC_ALL</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>B04529_gt</th>\n",
       "      <th>B04530_gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1</td>\n",
       "      <td>13656</td>\n",
       "      <td>.</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>3362.46</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:13,46:59:99:1876,0,402</td>\n",
       "      <td>0/1:33,38:71:99:1496,0,1267</td>\n",
       "      <td>splice_donor_variant+splice_region_variant+int...</td>\n",
       "      <td>HIGH</td>\n",
       "      <td></td>\n",
       "      <td>n.734+2_734+3delAG</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENST00000518655</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.4901</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1</td>\n",
       "      <td>15903</td>\n",
       "      <td>rs557514207</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>237.55</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:0,0:0:.:0,0,0</td>\n",
       "      <td>1/1:0,7:7:21:250,21,0</td>\n",
       "      <td>splice_region_variant+non_coding_transcript_ex...</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td>n.1346dupG</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENST00000538476</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5042</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>1</td>\n",
       "      <td>728421</td>\n",
       "      <td>rs780398542</td>\n",
       "      <td>AGT</td>\n",
       "      <td>A</td>\n",
       "      <td>627.88</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/0:37,0:37:99:0,101,1620</td>\n",
       "      <td>0/1:16,17:33:99:636,0,594</td>\n",
       "      <td>splice_donor_variant+intron_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td></td>\n",
       "      <td>n.258+1_258+2delGT</td>\n",
       "      <td>RP11-206L10.9</td>\n",
       "      <td>ENST00000586288</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1803</th>\n",
       "      <td>1</td>\n",
       "      <td>877831</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4465.09</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,55:55:99:2277,165,0</td>\n",
       "      <td>1/1:0,53:53:99:2204,159,0</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>MISSENSE</td>\n",
       "      <td>p.Trp169Arg/c.505T&gt;C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>.</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1804</th>\n",
       "      <td>1</td>\n",
       "      <td>878314</td>\n",
       "      <td>rs142558220</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>704.88</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/0:51,0:51:99:0,120,1800</td>\n",
       "      <td>0/1:17,20:37:99:713,0,650</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SILENT</td>\n",
       "      <td>p.Gly387Gly/c.1161G&gt;C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>.</td>\n",
       "      <td>0.1068</td>\n",
       "      <td>0.0659</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM     POS           ID  REF ALT     QUAL          FORMAT  \\\n",
       "15       1   13656            .  CAG   C  3362.46  GT:AD:DP:GQ:PL   \n",
       "39       1   15903  rs557514207    G  GC   237.55  GT:AD:DP:GQ:PL   \n",
       "1109     1  728421  rs780398542  AGT   A   627.88  GT:AD:DP:GQ:PL   \n",
       "1803     1  877831    rs6672356    T   C  4465.09  GT:AD:DP:GQ:PL   \n",
       "1804     1  878314  rs142558220    G   C   704.88  GT:AD:DP:GQ:PL   \n",
       "\n",
       "                          B04529                       B04530  \\\n",
       "15    0/1:13,46:59:99:1876,0,402  0/1:33,38:71:99:1496,0,1267   \n",
       "39             ./.:0,0:0:.:0,0,0        1/1:0,7:7:21:250,21,0   \n",
       "1109   0/0:37,0:37:99:0,101,1620    0/1:16,17:33:99:636,0,594   \n",
       "1803   1/1:0,55:55:99:2277,165,0    1/1:0,53:53:99:2204,159,0   \n",
       "1804   0/0:51,0:51:99:0,120,1800    0/1:17,20:37:99:713,0,650   \n",
       "\n",
       "                                                variant    impact  \\\n",
       "15    splice_donor_variant+splice_region_variant+int...      HIGH   \n",
       "39    splice_region_variant+non_coding_transcript_ex...       LOW   \n",
       "1109                splice_donor_variant+intron_variant      HIGH   \n",
       "1803                                   missense_variant  MODERATE   \n",
       "1804                                 synonymous_variant       LOW   \n",
       "\n",
       "     mutation_type              aa_change           gene       transcript  \\\n",
       "15                     n.734+2_734+3delAG        DDX11L1  ENST00000518655   \n",
       "39                             n.1346dupG         WASH7P  ENST00000538476   \n",
       "1109                   n.258+1_258+2delGT  RP11-206L10.9  ENST00000586288   \n",
       "1803      MISSENSE   p.Trp169Arg/c.505T>C         SAMD11  ENST00000455979   \n",
       "1804        SILENT  p.Gly387Gly/c.1161G>C         SAMD11  ENST00000341065   \n",
       "\n",
       "     cosmic70 ExAC_ALL gnomAD_genome_ALL B04529_gt B04530_gt  \n",
       "15          .        .            0.4901       0/1       0/1  \n",
       "39          .        .            0.5042       ./.       1/1  \n",
       "1109        .        .            0.0010       0/0       0/1  \n",
       "1803        .   0.9999            0.9998       1/1       1/1  \n",
       "1804        .   0.1068            0.0659       0/0       0/1  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfhml.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/da/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    }
   ],
   "source": [
    "dfhml['gnomAD_genome_ALL_tmp'] = dfhml['gnomAD_genome_ALL'].replace('.', .0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 25)"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dff[dff.gnomAD_genome_ALL.isna()].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "dff = dfhml[(dfhml.gnomAD_genome_ALL_tmp.astype(float) <0.01)|(dfhml.gnomAD_genome_ALL_tmp.isna())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>B04529</th>\n",
       "      <th>B04530</th>\n",
       "      <th>variant</th>\n",
       "      <th>...</th>\n",
       "      <th>mutation_type</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>ExAC_ALL</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>B04529_gt</th>\n",
       "      <th>B04530_gt</th>\n",
       "      <th>gnomAD_genome_ALL_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>1</td>\n",
       "      <td>728421</td>\n",
       "      <td>rs780398542</td>\n",
       "      <td>AGT</td>\n",
       "      <td>A</td>\n",
       "      <td>627.88</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/0:37,0:37:99:0,101,1620</td>\n",
       "      <td>0/1:16,17:33:99:636,0,594</td>\n",
       "      <td>splice_donor_variant+intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>n.258+1_258+2delGT</td>\n",
       "      <td>RP11-206L10.9</td>\n",
       "      <td>ENST00000586288</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2066</th>\n",
       "      <td>1</td>\n",
       "      <td>955724</td>\n",
       "      <td>rs750933418</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>452.88</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:20,14:34:99:461,0,726</td>\n",
       "      <td>0/0:35,0:35:84:0,84,1260</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>MISSENSE</td>\n",
       "      <td>p.Asp58Asn/c.172G&gt;A</td>\n",
       "      <td>AGRN</td>\n",
       "      <td>ENST00000379370</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2552</th>\n",
       "      <td>1</td>\n",
       "      <td>1133061</td>\n",
       "      <td>rs555101871</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>584.88</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>0/1:22,20:42:99:0|1:1133061_C_T:593,0,783</td>\n",
       "      <td>0/0:32,0:32:87:.:.:0,87,1305</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>MISSENSE</td>\n",
       "      <td>p.Pro619Leu/c.1856C&gt;T</td>\n",
       "      <td>TTLL10</td>\n",
       "      <td>ENST00000379289</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0070</td>\n",
       "      <td>0.0056</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0.0056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2566</th>\n",
       "      <td>1</td>\n",
       "      <td>1139873</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>28.88</td>\n",
       "      <td>GT:AD:DP:GQ:PGT:PID:PL</td>\n",
       "      <td>0/1:23,5:28:37:0|1:1139873_T_G:37,0,712</td>\n",
       "      <td>0/0:39,0:39:33:.:.:0,33,1042</td>\n",
       "      <td>splice_region_variant+intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>c.95-7A&gt;C</td>\n",
       "      <td>TNFRSF18</td>\n",
       "      <td>ENST00000486728</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2680</th>\n",
       "      <td>1</td>\n",
       "      <td>1167971</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1137.88</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:25,32:57:99:1146,0,836</td>\n",
       "      <td>0/0:39,0:39:99:0,104,1665</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>MISSENSE</td>\n",
       "      <td>p.Glu105Gln/c.313G&gt;C</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>ENST00000379198</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM      POS           ID  REF ALT     QUAL                  FORMAT  \\\n",
       "1109     1   728421  rs780398542  AGT   A   627.88          GT:AD:DP:GQ:PL   \n",
       "2066     1   955724  rs750933418    G   A   452.88          GT:AD:DP:GQ:PL   \n",
       "2552     1  1133061  rs555101871    C   T   584.88  GT:AD:DP:GQ:PGT:PID:PL   \n",
       "2566     1  1139873            .    T   G    28.88  GT:AD:DP:GQ:PGT:PID:PL   \n",
       "2680     1  1167971            .    G   C  1137.88          GT:AD:DP:GQ:PL   \n",
       "\n",
       "                                         B04529                        B04530  \\\n",
       "1109                  0/0:37,0:37:99:0,101,1620     0/1:16,17:33:99:636,0,594   \n",
       "2066                  0/1:20,14:34:99:461,0,726      0/0:35,0:35:84:0,84,1260   \n",
       "2552  0/1:22,20:42:99:0|1:1133061_C_T:593,0,783  0/0:32,0:32:87:.:.:0,87,1305   \n",
       "2566    0/1:23,5:28:37:0|1:1139873_T_G:37,0,712  0/0:39,0:39:33:.:.:0,33,1042   \n",
       "2680                 0/1:25,32:57:99:1146,0,836     0/0:39,0:39:99:0,104,1665   \n",
       "\n",
       "                                   variant  ... mutation_type  \\\n",
       "1109   splice_donor_variant+intron_variant  ...                 \n",
       "2066                      missense_variant  ...      MISSENSE   \n",
       "2552                      missense_variant  ...      MISSENSE   \n",
       "2566  splice_region_variant+intron_variant  ...                 \n",
       "2680                      missense_variant  ...      MISSENSE   \n",
       "\n",
       "                  aa_change           gene       transcript cosmic70 ExAC_ALL  \\\n",
       "1109     n.258+1_258+2delGT  RP11-206L10.9  ENST00000586288        .        .   \n",
       "2066    p.Asp58Asn/c.172G>A           AGRN  ENST00000379370        .   0.0002   \n",
       "2552  p.Pro619Leu/c.1856C>T         TTLL10  ENST00000379289        .   0.0070   \n",
       "2566              c.95-7A>C       TNFRSF18  ENST00000486728        .        .   \n",
       "2680   p.Glu105Gln/c.313G>C        B3GALT6  ENST00000379198        .        .   \n",
       "\n",
       "     gnomAD_genome_ALL B04529_gt B04530_gt gnomAD_genome_ALL_tmp  \n",
       "1109            0.0010       0/0       0/1                0.0010  \n",
       "2066            0.0003       0/1       0/0                0.0003  \n",
       "2552            0.0056       0/1       0/0                0.0056  \n",
       "2566                 .       0/1       0/0                     0  \n",
       "2680                 .       0/1       0/0                     0  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(3074, 21)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dff.head()\n",
    "dff.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# filter based on genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "ps = ['0/1', '1/0', '1/1', '0|1', '1|0', '1|1', './.', '.|']\n",
    "ns = ['0/0', '0|0', './.', '.|.']\n",
    "# affected = ['B04522', 'B04524', 'B04525']\n",
    "# unaffected = ['B04526', 'B04527']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "affecteds (B04529 and B04530) \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfff = dff[(dff.B04529_gt.isin(ps))&(dff.B04530_gt.isin(ps))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfff.drop('gnomAD_genome_ALL_tmp', axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>B04529</th>\n",
       "      <th>B04530</th>\n",
       "      <th>variant</th>\n",
       "      <th>impact</th>\n",
       "      <th>mutation_type</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>ExAC_ALL</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>B04529_gt</th>\n",
       "      <th>B04530_gt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>96329</th>\n",
       "      <td>1</td>\n",
       "      <td>41847872</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCCCGCGCCGGGACGCCCGCCTACTTCGGCGGCTGCAAGGGCGGC...</td>\n",
       "      <td>5616.13</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:6,27:33:99:2753,164,0</td>\n",
       "      <td>1/1:5,28:33:99:2879,171,0</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td></td>\n",
       "      <td>p.Gln337fs/c.1008_1009insGGGACGCCCGCCTACTTCGGC...</td>\n",
       "      <td>FOXO6</td>\n",
       "      <td>ENST00000372591</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268085</th>\n",
       "      <td>1</td>\n",
       "      <td>142805095</td>\n",
       "      <td>rs377094966</td>\n",
       "      <td>TCTGC</td>\n",
       "      <td>T</td>\n",
       "      <td>477.46</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:104,14:120:99:269,0,4375</td>\n",
       "      <td>0/1:92,12:104:99:218,0,4013</td>\n",
       "      <td>splice_region_variant+non_coding_transcript_ex...</td>\n",
       "      <td>LOW</td>\n",
       "      <td></td>\n",
       "      <td>n.351_354delGCAG</td>\n",
       "      <td>RP11-423O2.5</td>\n",
       "      <td>ENST00000423385</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       CHROM        POS           ID    REF  \\\n",
       "96329      1   41847872            .      G   \n",
       "268085     1  142805095  rs377094966  TCTGC   \n",
       "\n",
       "                                                      ALT     QUAL  \\\n",
       "96329   GGCCCGCGCCGGGACGCCCGCCTACTTCGGCGGCTGCAAGGGCGGC...  5616.13   \n",
       "268085                                                  T   477.46   \n",
       "\n",
       "                FORMAT                        B04529  \\\n",
       "96329   GT:AD:DP:GQ:PL     1/1:6,27:33:99:2753,164,0   \n",
       "268085  GT:AD:DP:GQ:PL  0/1:104,14:120:99:269,0,4375   \n",
       "\n",
       "                             B04530  \\\n",
       "96329     1/1:5,28:33:99:2879,171,0   \n",
       "268085  0/1:92,12:104:99:218,0,4013   \n",
       "\n",
       "                                                  variant impact  \\\n",
       "96329                                  frameshift_variant   HIGH   \n",
       "268085  splice_region_variant+non_coding_transcript_ex...    LOW   \n",
       "\n",
       "       mutation_type                                          aa_change  \\\n",
       "96329                 p.Gln337fs/c.1008_1009insGGGACGCCCGCCTACTTCGGC...   \n",
       "268085                                                 n.351_354delGCAG   \n",
       "\n",
       "                gene       transcript cosmic70 ExAC_ALL gnomAD_genome_ALL  \\\n",
       "96329          FOXO6  ENST00000372591        .        .                 .   \n",
       "268085  RP11-423O2.5  ENST00000423385        .        .                 .   \n",
       "\n",
       "       B04529_gt B04530_gt  \n",
       "96329        1/1       1/1  \n",
       "268085       0/1       0/1  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfff.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/da_workspace/DA-337/haplotype/B04529_30/haplotype_caller-1.1.4/joint/B04529_B04530/382/B04529_B04530_potential_causal_contributing_coding_variants_20190912.tsv'\n",
    "dfff.to_csv(of, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['FOXO6', 'RP11-423O2.5', 'FMO5', 'ANKRD45', 'RP11-318C24.2',\n",
       "       'SUCLA2P3', 'KANSL3', 'ANKRD30BL', 'AC097532.2', 'ZNF806', 'FIGN',\n",
       "       'ITGAV', 'PLEKHM3', 'TPRXL', 'POU5F1P6', 'RP11-680B3.2', 'KPNA4',\n",
       "       'RP11-1281K21.1', 'CENPK', 'PRR7-AS1', 'C5orf60', 'CCDC90A',\n",
       "       'HLA-DRB6', 'HLA-DQA1', 'HLA-DQB1', 'HLA-DQA2', 'RP11-569A2.2',\n",
       "       'SYNE1', 'MUC17', 'TRBV6-1', 'PRSS3P1', 'PRSS3P2',\n",
       "       'RP11-1102P16.1', 'EXOSC3', 'RP11-764K9.4', 'PCSK5', 'SVEP1',\n",
       "       'PTGR1', 'RALGDS', 'ARHGAP12', 'SLC9B1P3', 'GPRIN2', 'A1CF',\n",
       "       'PAPSS2', 'BTAF1', 'RP11-500G10.1', 'MUC6', 'ZNF195', 'OVCH2',\n",
       "       'KRTAP5-7', 'DDX6', 'TNFRSF1A', 'RP11-551L14.1', 'KRT4',\n",
       "       'RP11-248E9.1', 'KTN1-AS1', 'RP11-1152H15.1', 'GOLGA6L2', 'DMXL2',\n",
       "       'TPSB2', 'FLYWCH1', 'KIAA0430', 'MVP', 'CTD-2522B17.8',\n",
       "       'LINC00273', 'AC004158.2', 'RP11-55L4.1', 'NCOR1', 'RP11-344E13.3',\n",
       "       'MAP2K3', 'KCNJ12', 'RP11-260A9.6', 'SLC13A2', 'TANC2', 'ABCA6',\n",
       "       'CTD-2357A8.3', 'BAIAP2', 'OSBPL1A', 'KDSR', 'RP11-389J22.1',\n",
       "       'MADCAM1', 'CTB-25B13.12', 'KIAA0355', 'LMTK3', 'CST5', 'FAM182B',\n",
       "       'FAM182A', 'FRG1B', 'C20orf112', 'AC007731.1', 'SMCR7L',\n",
       "       'AC012078.2', 'AC134882.1', 'AC145205.1', 'AC011841.1',\n",
       "       'AL592183.1', 'AC011043.1', 'AL356585.1', 'AC018692.1',\n",
       "       'AC145212.1'], dtype=object)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(161, 20)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfff.gene.unique()\n",
    "dfff.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get the vcf entries based on index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FORMAT</th>\n",
       "      <th>B04529</th>\n",
       "      <th>B04530</th>\n",
       "      <th>variant</th>\n",
       "      <th>...</th>\n",
       "      <th>mutation_type</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>ExAC_ALL</th>\n",
       "      <th>gnomAD_genome_ALL</th>\n",
       "      <th>B04529_gt</th>\n",
       "      <th>B04530_gt</th>\n",
       "      <th>gnomAD_genome_ALL_tmp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1</td>\n",
       "      <td>13656</td>\n",
       "      <td>.</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>3362.46</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/1:13,46:59:99:1876,0,402</td>\n",
       "      <td>0/1:33,38:71:99:1496,0,1267</td>\n",
       "      <td>splice_donor_variant+splice_region_variant+int...</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>n.734+2_734+3delAG</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>ENST00000518655</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.4901</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.4901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1</td>\n",
       "      <td>15903</td>\n",
       "      <td>rs557514207</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>237.55</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>./.:0,0:0:.:0,0,0</td>\n",
       "      <td>1/1:0,7:7:21:250,21,0</td>\n",
       "      <td>splice_region_variant+non_coding_transcript_ex...</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>n.1346dupG</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>ENST00000538476</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.5042</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0.5042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>1</td>\n",
       "      <td>728421</td>\n",
       "      <td>rs780398542</td>\n",
       "      <td>AGT</td>\n",
       "      <td>A</td>\n",
       "      <td>627.88</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/0:37,0:37:99:0,101,1620</td>\n",
       "      <td>0/1:16,17:33:99:636,0,594</td>\n",
       "      <td>splice_donor_variant+intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td>n.258+1_258+2delGT</td>\n",
       "      <td>RP11-206L10.9</td>\n",
       "      <td>ENST00000586288</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.0010</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1803</th>\n",
       "      <td>1</td>\n",
       "      <td>877831</td>\n",
       "      <td>rs6672356</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4465.09</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>1/1:0,55:55:99:2277,165,0</td>\n",
       "      <td>1/1:0,53:53:99:2204,159,0</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>MISSENSE</td>\n",
       "      <td>p.Trp169Arg/c.505T&gt;C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>ENST00000455979</td>\n",
       "      <td>.</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>0.9998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1804</th>\n",
       "      <td>1</td>\n",
       "      <td>878314</td>\n",
       "      <td>rs142558220</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>704.88</td>\n",
       "      <td>GT:AD:DP:GQ:PL</td>\n",
       "      <td>0/0:51,0:51:99:0,120,1800</td>\n",
       "      <td>0/1:17,20:37:99:713,0,650</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>SILENT</td>\n",
       "      <td>p.Gly387Gly/c.1161G&gt;C</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>.</td>\n",
       "      <td>0.1068</td>\n",
       "      <td>0.0659</td>\n",
       "      <td>0/0</td>\n",
       "      <td>0/1</td>\n",
       "      <td>0.0659</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     CHROM     POS           ID  REF ALT     QUAL          FORMAT  \\\n",
       "15       1   13656            .  CAG   C  3362.46  GT:AD:DP:GQ:PL   \n",
       "39       1   15903  rs557514207    G  GC   237.55  GT:AD:DP:GQ:PL   \n",
       "1109     1  728421  rs780398542  AGT   A   627.88  GT:AD:DP:GQ:PL   \n",
       "1803     1  877831    rs6672356    T   C  4465.09  GT:AD:DP:GQ:PL   \n",
       "1804     1  878314  rs142558220    G   C   704.88  GT:AD:DP:GQ:PL   \n",
       "\n",
       "                          B04529                       B04530  \\\n",
       "15    0/1:13,46:59:99:1876,0,402  0/1:33,38:71:99:1496,0,1267   \n",
       "39             ./.:0,0:0:.:0,0,0        1/1:0,7:7:21:250,21,0   \n",
       "1109   0/0:37,0:37:99:0,101,1620    0/1:16,17:33:99:636,0,594   \n",
       "1803   1/1:0,55:55:99:2277,165,0    1/1:0,53:53:99:2204,159,0   \n",
       "1804   0/0:51,0:51:99:0,120,1800    0/1:17,20:37:99:713,0,650   \n",
       "\n",
       "                                                variant  ... mutation_type  \\\n",
       "15    splice_donor_variant+splice_region_variant+int...  ...                 \n",
       "39    splice_region_variant+non_coding_transcript_ex...  ...                 \n",
       "1109                splice_donor_variant+intron_variant  ...                 \n",
       "1803                                   missense_variant  ...      MISSENSE   \n",
       "1804                                 synonymous_variant  ...        SILENT   \n",
       "\n",
       "                  aa_change           gene       transcript cosmic70 ExAC_ALL  \\\n",
       "15       n.734+2_734+3delAG        DDX11L1  ENST00000518655        .        .   \n",
       "39               n.1346dupG         WASH7P  ENST00000538476        .        .   \n",
       "1109     n.258+1_258+2delGT  RP11-206L10.9  ENST00000586288        .        .   \n",
       "1803   p.Trp169Arg/c.505T>C         SAMD11  ENST00000455979        .   0.9999   \n",
       "1804  p.Gly387Gly/c.1161G>C         SAMD11  ENST00000341065        .   0.1068   \n",
       "\n",
       "     gnomAD_genome_ALL B04529_gt B04530_gt gnomAD_genome_ALL_tmp  \n",
       "15              0.4901       0/1       0/1                0.4901  \n",
       "39              0.5042       ./.       1/1                0.5042  \n",
       "1109            0.0010       0/0       0/1                0.0010  \n",
       "1803            0.9998       1/1       1/1                0.9998  \n",
       "1804            0.0659       0/0       0/1                0.0659  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfhml.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfvcf = df.reindex(dfff.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/da_workspace/DA-337/haplotype/B04529_30/haplotype_caller-1.1.4/joint/B04529_B04530/382/B04529_B04530_vcf_20190910.tsv'\n",
    "dfvcf.to_csv(of, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
