{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Thanks very much - Veronica, Ru Chen and I went over the pedigrees this morning and reverified the identities. I have attached the sample list here.\n",
    "\n",
    "The sibship defined by B04522, B04523, B04524, B04525, B04526, B04527 and B04528 should all be compared to each other. Since B04523 and B04528 could not be collected, we are left with assuming B04522, B04524, B04525, B04526, B04527 are full sibs. Of those 5 samples, 3 are affected and 2 are unaffected. We are interested in the list of rare variants (either coding or non-coding) that are shared among the affecteds (B04522, B04524 and B04525) and not shared with the unaffecteds (B04526, B04527). \n",
    "These can be homozygous (autosomal recessive model) or heterozygous (autosomal dominant model).\n",
    "\n",
    "B04529 and B04530 are third cousins to each other, both affected. We are interested in the list of rare variants that are shared between them.\n",
    "\n",
    "The other samples are all individuals from different families, unrelated to each other. It is possible (though not likely, given what we know about the genetic architecture of nonsyndromic brain aneurysms) that one or more of these samples has a rare variant in or near a gene that is flagged by one of the above approaches.\n",
    "If there is an algorithmic way to approach that, I'd appreciate you letting me know, but as far as I am aware the onus will be on me to scroll through the variant lists for those individuals one at a time to see by eye whether any rare variants in these singletons are in the same genes as are flagged by the filtering of the multiplex sibship and/or the two third cousins."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://www.bcgsc.ca/wiki/display/DA/Fetch+file+paths+and+other+meta-data+using+the+BioApps+API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "wkdir = '/projects/da_workspace/DA-337/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>external_identifier</th>\n",
       "      <th>project</th>\n",
       "      <th>upper_protocol</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Family #</th>\n",
       "      <th>Family position</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Affected Status</th>\n",
       "      <th>Relationship to Other Samples in this Run</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B04527</td>\n",
       "      <td>IBA1852</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-6</td>\n",
       "      <td>Female</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>Sibling to B04522, B04523, B04524, B04525, B04...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B04528</td>\n",
       "      <td>IBA1853</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>to exclude</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-3</td>\n",
       "      <td>Female</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>Sibling to B04522, B04523, B04524, B04525, B04...</td>\n",
       "      <td>EXCLUDE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     name external_identifier               project  \\\n",
       "0  B04527             IBA1852  William Gibson (335)   \n",
       "1  B04528             IBA1853  William Gibson (335)   \n",
       "\n",
       "               upper_protocol  Unnamed: 4 Family # Family position     Sex  \\\n",
       "0  Genome Shotgun PCRFree 1.3         NaN      BC1            II-6  Female   \n",
       "1  Genome Shotgun PCRFree 1.3  to exclude      BC1            II-3  Female   \n",
       "\n",
       "  Affected Status          Relationship to Other Samples in this Run  \\\n",
       "0      Unaffected  Sibling to B04522, B04523, B04524, B04525, B04...   \n",
       "1      Unaffected  Sibling to B04522, B04523, B04524, B04525, B04...   \n",
       "\n",
       "  Unnamed: 10  \n",
       "0         NaN  \n",
       "1     EXCLUDE  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = '/projects/da_workspace/DA-337/DA_337_WilliamGibsonPedigree_08_09_2019_WITH_FAM_ID.xlsx'\n",
    "df = pd.read_excel(f)\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>external_identifier</th>\n",
       "      <th>project</th>\n",
       "      <th>upper_protocol</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Family #</th>\n",
       "      <th>Family position</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Affected Status</th>\n",
       "      <th>Relationship to Other Samples in this Run</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B04527</td>\n",
       "      <td>IBA1852</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-6</td>\n",
       "      <td>Female</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>Sibling to B04522, B04523, B04524, B04525, B04...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B04528</td>\n",
       "      <td>IBA1853</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>to exclude</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-3</td>\n",
       "      <td>Female</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>Sibling to B04522, B04523, B04524, B04525, B04...</td>\n",
       "      <td>EXCLUDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>B04530</td>\n",
       "      <td>IBA1868</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC2</td>\n",
       "      <td>IV-2</td>\n",
       "      <td>Female</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Third cousin to B04529/IBA1824</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B04525</td>\n",
       "      <td>IBA1834</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-5</td>\n",
       "      <td>Female</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Sibling to B04522, B04523, B04524, B04526, B04...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>B04523</td>\n",
       "      <td>IBA1835</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>to exclude</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-1</td>\n",
       "      <td>Male</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Sibling to B04522, B04524, B04525, B04526, B04...</td>\n",
       "      <td>EXCLUDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>B04531</td>\n",
       "      <td>IBA1863</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC3</td>\n",
       "      <td>III-2</td>\n",
       "      <td>Female</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Unrelated</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>B04522</td>\n",
       "      <td>IBA1830</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-2</td>\n",
       "      <td>Male</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Sibling to B04523, B04524, B04525, B04526, B04...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>B04524</td>\n",
       "      <td>IBA1842</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-4</td>\n",
       "      <td>Male</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Sibling to B04522, B04523, B04525, B04526, B04...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>B04526</td>\n",
       "      <td>IBA1850</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC1</td>\n",
       "      <td>II-7</td>\n",
       "      <td>Female</td>\n",
       "      <td>Unaffected</td>\n",
       "      <td>Sibling to B04522, B04523, B04524, B04525, B04...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>B04529</td>\n",
       "      <td>IBA1824</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC2</td>\n",
       "      <td>IV-1</td>\n",
       "      <td>Female</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Third cousin to B04530/IBA1868</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>B04533</td>\n",
       "      <td>IBA1801</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC5</td>\n",
       "      <td>III-1</td>\n",
       "      <td>Female</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Unrelated</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>B04534</td>\n",
       "      <td>IBA1837</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC6</td>\n",
       "      <td>__</td>\n",
       "      <td>Female</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Unrelated</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>B04532</td>\n",
       "      <td>IBA1823</td>\n",
       "      <td>William Gibson (335)</td>\n",
       "      <td>Genome Shotgun PCRFree 1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>BC4</td>\n",
       "      <td>III-3</td>\n",
       "      <td>Female</td>\n",
       "      <td>Affected</td>\n",
       "      <td>Unrelated</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      name external_identifier               project  \\\n",
       "0   B04527             IBA1852  William Gibson (335)   \n",
       "1   B04528             IBA1853  William Gibson (335)   \n",
       "2   B04530             IBA1868  William Gibson (335)   \n",
       "3   B04525             IBA1834  William Gibson (335)   \n",
       "4   B04523             IBA1835  William Gibson (335)   \n",
       "5   B04531             IBA1863  William Gibson (335)   \n",
       "6   B04522             IBA1830  William Gibson (335)   \n",
       "7   B04524             IBA1842  William Gibson (335)   \n",
       "8   B04526             IBA1850  William Gibson (335)   \n",
       "9   B04529             IBA1824  William Gibson (335)   \n",
       "10  B04533             IBA1801  William Gibson (335)   \n",
       "11  B04534             IBA1837  William Gibson (335)   \n",
       "12  B04532             IBA1823  William Gibson (335)   \n",
       "\n",
       "                upper_protocol  Unnamed: 4 Family # Family position     Sex  \\\n",
       "0   Genome Shotgun PCRFree 1.3         NaN      BC1            II-6  Female   \n",
       "1   Genome Shotgun PCRFree 1.3  to exclude      BC1            II-3  Female   \n",
       "2   Genome Shotgun PCRFree 1.3         NaN      BC2            IV-2  Female   \n",
       "3   Genome Shotgun PCRFree 1.3         NaN      BC1            II-5  Female   \n",
       "4   Genome Shotgun PCRFree 1.3  to exclude      BC1            II-1    Male   \n",
       "5   Genome Shotgun PCRFree 1.3         NaN      BC3           III-2  Female   \n",
       "6   Genome Shotgun PCRFree 1.3         NaN      BC1            II-2    Male   \n",
       "7   Genome Shotgun PCRFree 1.3         NaN      BC1            II-4    Male   \n",
       "8   Genome Shotgun PCRFree 1.3         NaN      BC1            II-7  Female   \n",
       "9   Genome Shotgun PCRFree 1.3         NaN      BC2            IV-1  Female   \n",
       "10  Genome Shotgun PCRFree 1.3         NaN      BC5           III-1  Female   \n",
       "11  Genome Shotgun PCRFree 1.3         NaN      BC6              __  Female   \n",
       "12  Genome Shotgun PCRFree 1.3         NaN      BC4           III-3  Female   \n",
       "\n",
       "   Affected Status          Relationship to Other Samples in this Run  \\\n",
       "0       Unaffected  Sibling to B04522, B04523, B04524, B04525, B04...   \n",
       "1       Unaffected  Sibling to B04522, B04523, B04524, B04525, B04...   \n",
       "2         Affected                     Third cousin to B04529/IBA1824   \n",
       "3         Affected  Sibling to B04522, B04523, B04524, B04526, B04...   \n",
       "4         Affected  Sibling to B04522, B04524, B04525, B04526, B04...   \n",
       "5         Affected                                          Unrelated   \n",
       "6         Affected  Sibling to B04523, B04524, B04525, B04526, B04...   \n",
       "7         Affected  Sibling to B04522, B04523, B04525, B04526, B04...   \n",
       "8       Unaffected  Sibling to B04522, B04523, B04524, B04525, B04...   \n",
       "9         Affected                     Third cousin to B04530/IBA1868   \n",
       "10        Affected                                          Unrelated   \n",
       "11        Affected                                          Unrelated   \n",
       "12        Affected                                          Unrelated   \n",
       "\n",
       "   Unnamed: 10  \n",
       "0          NaN  \n",
       "1      EXCLUDE  \n",
       "2          NaN  \n",
       "3          NaN  \n",
       "4      EXCLUDE  \n",
       "5          NaN  \n",
       "6          NaN  \n",
       "7          NaN  \n",
       "8          NaN  \n",
       "9          NaN  \n",
       "10         NaN  \n",
       "11         NaN  \n",
       "12         NaN  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# test if permissions are setup correctly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'status': 'error', 'errors': [{'location': 'body', 'name': 'session', 'description': 'Username or password is incorrect.'}]}\n",
      "{'errors': [{'description': 'Either you do not have permission to use this '\n",
      "                            'resource, or you must send a valid web token in '\n",
      "                            'the X-Token header field.',\n",
      "             'name': '403 Forbidden'}],\n",
      " 'status': 'error'}\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "import os\n",
    "import requests\n",
    "import pprint\n",
    "import fileinput\n",
    " \n",
    "URL = 'http://sbs:8100/'\n",
    "# print(URL)\n",
    " \n",
    " \n",
    " \n",
    "# Create session object, which can be used for all requests.\n",
    "request_handle = requests.Session()\n",
    " \n",
    "# Set up appropriate headers.\n",
    "headers = {'Content-Type': 'application/json',\n",
    "           'Accept': 'application/json'}\n",
    " \n",
    "# Send the POST request, with your username & password, converted to json.\n",
    "response = request_handle.post(os.path.join(URL, 'session'),\n",
    "                               json={'username': 'xxxxxx',\n",
    "                                     'password': '######',\n",
    "                                     },\n",
    "                               headers=headers)\n",
    " \n",
    "# Check that the response is OK.\n",
    "if response.status_code == 200:\n",
    "     # Add the authentication token to the headers.\n",
    "     token = response.json().get('token')\n",
    "     headers.update({'X-Token': token})\n",
    "else:\n",
    "     # View error messages.\n",
    "     print(response.json())\n",
    "# response = request_handle.get(os.path.join(URL, 'patient_analysis'),\n",
    "#                           params={'pog_id': 'POG517, POG1022',\n",
    "#                                   'deleted': ''},\n",
    "#                           headers=headers)\n",
    "\n",
    "response = request_handle.patch(os.path.join(URL, 'haplotype_analysis/319'),\n",
    "                                json={'status': 'failed'}, \n",
    "                                headers=headers)\n",
    "\n",
    "pprint.pprint(response.json(), indent=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# get all the bams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "libs = ['B04522', 'B04524', 'B04525', 'B04526', 'B04527', 'B04529' ,'B04530', 'B04531', 'B04533', 'B04534', 'B04532']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "B04522 /projects/analysis/analysis32/B04522/merge44485_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04524 /projects/analysis/analysis32/B04524/merge44487_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04525 /projects/analysis/analysis32/B04525/merge44486_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04526 /projects/analysis/analysis32/B04526/merge44490_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04527 /projects/analysis/analysis32/B04527/merge44489_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04529 /projects/analysis/analysis32/B04529/merge44491_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04530 /projects/analysis/analysis32/B04530/merge44488_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04531 /projects/analysis/analysis32/B04531/merge39257_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04533 /projects/analysis/analysis32/B04533/merge39255_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04534 /projects/analysis/analysis32/B04534/merge39253_bwa-mem-0.7.6a-sb/150bp/hg19a\n",
      "B04532 /projects/analysis/analysis32/B04532/merge39256_bwa-mem-0.7.6a-sb/150bp/hg19a\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "import os\n",
    "import requests\n",
    "import pprint\n",
    "import fileinput\n",
    " \n",
    "URL = 'http://sbs:8100/'\n",
    "# print(URL)\n",
    " \n",
    " \n",
    " \n",
    "# Create session object, which can be used for all requests.\n",
    "request_handle = requests.Session()\n",
    " \n",
    "# Set up appropriate headers.\n",
    "headers = {'Content-Type': 'application/json',\n",
    "           'Accept': 'application/json'}\n",
    " \n",
    "# Send the POST request, with your username & password, converted to json.\n",
    "response = request_handle.post(os.path.join(URL, 'session'),\n",
    "                               json={'username': 'xxxx',\n",
    "                                     'password': '####',\n",
    "                                     },\n",
    "                               headers=headers)\n",
    " \n",
    "# Check that the response is OK.\n",
    "if response.status_code == 200:\n",
    "     # Add the authentication token to the headers.\n",
    "     token = response.json().get('token')\n",
    "     headers.update({'X-Token': token})\n",
    "else:\n",
    "     # View error messages.\n",
    "     print(response.json())\n",
    "\n",
    "\n",
    "# response = request_handle.get(os.path.join(URL, 'genome'), \n",
    "#                               params={'library': 'B04522'}, \n",
    "#                               headers=headers).json()\n",
    "# response = request_handle.get(os.path.join(URL, 'merge'), \n",
    "#                               params={'library':libs}, \n",
    "#                               headers=headers)\n",
    "\n",
    "bams = []\n",
    "for library in libs:\n",
    " \n",
    "#     library = library.rstrip()\n",
    "    response = request_handle.get(os.path.join(URL, 'merge'), \n",
    "                                  params={'library':library, 'status':'production'}, \n",
    "                                  headers=headers)\n",
    " \n",
    " \n",
    "    print (library, response.json()[0]['data_path'])\n",
    "    bam = [library, response.json()[0]['data_path']]\n",
    "    bams.append(bam)\n",
    "#     pprint.pprint(response.json(), indent=1)\n",
    "\n",
    "# pprint.pprint(response.json(), indent=1)\n",
    "# print(response)\n",
    "\n",
    "df = pd.DataFrame(bams)\n",
    "df.columns = ['lib', 'bam_path']\n",
    "of = ''.join([wkdir, 'bam_path.tsv'])\n",
    "df.to_csv(of, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# run pedigree caller"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "source /gsc/pipelines/pedigree_caller/v1.0.0/venv/bin/activate\n",
    "\n",
    " pedigree_setup -h\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
