{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import matplotlib\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import scipy.stats as ss\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import style, colors\n",
    "from matplotlib import gridspec\n",
    "import random\n",
    "import string\n",
    "import math\n",
    "\n",
    "# matplotlib.style.use('ggplot')\n",
    "%matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (12, 6)\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "this file has the viral gene coordinates:\n",
    "\n",
    "/projects/trans_scratch/software/circos/data/labels/hpv18.labels.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>library</th>\n",
       "      <th>contig(s)</th>\n",
       "      <th>contig-breakpoint(s)</th>\n",
       "      <th>human-breakpoint</th>\n",
       "      <th>human-strand</th>\n",
       "      <th>viral-breakpoint</th>\n",
       "      <th>viral-strand</th>\n",
       "      <th>contig-length(s)</th>\n",
       "      <th>human-length(s)</th>\n",
       "      <th>overlap-length(s)</th>\n",
       "      <th>viral-length(s)</th>\n",
       "      <th>virus</th>\n",
       "      <th>human-gene(s)</th>\n",
       "      <th>flanking-reads</th>\n",
       "      <th>spanning-reads</th>\n",
       "      <th>total-reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A37700</td>\n",
       "      <td>minus_k60_j_369025,plus_k64_f_28387</td>\n",
       "      <td>257,63</td>\n",
       "      <td>3_189526059</td>\n",
       "      <td>-</td>\n",
       "      <td>3631</td>\n",
       "      <td>-</td>\n",
       "      <td>389,123</td>\n",
       "      <td>255,61</td>\n",
       "      <td>2,2</td>\n",
       "      <td>132,60</td>\n",
       "      <td>883_gi|310698439|ref|NC_001526.2|_Human_papill...</td>\n",
       "      <td>TP63</td>\n",
       "      <td>36</td>\n",
       "      <td>70</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A37700</td>\n",
       "      <td>minus_k64_j_268601,plus_k66_j_196123</td>\n",
       "      <td>255,255</td>\n",
       "      <td>3_189526315</td>\n",
       "      <td>+</td>\n",
       "      <td>5695</td>\n",
       "      <td>+</td>\n",
       "      <td>1970,1970</td>\n",
       "      <td>254,254</td>\n",
       "      <td>1,1</td>\n",
       "      <td>1707,1707</td>\n",
       "      <td>32831_gi|196170262|gb|FJ006723.1|_Human_papill...</td>\n",
       "      <td>TP63</td>\n",
       "      <td>22</td>\n",
       "      <td>54</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  library                             contig(s) contig-breakpoint(s)  \\\n",
       "0  A37700   minus_k60_j_369025,plus_k64_f_28387               257,63   \n",
       "1  A37700  minus_k64_j_268601,plus_k66_j_196123              255,255   \n",
       "\n",
       "  human-breakpoint human-strand  viral-breakpoint viral-strand  \\\n",
       "0      3_189526059            -              3631            -   \n",
       "1      3_189526315            +              5695            +   \n",
       "\n",
       "  contig-length(s) human-length(s) overlap-length(s) viral-length(s)  \\\n",
       "0          389,123          255,61               2,2          132,60   \n",
       "1        1970,1970         254,254               1,1       1707,1707   \n",
       "\n",
       "                                               virus human-gene(s)  \\\n",
       "0  883_gi|310698439|ref|NC_001526.2|_Human_papill...          TP63   \n",
       "1  32831_gi|196170262|gb|FJ006723.1|_Human_papill...          TP63   \n",
       "\n",
       "   flanking-reads  spanning-reads  total-reads  \n",
       "0              36              70          106  \n",
       "1              22              54           76  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = '/projects/trans_scratch/validations/workspace/szong/Cervical/integration/Cervical_transcriptome_integration_summary_20190625.txt'\n",
    "df = pd.read_csv(f, sep='\\t')\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['hpv'] = df['virus'].apply(lambda x: x.split('|')[4].split('_')[4].replace(',', ''))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>library</th>\n",
       "      <th>contig(s)</th>\n",
       "      <th>contig-breakpoint(s)</th>\n",
       "      <th>human-breakpoint</th>\n",
       "      <th>human-strand</th>\n",
       "      <th>viral-breakpoint</th>\n",
       "      <th>viral-strand</th>\n",
       "      <th>contig-length(s)</th>\n",
       "      <th>human-length(s)</th>\n",
       "      <th>overlap-length(s)</th>\n",
       "      <th>viral-length(s)</th>\n",
       "      <th>virus</th>\n",
       "      <th>human-gene(s)</th>\n",
       "      <th>flanking-reads</th>\n",
       "      <th>spanning-reads</th>\n",
       "      <th>total-reads</th>\n",
       "      <th>hpv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A37700</td>\n",
       "      <td>minus_k60_j_369025,plus_k64_f_28387</td>\n",
       "      <td>257,63</td>\n",
       "      <td>3_189526059</td>\n",
       "      <td>-</td>\n",
       "      <td>3631</td>\n",
       "      <td>-</td>\n",
       "      <td>389,123</td>\n",
       "      <td>255,61</td>\n",
       "      <td>2,2</td>\n",
       "      <td>132,60</td>\n",
       "      <td>883_gi|310698439|ref|NC_001526.2|_Human_papill...</td>\n",
       "      <td>TP63</td>\n",
       "      <td>36</td>\n",
       "      <td>70</td>\n",
       "      <td>106</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A37700</td>\n",
       "      <td>minus_k64_j_268601,plus_k66_j_196123</td>\n",
       "      <td>255,255</td>\n",
       "      <td>3_189526315</td>\n",
       "      <td>+</td>\n",
       "      <td>5695</td>\n",
       "      <td>+</td>\n",
       "      <td>1970,1970</td>\n",
       "      <td>254,254</td>\n",
       "      <td>1,1</td>\n",
       "      <td>1707,1707</td>\n",
       "      <td>32831_gi|196170262|gb|FJ006723.1|_Human_papill...</td>\n",
       "      <td>TP63</td>\n",
       "      <td>22</td>\n",
       "      <td>54</td>\n",
       "      <td>76</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  library                             contig(s) contig-breakpoint(s)  \\\n",
       "0  A37700   minus_k60_j_369025,plus_k64_f_28387               257,63   \n",
       "1  A37700  minus_k64_j_268601,plus_k66_j_196123              255,255   \n",
       "\n",
       "  human-breakpoint human-strand  viral-breakpoint viral-strand  \\\n",
       "0      3_189526059            -              3631            -   \n",
       "1      3_189526315            +              5695            +   \n",
       "\n",
       "  contig-length(s) human-length(s) overlap-length(s) viral-length(s)  \\\n",
       "0          389,123          255,61               2,2          132,60   \n",
       "1        1970,1970         254,254               1,1       1707,1707   \n",
       "\n",
       "                                               virus human-gene(s)  \\\n",
       "0  883_gi|310698439|ref|NC_001526.2|_Human_papill...          TP63   \n",
       "1  32831_gi|196170262|gb|FJ006723.1|_Human_papill...          TP63   \n",
       "\n",
       "   flanking-reads  spanning-reads  total-reads hpv  \n",
       "0              36              70          106  16  \n",
       "1              22              54           76  16  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    16\n",
       "1    16\n",
       "Name: hpv, dtype: object"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.hpv.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df.hpv.str.contains('16')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def unique_items(x):\n",
    "    return ';'.join(x.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df.groupby('library')[['human-breakpoint', 'viral-breakpoint', 'hpv']].agg({'viral-breakpoint':['min', 'max'], 'hpv':[','.join, unique_items], 'human-breakpoint':','.join}).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>library</th>\n",
       "      <th colspan=\"2\" halign=\"left\">viral-breakpoint</th>\n",
       "      <th colspan=\"2\" halign=\"left\">hpv</th>\n",
       "      <th>human-breakpoint</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>join</th>\n",
       "      <th>unique_items</th>\n",
       "      <th>join</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A37700</td>\n",
       "      <td>408</td>\n",
       "      <td>5695</td>\n",
       "      <td>16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "      <td>3_189526059,3_189526315,3_189542597,3_18954296...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A37701</td>\n",
       "      <td>879</td>\n",
       "      <td>1723</td>\n",
       "      <td>52,52,52,52,52</td>\n",
       "      <td>52</td>\n",
       "      <td>22_43168251,22_43161526,22_43165135,22_4316152...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  library viral-breakpoint                            hpv               \\\n",
       "                       min   max                     join unique_items   \n",
       "0  A37700              408  5695  16,16,16,16,16,16,16,16           16   \n",
       "1  A37701              879  1723           52,52,52,52,52           52   \n",
       "\n",
       "                                    human-breakpoint  \n",
       "                                                join  \n",
       "0  3_189526059,3_189526315,3_189542597,3_18954296...  \n",
       "1  22_43168251,22_43161526,22_43165135,22_4316152...  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>join</th>\n",
       "      <th>unique_items</th>\n",
       "      <th>join</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A37700</td>\n",
       "      <td>408</td>\n",
       "      <td>5695</td>\n",
       "      <td>16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "      <td>3_189526059,3_189526315,3_189542597,3_18954296...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A37701</td>\n",
       "      <td>879</td>\n",
       "      <td>1723</td>\n",
       "      <td>52,52,52,52,52</td>\n",
       "      <td>52</td>\n",
       "      <td>22_43168251,22_43161526,22_43165135,22_4316152...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           min   max                     join unique_items  \\\n",
       "0  A37700  408  5695  16,16,16,16,16,16,16,16           16   \n",
       "1  A37701  879  1723           52,52,52,52,52           52   \n",
       "\n",
       "                                                join  \n",
       "0  3_189526059,3_189526315,3_189542597,3_18954296...  \n",
       "1  22_43168251,22_43161526,22_43165135,22_4316152...  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df1['unique_items']\n",
    "df2 = df1.droplevel(0, axis=1)\n",
    "df2.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(105, 6)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/integration/Cervical_transcriptome_integration_summary_hpv_integrated_sequence_20190625.txt'\n",
    "df1.to_csv(of, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>min</th>\n",
       "      <th>max</th>\n",
       "      <th>join</th>\n",
       "      <th>unique_items</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A37700</td>\n",
       "      <td>408</td>\n",
       "      <td>5695</td>\n",
       "      <td>16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A37702</td>\n",
       "      <td>880</td>\n",
       "      <td>3829</td>\n",
       "      <td>16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A37707</td>\n",
       "      <td>880</td>\n",
       "      <td>1367</td>\n",
       "      <td>16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A37708</td>\n",
       "      <td>226</td>\n",
       "      <td>3316</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A37717</td>\n",
       "      <td>226</td>\n",
       "      <td>4325</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A37718</td>\n",
       "      <td>834</td>\n",
       "      <td>7394</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>A37723</td>\n",
       "      <td>226</td>\n",
       "      <td>7828</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>A37725</td>\n",
       "      <td>226</td>\n",
       "      <td>3419</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>A37729</td>\n",
       "      <td>226</td>\n",
       "      <td>4998</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>A37736</td>\n",
       "      <td>2077</td>\n",
       "      <td>5178</td>\n",
       "      <td>16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>A50342</td>\n",
       "      <td>880</td>\n",
       "      <td>1533</td>\n",
       "      <td>16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>A50346</td>\n",
       "      <td>407</td>\n",
       "      <td>6357</td>\n",
       "      <td>16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>A52345</td>\n",
       "      <td>227</td>\n",
       "      <td>3113</td>\n",
       "      <td>16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>A52351</td>\n",
       "      <td>228</td>\n",
       "      <td>7286</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>A52359</td>\n",
       "      <td>226</td>\n",
       "      <td>881</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>A52360</td>\n",
       "      <td>226</td>\n",
       "      <td>7148</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>A52364</td>\n",
       "      <td>880</td>\n",
       "      <td>3426</td>\n",
       "      <td>16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>A52365</td>\n",
       "      <td>226</td>\n",
       "      <td>2731</td>\n",
       "      <td>16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>A52366</td>\n",
       "      <td>682</td>\n",
       "      <td>682</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>A52367</td>\n",
       "      <td>880</td>\n",
       "      <td>2687</td>\n",
       "      <td>16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>A54468</td>\n",
       "      <td>3770</td>\n",
       "      <td>3834</td>\n",
       "      <td>16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>A54477</td>\n",
       "      <td>881</td>\n",
       "      <td>881</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>A54479</td>\n",
       "      <td>880</td>\n",
       "      <td>3525</td>\n",
       "      <td>16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>A57112</td>\n",
       "      <td>520</td>\n",
       "      <td>4768</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>A57115</td>\n",
       "      <td>227</td>\n",
       "      <td>3452</td>\n",
       "      <td>16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>A57122</td>\n",
       "      <td>227</td>\n",
       "      <td>3105</td>\n",
       "      <td>16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>A57123</td>\n",
       "      <td>226</td>\n",
       "      <td>4091</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>A68255</td>\n",
       "      <td>3505</td>\n",
       "      <td>3665</td>\n",
       "      <td>16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>A68262</td>\n",
       "      <td>2675</td>\n",
       "      <td>6230</td>\n",
       "      <td>16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>A68267</td>\n",
       "      <td>226</td>\n",
       "      <td>3377</td>\n",
       "      <td>16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>A68268</td>\n",
       "      <td>880</td>\n",
       "      <td>7111</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>A71874</td>\n",
       "      <td>4574</td>\n",
       "      <td>4574</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>A78464</td>\n",
       "      <td>881</td>\n",
       "      <td>881</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>A78477</td>\n",
       "      <td>734</td>\n",
       "      <td>3421</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>A78488</td>\n",
       "      <td>880</td>\n",
       "      <td>3099</td>\n",
       "      <td>16,16,16,16,16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>A78492</td>\n",
       "      <td>226</td>\n",
       "      <td>7100</td>\n",
       "      <td>16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             min   max                                               join  \\\n",
       "0   A37700   408  5695                            16,16,16,16,16,16,16,16   \n",
       "1   A37702   880  3829                                           16,16,16   \n",
       "2   A37707   880  1367                                              16,16   \n",
       "3   A37708   226  3316                      16,16,16,16,16,16,16,16,16,16   \n",
       "4   A37717   226  4325          16,16,16,16,16,16,16,16,16,16,16,16,16,16   \n",
       "5   A37718   834  7394  16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...   \n",
       "6   A37723   226  7828  16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...   \n",
       "7   A37725   226  3419                         16,16,16,16,16,16,16,16,16   \n",
       "8   A37729   226  4998                16,16,16,16,16,16,16,16,16,16,16,16   \n",
       "9   A37736  2077  5178                                              16,16   \n",
       "10  A50342   880  1533                                        16,16,16,16   \n",
       "11  A50346   407  6357                                              16,16   \n",
       "12  A52345   227  3113                                  16,16,16,16,16,16   \n",
       "13  A52351   228  7286       16,16,16,16,16,16,16,16,16,16,16,16,16,16,16   \n",
       "14  A52359   226   881          16,16,16,16,16,16,16,16,16,16,16,16,16,16   \n",
       "15  A52360   226  7148  16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...   \n",
       "16  A52364   880  3426                                  16,16,16,16,16,16   \n",
       "17  A52365   226  2731                                  16,16,16,16,16,16   \n",
       "18  A52366   682   682                                                 16   \n",
       "19  A52367   880  2687                                  16,16,16,16,16,16   \n",
       "20  A54468  3770  3834                                              16,16   \n",
       "21  A54477   881   881                                                 16   \n",
       "22  A54479   880  3525                            16,16,16,16,16,16,16,16   \n",
       "23  A57112   520  4768                16,16,16,16,16,16,16,16,16,16,16,16   \n",
       "24  A57115   227  3452                                  16,16,16,16,16,16   \n",
       "25  A57122   227  3105                                        16,16,16,16   \n",
       "26  A57123   226  4091       16,16,16,16,16,16,16,16,16,16,16,16,16,16,16   \n",
       "27  A68255  3505  3665                                              16,16   \n",
       "28  A68262  2675  6230                                        16,16,16,16   \n",
       "29  A68267   226  3377                                  16,16,16,16,16,16   \n",
       "30  A68268   880  7111  16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...   \n",
       "31  A71874  4574  4574                                                 16   \n",
       "32  A78464   881   881                                                 16   \n",
       "33  A78477   734  3421                      16,16,16,16,16,16,16,16,16,16   \n",
       "34  A78488   880  3099                                     16,16,16,16,16   \n",
       "35  A78492   226  7100  16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,1...   \n",
       "\n",
       "   unique_items  \n",
       "0            16  \n",
       "1            16  \n",
       "2            16  \n",
       "3            16  \n",
       "4            16  \n",
       "5            16  \n",
       "6            16  \n",
       "7            16  \n",
       "8            16  \n",
       "9            16  \n",
       "10           16  \n",
       "11           16  \n",
       "12           16  \n",
       "13           16  \n",
       "14           16  \n",
       "15           16  \n",
       "16           16  \n",
       "17           16  \n",
       "18           16  \n",
       "19           16  \n",
       "20           16  \n",
       "21           16  \n",
       "22           16  \n",
       "23           16  \n",
       "24           16  \n",
       "25           16  \n",
       "26           16  \n",
       "27           16  \n",
       "28           16  \n",
       "29           16  \n",
       "30           16  \n",
       "31           16  \n",
       "32           16  \n",
       "33           16  \n",
       "34           16  \n",
       "35           16  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
