{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "INSTRUCTIONS:\n",
    "    run in this Rstudio: http://gphost03.bcgsc.ca:8787/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://simon-coetzee.github.io/motifBreakR/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4653394/  \n",
    "Transcription factor binding sites (TFBS) are typically short DNA sequence motifs that facilitate binding of a specific transcription factors via protein–DNA interactions.  \n",
    "MotifbreakR will produce a GRanges table with match statistics describing the percent of maximum score for a matrix for both alleles of the SNP, the matrix values for each allele (useful for determining the severity of the disruption), the strand, whether the disruption is strong or weak."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://bioconductor.org/packages/devel/bioc/manuals/motifbreakR/man/motifbreakR.pdf  \n",
    "**seqMatch** the sequence on the 5’ -> 3’ direction of the \"+\" strand that corresponds to DNA at the position that the TF binding motif was found.  \n",
    "\n",
    "**pctRef** The score as determined by the scoring method, when the sequence contains the reference SNP allele, normalized to a scale from 0 - 1. If filterp = FALSE, this is the value that is thresholded.  \n",
    "\n",
    "**pctAlt** The score as determined by the scoring method, when the sequence contains the alternate SNP allele, normalized to a scale from 0 - 1. If filterp = FALSE, this is the value that is thresholded.  \n",
    "\n",
    "**scoreRef** The score as determined by the scoring method, when the sequence contains the reference SNP allele\n",
    "\n",
    "**scoreAlt** The score as determined by the scoring method, when the sequence contains the alternate SNP allele\n",
    "\n",
    "**Refpvalue** p-value for the match for the pctRef score, initially set to NA. see calculatePvalue for more information. this is the significance of the match for PWM, position weight matrix. \n",
    "**Altpvalue** p-value for the match for the pctAlt score, initially set to NA. see calculatePvalue for more information  \n",
    "**alleleRef** The proportional frequency of the reference allele at position motifPos in the motif    \n",
    "\n",
    "**alleleAlt** The proportional frequency of the alternate allele at position motifPos in the motif    \n",
    "\n",
    "effect one of weak, strong, or neutral indicating the strength of the effect."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "use Rstudio on: http://gphost03.bcgsc.ca:8787/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "these are the format of the bed file as input. if you have rs_ids, you can just use a list of rs_ids as input as well.\n",
    "\n",
    "[szong@szong01 motifbreakR]$ head /projects/da_workspace/software/motifbreakR/motifBreakR/inst/extdata/snps.bed\n",
    "\n",
    "chr2\t12581137\t12581138\trs10170896\t0\t+\n",
    "\n",
    "chr2\t12594017\t12594018\tchr2:12594018:G:A\t0\t+\n",
    "\n",
    "chr3\t192388677\t192388678\trs13068005\t0\t+\n",
    "\n",
    "chr4\t122361479\t122361480\trs12644995\t0\t+\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "reference:\n",
    "    \n",
    "    http://simon-coetzee.github.io/motifBreakR/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### input file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start_x</th>\n",
       "      <th>end_x</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295229</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>1295250</td>\n",
       "      <td>1295251</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr  start_x    end_x Reference_Allele Tumor_Seq_Allele2\n",
       "0    5  1295228  1295229                G                 A\n",
       "1    5  1295250  1295251                G                 A"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = '/projects/trans_scratch/validations/workspace/szong/Cervical/hotspots/rainstorm/hotspots_20190426.txt'\n",
    "df = pd.read_csv(f, sep='\\t', usecols=['chr', 'start_x', 'end_x','Reference_Allele', 'Tumor_Seq_Allele2'])\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start_x</th>\n",
       "      <th>end_x</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>1295228</td>\n",
       "      <td>1295229</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>1295250</td>\n",
       "      <td>1295251</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr  start_x    end_x Reference_Allele alt\n",
       "0    5  1295228  1295229                G   A\n",
       "1    5  1295250  1295251                G   A"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s = df.Tumor_Seq_Allele2.str.split(';', expand=True).stack()\n",
    "idx = s.index.get_level_values(0)\n",
    "dfn = df.loc[idx]\n",
    "dfn['alt'] = list(s)\n",
    "dfn.drop('Tumor_Seq_Allele2', axis=1, inplace=True)\n",
    "\n",
    "dfn.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### input file need 'chr' in front of chromosome number, for some reason, it looks it require the second position to be the mutation. may be 0 based numbering?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>start_x</th>\n",
       "      <th>rsid</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295227</td>\n",
       "      <td>1295228</td>\n",
       "      <td>chr5:1295228:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295249</td>\n",
       "      <td>1295250</td>\n",
       "      <td>chr5:1295250:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>92890259</td>\n",
       "      <td>92890260</td>\n",
       "      <td>chr6:92890260:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6</td>\n",
       "      <td>142706205</td>\n",
       "      <td>142706206</td>\n",
       "      <td>chr6:142706206:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6</td>\n",
       "      <td>142706208</td>\n",
       "      <td>142706209</td>\n",
       "      <td>chr6:142706209:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6</td>\n",
       "      <td>142706208</td>\n",
       "      <td>142706209</td>\n",
       "      <td>chr6:142706209:C:G</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>83993736</td>\n",
       "      <td>83993737</td>\n",
       "      <td>chr8:83993737:T:A</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr11</td>\n",
       "      <td>39877968</td>\n",
       "      <td>39877969</td>\n",
       "      <td>chr11:39877969:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr      start    start_x                rsid  4  5\n",
       "0   chr5    1295227    1295228    chr5:1295228:G:A  0  +\n",
       "1   chr5    1295249    1295250    chr5:1295250:G:A  0  +\n",
       "2   chr6   92890259   92890260   chr6:92890260:A:T  0  +\n",
       "3   chr6  142706205  142706206  chr6:142706206:G:A  0  +\n",
       "4   chr6  142706208  142706209  chr6:142706209:C:T  0  +\n",
       "4   chr6  142706208  142706209  chr6:142706209:C:G  0  +\n",
       "5   chr8   83993736   83993737   chr8:83993737:T:A  0  +\n",
       "6  chr11   39877968   39877969  chr11:39877969:A:G  0  +"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfn['rsid'] = dfn.apply(lambda x: ''.join(['chr', ':'.join(np.array([str(i) for i in x])[[0,1,3, 4]])]), axis=1)\n",
    "dfn = dfn.iloc[:, [0,1,5]]\n",
    "dfn[4] = 0\n",
    "dfn[5] = \"+\"\n",
    "dfn['start'] = dfn.start_x -1\n",
    "dfn = dfn.iloc[:,[0,5,1,2,3,4]].drop_duplicates()\n",
    "dfn['chr'] = dfn['chr'].apply(lambda x: ''.join(['chr', str(x)]))\n",
    "dfn\n",
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/hotspots/rainstorm/mt_input.txt'\n",
    "dfn.to_csv(of, sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### this is the R script to run motifBreakR\n",
    "file.path(R.home(\"bin\"), \"R\")  \n",
    ".libPaths( c('/projects/da_workspace/software/motifbreakR_libs', \"/gsc/software/linux-x86_64-centos7/R-3.5.1/lib64/R/library\", .libPaths()) )  \n",
    ".libPaths()  \n",
    "library(motifbreakR)  \n",
    "library(BSgenome)  \n",
    "library(SNPlocs.Hsapiens.dbSNP142.GRCh37) # dbSNP137 in hg19  \n",
    "library(BSgenome.Hsapiens.UCSC.hg19)     # hg19 genome  \n",
    "library(MotifDb)  \n",
    "\n",
    "snps.bed.file <- '/projects/trans_scratch/validations/workspace/szong/Cervical/hotspots/rainstorm/mt_input.txt'\n",
    "read.table(snps.bed.file, header = FALSE)\n",
    "\n",
    "#import the BED file\n",
    "snps.mb.frombed <- snps.from.file(file = snps.bed.file,\n",
    "                                  #dbSNP = SNPlocs.Hsapiens.dbSNP142.GRCh37,\n",
    "                                  search.genome = BSgenome.Hsapiens.UCSC.hg19,\n",
    "                                  format = \"bed\")\n",
    "snps.mb.frombed\n",
    "\n",
    "data(motifbreakR_motif)\n",
    "motifbreakR_motif\n",
    "\n",
    "\n",
    "results <- motifbreakR(snpList = snps.mb.frombed, filterp = TRUE,  \n",
    "                       pwmList = motifbreakR_motif,  \n",
    "                       threshold = 1e-4,  \n",
    "                       method = \"ic\",  \n",
    "                       bkg = c(A=0.25, C=0.25, G=0.25, T=0.25),  \n",
    "                       BPPARAM = BiocParallel::bpparam())\n",
    "\n",
    "results\n",
    "\n",
    "f <- '/projects/da_workspace/software/motifbreakR/results.csv'  \n",
    "write.csv(results, file = f)\n",
    "\n",
    "### calculate p values\n",
    "sub_res <- results[names(results) %in% 'chr5:1295228:G:A']  \n",
    "pvalue <- calculatePvalue(sub_res)  \n",
    "pvalue\n",
    "\n",
    "plotMB(results = results, rsid = \"6:142706209:142706210:C:G\", effect = \"strong\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### filter output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>snpPos</th>\n",
       "      <th>motifPos</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>...</th>\n",
       "      <th>seqMatch</th>\n",
       "      <th>pctRef</th>\n",
       "      <th>pctAlt</th>\n",
       "      <th>scoreRef</th>\n",
       "      <th>scoreAlt</th>\n",
       "      <th>Refpvalue</th>\n",
       "      <th>Altpvalue</th>\n",
       "      <th>alleleRef</th>\n",
       "      <th>alleleAlt</th>\n",
       "      <th>effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295222</td>\n",
       "      <td>1295231</td>\n",
       "      <td>10</td>\n",
       "      <td>+</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1295228</td>\n",
       "      <td>7</td>\n",
       "      <td>BCLAF1</td>\n",
       "      <td>...</td>\n",
       "      <td>cccggaGggg</td>\n",
       "      <td>0.771581</td>\n",
       "      <td>0.910434</td>\n",
       "      <td>8.099010</td>\n",
       "      <td>9.556510</td>\n",
       "      <td>0.001601</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.971154</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295220</td>\n",
       "      <td>1295235</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1295228</td>\n",
       "      <td>9</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>...</td>\n",
       "      <td>ggcccggaGggggctg</td>\n",
       "      <td>0.942173</td>\n",
       "      <td>0.855357</td>\n",
       "      <td>5.523480</td>\n",
       "      <td>5.014519</td>\n",
       "      <td>0.000084</td>\n",
       "      <td>0.001699</td>\n",
       "      <td>0.612805</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>weak</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295219</td>\n",
       "      <td>1295237</td>\n",
       "      <td>19</td>\n",
       "      <td>+</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1295228</td>\n",
       "      <td>10</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>...</td>\n",
       "      <td>gggcccggaGggggctggg</td>\n",
       "      <td>0.898012</td>\n",
       "      <td>0.752305</td>\n",
       "      <td>10.995074</td>\n",
       "      <td>9.211073</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.001171</td>\n",
       "      <td>0.979328</td>\n",
       "      <td>0.002839</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295219</td>\n",
       "      <td>1295237</td>\n",
       "      <td>19</td>\n",
       "      <td>+</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1295228</td>\n",
       "      <td>10</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>...</td>\n",
       "      <td>gggcccggaGggggctggg</td>\n",
       "      <td>0.857465</td>\n",
       "      <td>0.724539</td>\n",
       "      <td>10.114097</td>\n",
       "      <td>8.546181</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>0.001859</td>\n",
       "      <td>0.991218</td>\n",
       "      <td>0.005488</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295222</td>\n",
       "      <td>1295235</td>\n",
       "      <td>14</td>\n",
       "      <td>+</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1295228</td>\n",
       "      <td>7</td>\n",
       "      <td>CTCF</td>\n",
       "      <td>...</td>\n",
       "      <td>cccggaGggggctg</td>\n",
       "      <td>0.985911</td>\n",
       "      <td>0.702726</td>\n",
       "      <td>5.706734</td>\n",
       "      <td>4.067583</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.032342</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>strong</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqnames    start      end  width strand REF ALT   snpPos  motifPos  \\\n",
       "0     chr5  1295222  1295231     10      +   G   A  1295228         7   \n",
       "1     chr5  1295220  1295235     16      +   G   A  1295228         9   \n",
       "2     chr5  1295219  1295237     19      +   G   A  1295228        10   \n",
       "3     chr5  1295219  1295237     19      +   G   A  1295228        10   \n",
       "4     chr5  1295222  1295235     14      +   G   A  1295228         7   \n",
       "\n",
       "  geneSymbol  ...                                 seqMatch    pctRef  \\\n",
       "0     BCLAF1  ...     cccggaGggg                            0.771581   \n",
       "1       CTCF  ...   ggcccggaGggggctg                        0.942173   \n",
       "2       CTCF  ...  gggcccggaGggggctggg                      0.898012   \n",
       "3       CTCF  ...  gggcccggaGggggctggg                      0.857465   \n",
       "4       CTCF  ...     cccggaGggggctg                        0.985911   \n",
       "\n",
       "     pctAlt   scoreRef  scoreAlt  Refpvalue  Altpvalue  alleleRef  alleleAlt  \\\n",
       "0  0.910434   8.099010  9.556510   0.001601   0.000060   0.000000   0.971154   \n",
       "1  0.855357   5.523480  5.014519   0.000084   0.001699   0.612805   0.000000   \n",
       "2  0.752305  10.995074  9.211073   0.000010   0.001171   0.979328   0.002839   \n",
       "3  0.724539  10.114097  8.546181   0.000039   0.001859   0.991218   0.005488   \n",
       "4  0.702726   5.706734  4.067583   0.000003   0.032342   1.000000   0.000000   \n",
       "\n",
       "   effect  \n",
       "0  strong  \n",
       "1    weak  \n",
       "2  strong  \n",
       "3  strong  \n",
       "4  strong  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f = '/projects/da_workspace/software/motifbreakR/results_pvalues.csv'\n",
    "df = pd.read_csv(f)\n",
    "df.drop('Unnamed: 0', axis=1, inplace=True)\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['pvalue_diff'] = (df.Refpvalue - df.Altpvalue).abs()\n",
    "df['score_diff'] = (df.scoreRef - df.scoreAlt).abs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(232, 26)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df.sort_values(['pvalue_diff', 'score_diff'], ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/da_workspace/software/motifbreakR/motifbreakr_results_pvalues_20190528.csv'\n",
    "df1.to_csv(of, sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>snpPos</th>\n",
       "      <th>motifPos</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>...</th>\n",
       "      <th>pctAlt</th>\n",
       "      <th>scoreRef</th>\n",
       "      <th>scoreAlt</th>\n",
       "      <th>Refpvalue</th>\n",
       "      <th>Altpvalue</th>\n",
       "      <th>alleleRef</th>\n",
       "      <th>alleleAlt</th>\n",
       "      <th>effect</th>\n",
       "      <th>pvalue_diff</th>\n",
       "      <th>score_diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295225</td>\n",
       "      <td>1295237</td>\n",
       "      <td>13</td>\n",
       "      <td>+</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1295228</td>\n",
       "      <td>4</td>\n",
       "      <td>POU2F2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.643708</td>\n",
       "      <td>4.886466</td>\n",
       "      <td>3.247314</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.092089</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>strong</td>\n",
       "      <td>0.092009</td>\n",
       "      <td>1.639152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr5</td>\n",
       "      <td>1295223</td>\n",
       "      <td>1295235</td>\n",
       "      <td>13</td>\n",
       "      <td>+</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1295228</td>\n",
       "      <td>6</td>\n",
       "      <td>EBF1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.682239</td>\n",
       "      <td>5.717843</td>\n",
       "      <td>4.078691</td>\n",
       "      <td>0.000083</td>\n",
       "      <td>0.051075</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>strong</td>\n",
       "      <td>0.050993</td>\n",
       "      <td>1.639152</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   seqnames    start      end  width strand REF ALT   snpPos  motifPos  \\\n",
       "46     chr5  1295225  1295237     13      +   G   A  1295228         4   \n",
       "7      chr5  1295223  1295235     13      +   G   A  1295228         6   \n",
       "\n",
       "   geneSymbol  ...    pctAlt  scoreRef  scoreAlt Refpvalue  Altpvalue  \\\n",
       "46     POU2F2  ...  0.643708  4.886466  3.247314  0.000079   0.092089   \n",
       "7        EBF1  ...  0.682239  5.717843  4.078691  0.000083   0.051075   \n",
       "\n",
       "    alleleRef  alleleAlt  effect  pvalue_diff  score_diff  \n",
       "46        1.0        0.0  strong     0.092009    1.639152  \n",
       "7         1.0        0.0  strong     0.050993    1.639152  \n",
       "\n",
       "[2 rows x 25 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.snpPos == 1295228].sort_values(['pvalue_diff', 'score_diff'], ascending=False).head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### transcription databased used by motifBreakR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "HOMMER, INCLUDED JASPER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "*Warning?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Object `%cpaste` not found.\n"
     ]
    }
   ],
   "source": [
    "%cpaste?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/json": {
       "cell": {
        "!": "OSMagics",
        "HTML": "Other",
        "SVG": "Other",
        "bash": "Other",
        "capture": "ExecutionMagics",
        "debug": "ExecutionMagics",
        "file": "Other",
        "html": "DisplayMagics",
        "javascript": "DisplayMagics",
        "js": "DisplayMagics",
        "latex": "DisplayMagics",
        "markdown": "DisplayMagics",
        "perl": "Other",
        "prun": "ExecutionMagics",
        "pypy": "Other",
        "python": "Other",
        "python2": "Other",
        "python3": "Other",
        "ruby": "Other",
        "script": "ScriptMagics",
        "sh": "Other",
        "svg": "DisplayMagics",
        "sx": "OSMagics",
        "system": "OSMagics",
        "time": "ExecutionMagics",
        "timeit": "ExecutionMagics",
        "writefile": "OSMagics"
       },
       "line": {
        "alias": "OSMagics",
        "alias_magic": "BasicMagics",
        "autocall": "AutoMagics",
        "automagic": "AutoMagics",
        "autosave": "KernelMagics",
        "bookmark": "OSMagics",
        "cat": "Other",
        "cd": "OSMagics",
        "clear": "KernelMagics",
        "colors": "BasicMagics",
        "config": "ConfigMagics",
        "connect_info": "KernelMagics",
        "cp": "Other",
        "debug": "ExecutionMagics",
        "dhist": "OSMagics",
        "dirs": "OSMagics",
        "doctest_mode": "BasicMagics",
        "ed": "Other",
        "edit": "KernelMagics",
        "env": "OSMagics",
        "gui": "BasicMagics",
        "hist": "Other",
        "history": "HistoryMagics",
        "killbgscripts": "ScriptMagics",
        "ldir": "Other",
        "less": "KernelMagics",
        "lf": "Other",
        "lk": "Other",
        "ll": "Other",
        "load": "CodeMagics",
        "load_ext": "ExtensionMagics",
        "loadpy": "CodeMagics",
        "logoff": "LoggingMagics",
        "logon": "LoggingMagics",
        "logstart": "LoggingMagics",
        "logstate": "LoggingMagics",
        "logstop": "LoggingMagics",
        "ls": "Other",
        "lsmagic": "BasicMagics",
        "lx": "Other",
        "macro": "ExecutionMagics",
        "magic": "BasicMagics",
        "man": "KernelMagics",
        "matplotlib": "PylabMagics",
        "mkdir": "Other",
        "more": "KernelMagics",
        "mv": "Other",
        "notebook": "BasicMagics",
        "page": "BasicMagics",
        "pastebin": "CodeMagics",
        "pdb": "ExecutionMagics",
        "pdef": "NamespaceMagics",
        "pdoc": "NamespaceMagics",
        "pfile": "NamespaceMagics",
        "pinfo": "NamespaceMagics",
        "pinfo2": "NamespaceMagics",
        "pip": "BasicMagics",
        "popd": "OSMagics",
        "pprint": "BasicMagics",
        "precision": "BasicMagics",
        "profile": "BasicMagics",
        "prun": "ExecutionMagics",
        "psearch": "NamespaceMagics",
        "psource": "NamespaceMagics",
        "pushd": "OSMagics",
        "pwd": "OSMagics",
        "pycat": "OSMagics",
        "pylab": "PylabMagics",
        "qtconsole": "KernelMagics",
        "quickref": "BasicMagics",
        "recall": "HistoryMagics",
        "rehashx": "OSMagics",
        "reload_ext": "ExtensionMagics",
        "rep": "Other",
        "rerun": "HistoryMagics",
        "reset": "NamespaceMagics",
        "reset_selective": "NamespaceMagics",
        "rm": "Other",
        "rmdir": "Other",
        "run": "ExecutionMagics",
        "save": "CodeMagics",
        "sc": "OSMagics",
        "set_env": "OSMagics",
        "store": "StoreMagics",
        "sx": "OSMagics",
        "system": "OSMagics",
        "tb": "ExecutionMagics",
        "time": "ExecutionMagics",
        "timeit": "ExecutionMagics",
        "unalias": "OSMagics",
        "unload_ext": "ExtensionMagics",
        "who": "NamespaceMagics",
        "who_ls": "NamespaceMagics",
        "whos": "NamespaceMagics",
        "xdel": "NamespaceMagics",
        "xmode": "BasicMagics"
       }
      },
      "text/plain": [
       "Available line magics:\n",
       "%alias  %alias_magic  %autocall  %automagic  %autosave  %bookmark  %cat  %cd  %clear  %colors  %config  %connect_info  %cp  %debug  %dhist  %dirs  %doctest_mode  %ed  %edit  %env  %gui  %hist  %history  %killbgscripts  %ldir  %less  %lf  %lk  %ll  %load  %load_ext  %loadpy  %logoff  %logon  %logstart  %logstate  %logstop  %ls  %lsmagic  %lx  %macro  %magic  %man  %matplotlib  %mkdir  %more  %mv  %notebook  %page  %pastebin  %pdb  %pdef  %pdoc  %pfile  %pinfo  %pinfo2  %popd  %pprint  %precision  %profile  %prun  %psearch  %psource  %pushd  %pwd  %pycat  %pylab  %qtconsole  %quickref  %recall  %rehashx  %reload_ext  %rep  %rerun  %reset  %reset_selective  %rm  %rmdir  %run  %save  %sc  %set_env  %store  %sx  %system  %tb  %time  %timeit  %unalias  %unload_ext  %who  %who_ls  %whos  %xdel  %xmode\n",
       "\n",
       "Available cell magics:\n",
       "%%!  %%HTML  %%SVG  %%bash  %%capture  %%debug  %%file  %%html  %%javascript  %%js  %%latex  %%markdown  %%perl  %%prun  %%pypy  %%python  %%python2  %%python3  %%ruby  %%script  %%sh  %%svg  %%sx  %%system  %%time  %%timeit  %%writefile\n",
       "\n",
       "Automagic is ON, % prefix IS NOT needed for line magics."
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%lsmagic"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
