{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### this version adds more annotation to the summary file including amino acid change in order to make the lollipop plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import matplotlib\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import scipy.stats as ss\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import style, colors\n",
    "from matplotlib import gridspec\n",
    "import random\n",
    "import string\n",
    "import math\n",
    "\n",
    "# matplotlib.style.use('ggplot')\n",
    "%matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (12, 6)\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### what does multiple means?\n",
    "* multi_hit means a gene hit by two or more mutations. \n",
    "\n",
    "* use the new strelka results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "use all calls from strelka: this is prefered since this takes into consideration of all soamtic calls:snvs and indels, high/moderate/low/modifier impacts.\n",
    "\n",
    "in most papers, they only report mutation in coding regions including splicing site, excluding non-coding regions, i will get these numbers as well. synonymous/non-synonymous mutation are only in coding regions."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "a good post about match\n",
    "\n",
    "https://stackoverflow.com/questions/24958358/use-regular-expression-in-python-to-find-two-strings-in-line"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "a table details the equivalence of ontology an classic effect description\n",
    "\n",
    "http://snpeff.sourceforge.net/SnpEff_manual.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. parsing strelka vcf files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_strelka_vcf(vcf):\n",
    "    df = pd.read_csv(vcf, comment='#', sep='\\t', header=None, low_memory=False)\n",
    "    patient = vcf.split('/')[4]\n",
    "    df = df[[0,1,3,4,7]]\n",
    "    df.columns = ['chr', 'pos', 'ref', 'alt', 'effects']\n",
    "    df = df[(df['effects'].str.contains(\"HIGH\"))|(df['effects'].str.contains(\"MODERATE\"))|(df['effects'].str.contains(\"LOW\"))]\n",
    "#     print(df.head(2))\n",
    "    if not df.empty:\n",
    "        df['annotations'] = df['effects'].apply(lambda x: get_annotations(x))      \n",
    "        df['patient'] = patient\n",
    "\n",
    "    df = df.drop('effects', axis=1)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "def wide_to_long(df):\n",
    "    s = df.annotations.str.split('$', expand=True).stack()\n",
    "    idx = s.index.get_level_values(0)\n",
    "    print('length of s and idx', len(s), len(idx))\n",
    "    dfn = df.loc[idx]\n",
    "    dfn['effect'] = list(s)\n",
    "    dfn.drop('annotations', inplace=True, axis=1)\n",
    "    dfn['impact'], dfn['impact_type'], dfn['type'], dfn['AA_change'], dfn['gene'], dfn['transcript'] = dfn['effect'].apply(lambda x: x.split('@')).str\n",
    "    dfn.drop('effect', axis=1, inplace=True)\n",
    "    return dfn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep it easy for now pick HIGH, MODERATE and then LOW and the first transcript. could pull out all transcripts if needed\n",
    "def get_annotations(line):\n",
    "    effs = line.split('EFF=')[1].split(',')\n",
    "    #     extract impact, impact_type, amino acid change, and gene\n",
    "    effs = ['@'.join(list(np.array(re.split('\\(|\\|',ef))[[0,1,2,4,6,9]])) for ef in effs if ('HIGH' in  ef) or ('MODERATE' in ef) or ('LOW' in ef)]\n",
    "    effs = list(set(effs))\n",
    "    high = [ef for ef in effs if 'HIGH' in ef]\n",
    "    moderate = [ef for ef in effs if 'MODERATE' in ef]\n",
    "    low = [ef for ef in effs if 'LOW' in ef]\n",
    "    if high:\n",
    "        anno = '$'.join(high)\n",
    "    elif moderate:\n",
    "        anno = '$'.join(moderate)\n",
    "    elif low:\n",
    "        anno = '$'.join(low)\n",
    "    else:\n",
    "        print('ERROR!')\n",
    "#     make sure the genes have the same name\n",
    "#     genes = [ef.split('@')[2] for ef in effs]\n",
    "#     print(anno)\n",
    "    # anno= list of annotations\n",
    "    return anno\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length of s and idx 745 745\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>annotations</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>908292</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@D352N@...</td>\n",
       "      <td>HTMCP-03-06-02020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>935319</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@S12L@H...</td>\n",
       "      <td>HTMCP-03-06-02020</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr     pos ref alt                                        annotations  \\\n",
       "6   1  908292   G   A  NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@D352N@...   \n",
       "7   1  935319   G   A  NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@S12L@H...   \n",
       "\n",
       "             patient  \n",
       "6  HTMCP-03-06-02020  \n",
       "7  HTMCP-03-06-02020  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>type</th>\n",
       "      <th>AA_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>908292</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>HTMCP-03-06-02020</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>MISSENSE</td>\n",
       "      <td>D352N</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>ENST00000379410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>908292</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>HTMCP-03-06-02020</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>MISSENSE</td>\n",
       "      <td>D404N</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>ENST00000379409</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr     pos ref alt            patient                 impact impact_type  \\\n",
       "6   1  908292   G   A  HTMCP-03-06-02020  NON_SYNONYMOUS_CODING    MODERATE   \n",
       "6   1  908292   G   A  HTMCP-03-06-02020  NON_SYNONYMOUS_CODING    MODERATE   \n",
       "\n",
       "       type AA_change     gene       transcript  \n",
       "6  MISSENSE     D352N  PLEKHN1  ENST00000379410  \n",
       "6  MISSENSE     D404N  PLEKHN1  ENST00000379409  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # example run\n",
    "ft = '/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-03-06-02020/hg19a/GSH/A37247_A56468/strelka/23223/bwa/results/passed.somatic.snvs.eff.dbSNP_v137.cosmic_v64.annotations.vcf'\n",
    "ft = '/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-03-06-02020/hg19a/GSH/A37247_A56468/strelka/59869/bwa/results/passed.somatic.snvs.eff.dbSNP_v149.cosmic_v82.clinvar_20170801.annotations.classic.vcf'\n",
    "# assert count_coding_mutations(ft) == 266\n",
    "df = parse_strelka_vcf(ft)\n",
    "dfn = wide_to_long(df)\n",
    "\n",
    "df.head(2)\n",
    "dfn.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Scope: genes and cohort size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 118 patients\n",
    "f = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients.txt'\n",
    "with open(f) as file:\n",
    "    patients = [line.strip() for line in file]\n",
    "assert len(patients) == 118"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 12 SMGS\n",
    "f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients/smgs_reviewed.txt'\n",
    "f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/12_sig_genes.txt'\n",
    "with open (f2) as file:\n",
    "    genes = [line.strip() for line in file]\n",
    "assert len(genes) == 13"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. Strelka vcf files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>DNA_lib</th>\n",
       "      <th>DNA_single_vcf</th>\n",
       "      <th>strelka_indel_vcf</th>\n",
       "      <th>strelka_snv_vcf</th>\n",
       "      <th>source</th>\n",
       "      <th>status</th>\n",
       "      <th>RNA_lib</th>\n",
       "      <th>ribodepleted_lib</th>\n",
       "      <th>DNA_bam</th>\n",
       "      <th>...</th>\n",
       "      <th>other_vcf</th>\n",
       "      <th>DNA_tc</th>\n",
       "      <th>RNA_tc</th>\n",
       "      <th>cnv</th>\n",
       "      <th>bbt_transcriptome</th>\n",
       "      <th>bbt_genome</th>\n",
       "      <th>bbt_transcriptome_other_bacterial</th>\n",
       "      <th>bbt_genome_other_bacterial</th>\n",
       "      <th>bbt_transcriptome_other_viral</th>\n",
       "      <th>bbt_genome_other_viral</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <td>Positive</td>\n",
       "      <td>A37234</td>\n",
       "      <td>/projects/analysis31/A37234/vcall21989_merge13...</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>HTMCP_124</td>\n",
       "      <td>Malignant</td>\n",
       "      <td>A37700</td>\n",
       "      <td>A56295</td>\n",
       "      <td>/projects/analysis/analysis22/A37234/merge_bwa...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <td>Negative</td>\n",
       "      <td>A37235</td>\n",
       "      <td>/projects/analysis31/A37235/vcall21991_merge13...</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>HTMCP_125</td>\n",
       "      <td>Malignant</td>\n",
       "      <td>A37701</td>\n",
       "      <td>A56296</td>\n",
       "      <td>/projects/analysis/analysis22/A37235/merge_bwa...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  HIV_status DNA_lib  \\\n",
       "patient                                \n",
       "HTMCP-03-06-02001   Positive  A37234   \n",
       "HTMCP-03-06-02002   Negative  A37235   \n",
       "\n",
       "                                                      DNA_single_vcf  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/analysis31/A37234/vcall21989_merge13...   \n",
       "HTMCP-03-06-02002  /projects/analysis31/A37235/vcall21991_merge13...   \n",
       "\n",
       "                                                   strelka_indel_vcf  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "HTMCP-03-06-02002  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "\n",
       "                                                     strelka_snv_vcf  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "HTMCP-03-06-02002  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "\n",
       "                      source     status RNA_lib ribodepleted_lib  \\\n",
       "patient                                                            \n",
       "HTMCP-03-06-02001  HTMCP_124  Malignant  A37700           A56295   \n",
       "HTMCP-03-06-02002  HTMCP_125  Malignant  A37701           A56296   \n",
       "\n",
       "                                                             DNA_bam  ...  \\\n",
       "patient                                                               ...   \n",
       "HTMCP-03-06-02001  /projects/analysis/analysis22/A37234/merge_bwa...  ...   \n",
       "HTMCP-03-06-02002  /projects/analysis/analysis22/A37235/merge_bwa...  ...   \n",
       "\n",
       "                  other_vcf DNA_tc  RNA_tc  \\\n",
       "patient                                      \n",
       "HTMCP-03-06-02001       NaN   55.0    55.0   \n",
       "HTMCP-03-06-02002       NaN   70.0    70.0   \n",
       "\n",
       "                                                                 cnv  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "HTMCP-03-06-02002  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "\n",
       "                                                   bbt_transcriptome  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                                          bbt_genome  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                   bbt_transcriptome_other_bacterial  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                          bbt_genome_other_bacterial  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                       bbt_transcriptome_other_viral  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                              bbt_genome_other_viral  \n",
       "patient                                                               \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...  \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...  \n",
       "\n",
       "[2 rows x 28 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# f1 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/124_patients_bam_vcf.txt'\n",
    "f1 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/124_patients_bam_vcf_20190115.csv'\n",
    "df = pd.read_csv(f1, index_col='patient')\n",
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>strelka_snv_vcf</th>\n",
       "      <th>strelka_indel_vcf</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     strelka_snv_vcf  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "HTMCP-03-06-02002  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "\n",
       "                                                   strelka_indel_vcf  \n",
       "patient                                                               \n",
       "HTMCP-03-06-02001  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...  \n",
       "HTMCP-03-06-02002  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...  "
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ndf = df[['strelka_snv_vcf', 'strelka_indel_vcf']].dropna()\n",
    "ndf.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "ndf = ndf.reindex(patients)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(118, 2)"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ndf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-03-06-02001/hg19a/GSH/A37234_A37159/strelka/59844/bwa/results/passed.somatic.indels.eff.dbSNP_v149.cosmic_v82.clinvar_20170801.annotations.classic.vcf\n",
      "/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-03-06-02002/hg19a/GSH/A37235_A37160/strelka/59848/bwa/results/passed.somatic.indels.eff.dbSNP_v149.cosmic_v82.clinvar_20170801.annotations.classic.vcf\n"
     ]
    }
   ],
   "source": [
    "for i in ndf['strelka_indel_vcf'].head(2):\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. concatenate snvs and indels "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfmg = pd.DataFrame()\n",
    "for ix, row in ndf.iterrows():\n",
    "    patient = ix\n",
    "    snv_vcf = row['strelka_snv_vcf']\n",
    "    indel_vcf = row['strelka_indel_vcf']\n",
    "    snv_df = parse_strelka_vcf(snv_vcf)\n",
    "    indel_df = parse_strelka_vcf(indel_vcf)\n",
    "    mdf = pd.concat([snv_df, indel_df], sort=False)\n",
    "    mdf = mdf.drop_duplicates(keep='first')\n",
    "\n",
    "    if dfmg.empty:\n",
    "        dfmg = mdf\n",
    "    else:\n",
    "        dfmg = pd.concat([dfmg, mdf], sort=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>annotations</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1</td>\n",
       "      <td>2706109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SYNONYMOUS_CODING@LOW@SILENT@R16@TTC34@ENST000...</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>1</td>\n",
       "      <td>4771973</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@R15C@A...</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>1</td>\n",
       "      <td>6146051</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@G104A@...</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>1</td>\n",
       "      <td>6579534</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@R13Q@P...</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>1</td>\n",
       "      <td>12170207</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@L208V@...</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr       pos ref alt                                        annotations  \\\n",
       "39    1   2706109   C   G  SYNONYMOUS_CODING@LOW@SILENT@R16@TTC34@ENST000...   \n",
       "58    1   4771973   C   T  NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@R15C@A...   \n",
       "78    1   6146051   G   C  NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@G104A@...   \n",
       "83    1   6579534   C   T  NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@R13Q@P...   \n",
       "145   1  12170207   C   G  NON_SYNONYMOUS_CODING@MODERATE@MISSENSE@L208V@...   \n",
       "\n",
       "               patient  \n",
       "39   HTMCP-03-06-02001  \n",
       "58   HTMCP-03-06-02001  \n",
       "78   HTMCP-03-06-02001  \n",
       "83   HTMCP-03-06-02001  \n",
       "145  HTMCP-03-06-02001  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(36739, 6)"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfmg.head()\n",
    "dfmg.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfmgn = dfmg.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length of s and idx 102013 102013\n"
     ]
    }
   ],
   "source": [
    "dfn = wide_to_long(dfmgn)\n",
    "# wide_to_long(dfmg.iloc[46:416,:].reset_index(drop=True))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(102013, 10)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>AA_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2706109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>LOW</td>\n",
       "      <td>R16</td>\n",
       "      <td>TTC34</td>\n",
       "      <td>ENST00000401095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2706109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>LOW</td>\n",
       "      <td>R16</td>\n",
       "      <td>TTC34</td>\n",
       "      <td>ENST00000574621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4771973</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>R15C</td>\n",
       "      <td>AJAP1</td>\n",
       "      <td>ENST00000378191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>4771973</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>R15C</td>\n",
       "      <td>AJAP1</td>\n",
       "      <td>ENST00000378190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>6146051</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>G104A</td>\n",
       "      <td>KCNAB2</td>\n",
       "      <td>ENST00000462676</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr      pos ref alt            patient                 impact impact_type  \\\n",
       "0   1  2706109   C   G  HTMCP-03-06-02001      SYNONYMOUS_CODING         LOW   \n",
       "0   1  2706109   C   G  HTMCP-03-06-02001      SYNONYMOUS_CODING         LOW   \n",
       "1   1  4771973   C   T  HTMCP-03-06-02001  NON_SYNONYMOUS_CODING    MODERATE   \n",
       "1   1  4771973   C   T  HTMCP-03-06-02001  NON_SYNONYMOUS_CODING    MODERATE   \n",
       "2   1  6146051   G   C  HTMCP-03-06-02001  NON_SYNONYMOUS_CODING    MODERATE   \n",
       "\n",
       "  AA_change    gene       transcript  \n",
       "0       R16   TTC34  ENST00000401095  \n",
       "0       R16   TTC34  ENST00000574621  \n",
       "1      R15C   AJAP1  ENST00000378191  \n",
       "1      R15C   AJAP1  ENST00000378190  \n",
       "2     G104A  KCNAB2  ENST00000462676  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfn.shape\n",
    "dfn.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_20190527.txt'\n",
    "# dfn.to_csv(of, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make supplementary table, gene by patient matrix to indicate if a gene is mutated in each patient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>type</th>\n",
       "      <th>AA_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2706109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SILENT</td>\n",
       "      <td>R16</td>\n",
       "      <td>TTC34</td>\n",
       "      <td>ENST00000401095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2706109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SILENT</td>\n",
       "      <td>R16</td>\n",
       "      <td>TTC34</td>\n",
       "      <td>ENST00000574621</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr      pos ref alt            patient             impact impact_type  \\\n",
       "0   1  2706109   C   G  HTMCP-03-06-02001  SYNONYMOUS_CODING         LOW   \n",
       "1   1  2706109   C   G  HTMCP-03-06-02001  SYNONYMOUS_CODING         LOW   \n",
       "\n",
       "     type AA_change   gene       transcript  \n",
       "0  SILENT       R16  TTC34  ENST00000401095  \n",
       "1  SILENT       R16  TTC34  ENST00000574621  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(102013, 11)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_20190527.txt'\n",
    "df = pd.read_csv(of, sep='\\t')\n",
    "df.head(2)\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = pd.pivot_table(df[['gene', 'patient', 'type']], index='gene', columns='patient', values='type', aggfunc=lambda x: 'mutated' if not x.empty else 'WT')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02417</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7SK</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mutated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02006 HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02013 HTMCP-03-06-02020 HTMCP-03-06-02036  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02037  ... HTMCP-03-06-02417 HTMCP-03-06-02424  \\\n",
       "gene                       ...                                       \n",
       "7SK                   NaN  ...               NaN               NaN   \n",
       "A1BG                  NaN  ...               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "gene                                                            \n",
       "7SK               mutated               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02442 HTMCP-03-06-02447  \n",
       "gene                                         \n",
       "7SK                   NaN               NaN  \n",
       "A1BG                  NaN               NaN  \n",
       "\n",
       "[2 rows x 118 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>number_patients_mutated</th>\n",
       "      <th>percent_patients_mutated</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7SK</th>\n",
       "      <td>1</td>\n",
       "      <td>0.008475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>2</td>\n",
       "      <td>0.016949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>2</td>\n",
       "      <td>0.016949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>1</td>\n",
       "      <td>0.008475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>6</td>\n",
       "      <td>0.050847</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "patient  number_patients_mutated  percent_patients_mutated\n",
       "gene                                                      \n",
       "7SK                            1                  0.008475\n",
       "A1BG                           2                  0.016949\n",
       "A1CF                           2                  0.016949\n",
       "A2LD1                          1                  0.008475\n",
       "A2M                            6                  0.050847"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.head(2)\n",
    "df1['number_patients_mutated'] = df1.notnull().sum(axis=1)\n",
    "df1['percent_patients_mutated'] = df1['number_patients_mutated']/118.0\n",
    "df2 = df1[['number_patients_mutated', 'percent_patients_mutated']]\n",
    "\n",
    "df2.head()\n",
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/somatic_mutation_frequency_summary_20190529.tsv'\n",
    "df2.to_csv(of, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(df1 == 'WT').any().any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "df1 = df1.fillna('wt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/somatic_mutation_gene_by_patient_summary_20190529.tsv'\n",
    "df1.to_csv(of, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02417</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7SK</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mutated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02006 HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02013 HTMCP-03-06-02020 HTMCP-03-06-02036  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02037  ... HTMCP-03-06-02417 HTMCP-03-06-02424  \\\n",
       "gene                       ...                                       \n",
       "7SK                   NaN  ...               NaN               NaN   \n",
       "A1BG                  NaN  ...               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "gene                                                            \n",
       "7SK                   NaN               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "gene                                                            \n",
       "7SK               mutated               NaN               NaN   \n",
       "A1BG                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02442 HTMCP-03-06-02447  \n",
       "gene                                         \n",
       "7SK                   NaN               NaN  \n",
       "A1BG                  NaN               NaN  \n",
       "\n",
       "[2 rows x 118 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "patient\n",
       "HTMCP-03-06-02002    mutated\n",
       "HTMCP-03-06-02120    mutated\n",
       "HTMCP-03-06-02260    mutated\n",
       "HTMCP-03-06-02320    mutated\n",
       "HTMCP-03-06-02326    mutated\n",
       "HTMCP-03-06-02441    mutated\n",
       "Name: PSPC1, dtype: object"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.head(2)\n",
    "df1.loc['PSPC1', :].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(51, 11)"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfn[dfn.gene=='PIK3CA'].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(305, 11)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>AA_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>E545K</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>ENST00000263967</td>\n",
       "      <td>178936091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>3</td>\n",
       "      <td>178952117</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>I1058F</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>ENST00000263967</td>\n",
       "      <td>178952117</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr        pos ref alt            patient                 impact  \\\n",
       "71   3  178936091   G   A  HTMCP-03-06-02001  NON_SYNONYMOUS_CODING   \n",
       "72   3  178952117   A   T  HTMCP-03-06-02001  NON_SYNONYMOUS_CODING   \n",
       "\n",
       "   impact_type AA_change    gene       transcript        end  \n",
       "71    MODERATE     E545K  PIK3CA  ENST00000263967  178936091  \n",
       "72    MODERATE    I1058F  PIK3CA  ENST00000263967  178952117  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset 12 genes of interest\n",
    "dfs = dfn[dfn.gene.isin(genes)&dfn.patient.isin(patients)]\n",
    "# dfs = dfmg[dfmg.patient.isin(patients)]\n",
    "\n",
    "dfs['end'] = dfs.pos\n",
    "dfs.shape\n",
    "dfs.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_sig_genes_20190527.txt'\n",
    "dfs.to_csv(of, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### make input files for cbioportal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>type</th>\n",
       "      <th>AA_change</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2706109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SILENT</td>\n",
       "      <td>R16</td>\n",
       "      <td>TTC34</td>\n",
       "      <td>ENST00000401095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2706109</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>LOW</td>\n",
       "      <td>SILENT</td>\n",
       "      <td>R16</td>\n",
       "      <td>TTC34</td>\n",
       "      <td>ENST00000574621</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr      pos ref alt            patient             impact impact_type  \\\n",
       "0   1  2706109   C   G  HTMCP-03-06-02001  SYNONYMOUS_CODING         LOW   \n",
       "0   1  2706109   C   G  HTMCP-03-06-02001  SYNONYMOUS_CODING         LOW   \n",
       "\n",
       "     type AA_change   gene       transcript  \n",
       "0  SILENT       R16  TTC34  ENST00000401095  \n",
       "0  SILENT       R16  TTC34  ENST00000574621  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfn.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "process PIK3CA: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_PIK3CA_ENST00000263967_20190527.txt\n",
      "process MAPK1: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_MAPK1_ENST00000544786_20190527.txt\n",
      "process CASP8: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_CASP8_ENST00000432109_20190527.txt\n",
      "process FBXW7: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_FBXW7_ENST00000281708_20190527.txt\n",
      "process MLL2: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_MLL2_ENST00000301067_20190527.txt\n",
      "process ZC3H6: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_ZC3H6_ENST00000409871_20190527.txt\n",
      "process PCDHGA12: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_PCDHGA12_ENST00000252085_20190527.txt\n",
      "process FAT1: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_FAT1_ENST00000441802_20190527.txt\n",
      "process ZNF750: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_ZNF750_ENST00000269394_20190527.txt\n",
      "process PCDHA9: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_PCDHA9_ENST00000532602_20190527.txt\n",
      "process PSPC1: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_PSPC1_ENST00000338910_20190527.txt\n",
      "process SLC35G5: /projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_SLC35G5_ENST00000382435_20190527.txt\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "index 0 is out of bounds for axis 0 with size 0",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-92-b80ccfbe85b7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0mdfs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdfn\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdfn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgene\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mgene\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m&\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdfn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpatient\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0misin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpatients\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mdfs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'end'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdfs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpos\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m     \u001b[0mhf_trans\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdfs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'transcript'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'transcript'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mascending\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m     \u001b[0mdfs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdfs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdfs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtranscript\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mhf_trans\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mdfs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3955\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3956\u001b[0m             \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcast_scalar_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3957\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mgetitem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3958\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3959\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mslice\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: index 0 is out of bounds for axis 0 with size 0"
     ]
    }
   ],
   "source": [
    "headers = ['Chromosome', 'Start_Position', 'Reference_Allele', 'Variant_Allele', 'Sample_ID',  'type', 'impact_type', 'Mutation_Type','Protein_Change', 'Hugo_Symbol', 'transcript', 'End_Position' ]\n",
    "for gene in genes:\n",
    "    dfs = dfn[(dfn.gene == gene)&(dfn.patient.isin(patients))]\n",
    "    dfs['end'] = dfs.pos   \n",
    "    hf_trans = dfs.groupby('transcript')['transcript'].count().sort_values(ascending=False).index[0]\n",
    "    dfs = dfs[dfs.transcript == hf_trans]\n",
    "    dfs.columns = headers\n",
    "    of = f'/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/strelka_HML_summary_{gene}_{hf_trans}_20190527.txt'\n",
    "    print(f'process {gene}: {of}')\n",
    "    dfs.to_csv(of, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_impact(impact):\n",
    "    impact = impact.upper()\n",
    "    if '+' in impact:\n",
    "        impact = impact.split('+')[0]\n",
    "    return impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/pandas/core/frame.py:3940: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  errors=errors)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(171, 8)"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>gene</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chr, pos, ref, alt, impact_type, gene, patient, impact_new]\n",
       "Index: []"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# These are not reasonable, it will remove multi-hits\n",
    "dfs['impact_new'] = dfs['impact'].apply(format_impact)\n",
    "dfs.drop('impact', axis=1, inplace=True)\n",
    "dfs.shape\n",
    "dfs[dfs.duplicated()]\n",
    "\n",
    "# dfs.drop_duplicates(keep='first', inplace=True)\n",
    "# dfs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NON_SYNONYMOUS_CODING', 'FRAME_SHIFT', 'STOP_GAINED',\n",
       "       'SYNONYMOUS_CODING', 'SPLICE_SITE_ACCEPTOR', 'SPLICE_SITE_REGION',\n",
       "       'START_GAINED', 'CODON_DELETION',\n",
       "       'CODON_CHANGE_PLUS_CODON_DELETION'], dtype=object)"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['impact_new'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(171, 8)"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs.drop_duplicates().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02012</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02036</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02239</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02260</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene            patient             impact_new\n",
       "0  CASP8  HTMCP-03-06-02001            FRAME_SHIFT\n",
       "1  CASP8  HTMCP-03-06-02012            STOP_GAINED\n",
       "2  CASP8  HTMCP-03-06-02036  NON_SYNONYMOUS_CODING\n",
       "3  CASP8  HTMCP-03-06-02239  NON_SYNONYMOUS_CODING\n",
       "4  CASP8  HTMCP-03-06-02260            FRAME_SHIFT"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = dfs.drop_duplicates().groupby(['gene', 'patient']).agg({'impact_new': ','.join}).reset_index()\n",
    "dfs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "def impact_type(x):\n",
    "#     print(x)\n",
    "    xsplit = x.split(',')\n",
    "#     print(x)\n",
    "#     type = xsplit[0].upper()\n",
    "    if len(xsplit) > 1:\n",
    "        type = 'Multi_hit'\n",
    "    elif len(xsplit) == 1:\n",
    "        type = xsplit[0]\n",
    "        if type == 'MISSENSE_VARIANT' or type == 'NON_SYNONYMOUS_CODING':\n",
    "           type = 'Non-synonymous'\n",
    "        elif type == 'SYNONYMOUS_VARIANT' or type == 'SYNONYMOUS_CODING':\n",
    "           type = 'Synonymous'\n",
    "        elif  type == 'SPLICE_SITE_ACCEPTOR' or  type == 'SPLICE_SITE_DONOR' or type == 'SPLICE_SITE_REGION':\n",
    "           type = 'Splice site'\n",
    "        elif type == 'STOP_LOST':\n",
    "           type = 'Stop lost'\n",
    "        elif type == 'STOP_GAINED':\n",
    "           type = 'Stop gained'\n",
    "        elif  type == 'START_LOST':\n",
    "           type = 'Start lost'\n",
    "        elif type == 'START_GAINED':\n",
    "           type = 'Start gained'\n",
    "        elif type == 'FRAME_SHIFT' or type == 'FRAMESHIFT_VARIANT':\n",
    "           type = 'Frameshift'\n",
    "        elif type == 'INFRAME_DELETION' or type == 'CODON_INSERTION' or type == 'CODON_DELETION' or type == 'CODON_CHANGE_PLUS_CODON_INSERTION' or type == 'CODON_CHANGE_PLUS_CODON_DELETION' or type.lower() == 'disruptive_inframe_deletion':\n",
    "           type = 'Codon indel'                \n",
    "        else: exit(1);print('ERROR');print(x)\n",
    "    return type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(139, 4)"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(139, 3)"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02040</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02047</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02344</th>\n",
       "      <th>HTMCP-03-06-02346</th>\n",
       "      <th>HTMCP-03-06-02354</th>\n",
       "      <th>HTMCP-03-06-02392</th>\n",
       "      <th>HTMCP-03-06-02411</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CASP8</th>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAT1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 80 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02006  \\\n",
       "gene                                                            \n",
       "CASP8          Frameshift               NaN               NaN   \n",
       "FAT1                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02012 HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "gene                                                            \n",
       "CASP8         Stop gained               NaN               NaN   \n",
       "FAT1                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02036 HTMCP-03-06-02040 HTMCP-03-06-02042  \\\n",
       "gene                                                            \n",
       "CASP8      Non-synonymous               NaN               NaN   \n",
       "FAT1          Stop gained        Synonymous               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02047  ... HTMCP-03-06-02344 HTMCP-03-06-02346  \\\n",
       "gene                       ...                                       \n",
       "CASP8                 NaN  ...               NaN               NaN   \n",
       "FAT1                  NaN  ...               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02354 HTMCP-03-06-02392 HTMCP-03-06-02411  \\\n",
       "gene                                                            \n",
       "CASP8                 NaN               NaN               NaN   \n",
       "FAT1          Stop gained               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02435  \\\n",
       "gene                                                            \n",
       "CASP8                 NaN               NaN               NaN   \n",
       "FAT1           Frameshift         Multi_hit    Non-synonymous   \n",
       "\n",
       "patient HTMCP-03-06-02441 HTMCP-03-06-02447  \n",
       "gene                                         \n",
       "CASP8                 NaN               NaN  \n",
       "FAT1                  NaN               NaN  \n",
       "\n",
       "[2 rows x 80 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(12, 80)"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['impact'] = dfs['impact_new'].apply(lambda x: impact_type(x))\n",
    "dfs.shape\n",
    "dfs = dfs[['gene', 'patient', 'impact']].drop_duplicates()\n",
    "dfs.shape\n",
    "dfs = dfs.set_index(['gene', 'patient'])['impact'].unstack()\n",
    "dfs.head(2)\n",
    "dfs.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5. add in covariates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Squamous', 'Adenosquamous', 'Adeno', 'Neuroendocrine'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02417</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient            HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "HIV_status                  Positive          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02006 HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02013 HTMCP-03-06-02020 HTMCP-03-06-02036  \\\n",
       "HIV_status                  Negative          Positive          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02037  ... HTMCP-03-06-02417 HTMCP-03-06-02424  \\\n",
       "HIV_status                  Negative  ...          Positive          Positive   \n",
       "Putative_histology          Squamous  ...          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02427 HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02435 HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous             Adeno   \n",
       "\n",
       "patient            HTMCP-03-06-02442 HTMCP-03-06-02447  \n",
       "HIV_status                  Positive          Positive  \n",
       "Putative_histology          Squamous          Squamous  \n",
       "\n",
       "[2 rows x 118 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(2, 118)"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get meta track for example histology\n",
    "f3 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutation_load_clinic.txt'\n",
    "ddf = pd.read_csv(f3, sep='\\t')\n",
    "ddf = ddf[['patient', 'HIV_status', 'Putative_histology']]\n",
    "edf = ddf.set_index('patient').T\n",
    "edf.loc['Putative_histology',].unique()\n",
    "edf = edf[patients]\n",
    "edf.head(2)\n",
    "edf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "patient\n",
       "HTMCP-03-06-02047    True\n",
       "HTMCP-03-06-02110    True\n",
       "HTMCP-03-06-02176    True\n",
       "HTMCP-03-06-02203    True\n",
       "HTMCP-03-06-02216    True\n",
       "HTMCP-03-06-02235    True\n",
       "Name: Putative_histology, dtype: bool"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02047</th>\n",
       "      <th>HTMCP-03-06-02110</th>\n",
       "      <th>HTMCP-03-06-02176</th>\n",
       "      <th>HTMCP-03-06-02203</th>\n",
       "      <th>HTMCP-03-06-02216</th>\n",
       "      <th>HTMCP-03-06-02235</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Neuroendocrine</td>\n",
       "      <td>Adenosquamous</td>\n",
       "      <td>Adenosquamous</td>\n",
       "      <td>Adenosquamous</td>\n",
       "      <td>Adenosquamous</td>\n",
       "      <td>Neuroendocrine</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "patient            HTMCP-03-06-02047 HTMCP-03-06-02110 HTMCP-03-06-02176  \\\n",
       "HIV_status                  Negative          Positive          Negative   \n",
       "Putative_histology    Neuroendocrine     Adenosquamous     Adenosquamous   \n",
       "\n",
       "patient            HTMCP-03-06-02203 HTMCP-03-06-02216 HTMCP-03-06-02235  \n",
       "HIV_status                  Negative          Positive          Positive  \n",
       "Putative_histology     Adenosquamous     Adenosquamous    Neuroendocrine  "
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s = (edf.loc['Putative_histology',:].isin(['Adenosquamous', 'Neuroendocrine']))\n",
    "s[s]\n",
    "edf[s[s].index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "cases other than squamous or adeno\n",
    "# patient\tHIV_status\tHistology\tAPOBEC_signature\tClinical_stage\tAge_at_diagnosis\n",
    "# HTMCP-03-06-02235\tPositive\tNeuroendocrine\tweak\tStage IV\t<45\n",
    "# HTMCP-03-06-02110\tPositive\tAdenosquamous\tweak\tStage IV\t<45\n",
    "# HTMCP-03-06-02216\tPositive\tAdenosquamous\tweak\tStage III\t45-65\n",
    "# HTMCP-03-06-02203\tNegative\tAdenosquamous\tmoderate\tStage II\t45-65\n",
    "# HTMCP-03-06-02047\tNegative\tNeuroendocrine\tweak\tStage IV\t45-65\n",
    "# HTMCP-03-06-02176\tNegative\tAdenosquamous\tmoderate\tStage III\t45-65"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs = pd.concat([edf, dfs], sort=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(14, 118)"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs = dfs[patients]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 6. add in mutation rate (fraction of patients)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs['occurrence'] = (dfs.notnull().sum(axis=1)/(dfs.shape[1] -2))\n",
    "dfs['percentage'] = ['{0}({1}%)'.format(i[0], int(round(i[1]*100))) for i in zip(dfs.index.tolist(), dfs.occurrence)]\n",
    "# dfs.loc['HIV_status', 'percentage'] = 'HIV_Status'\n",
    "# dfs.loc['Putative_histology', 'percentage' ] = 'Puatative_histology'\n",
    "sdf = dfs.drop('occurrence', axis=1).set_index('percentage', drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02417</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>percentage</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status(102%)</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology(102%)</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CASP8(7%)</th>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAT1(19%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FBXW7(10%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAPK1(5%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL2(15%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCDHA9(3%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCDHGA12(5%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA(35%)</th>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSPC1(5%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC35G5(7%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZC3H6(3%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF750(4%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14 rows × 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                       Frameshift               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN       Stop gained   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                      Multi_hit    Non-synonymous   \n",
       "PSPC1(5%)                              NaN    Non-synonymous   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN        Frameshift   \n",
       "\n",
       "                         HTMCP-03-06-02003 HTMCP-03-06-02006  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN    Non-synonymous   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN    Non-synonymous   \n",
       "PIK3CA(35%)                            NaN               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Negative          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN       Stop gained   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                            NaN         Multi_hit   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Negative          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                 Non-synonymous               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN         Multi_hit   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02036 HTMCP-03-06-02037  ...  \\\n",
       "percentage                                                    ...   \n",
       "HIV_status(102%)                  Negative          Negative  ...   \n",
       "Putative_histology(102%)          Squamous          Squamous  ...   \n",
       "CASP8(7%)                   Non-synonymous               NaN  ...   \n",
       "FAT1(19%)                      Stop gained               NaN  ...   \n",
       "FBXW7(10%)                             NaN               NaN  ...   \n",
       "MAPK1(5%)                              NaN               NaN  ...   \n",
       "MLL2(15%)                              NaN               NaN  ...   \n",
       "PCDHA9(3%)                             NaN               NaN  ...   \n",
       "PCDHGA12(5%)                           NaN               NaN  ...   \n",
       "PIK3CA(35%)                 Non-synonymous               NaN  ...   \n",
       "PSPC1(5%)                              NaN               NaN  ...   \n",
       "SLC35G5(7%)                            NaN               NaN  ...   \n",
       "ZC3H6(3%)                              NaN               NaN  ...   \n",
       "ZNF750(4%)                             NaN               NaN  ...   \n",
       "\n",
       "                         HTMCP-03-06-02417 HTMCP-03-06-02424  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                            NaN               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02427 HTMCP-03-06-02428  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                       Frameshift         Multi_hit   \n",
       "FBXW7(10%)                             NaN    Non-synonymous   \n",
       "MAPK1(5%)                              NaN    Non-synonymous   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                            NaN    Non-synonymous   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN         Multi_hit   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN    Non-synonymous   \n",
       "FBXW7(10%)                             NaN        Frameshift   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN        Frameshift   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN    Non-synonymous   \n",
       "PIK3CA(35%)                            NaN               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous             Adeno   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                            NaN               NaN   \n",
       "PSPC1(5%)                              NaN    Non-synonymous   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02442 HTMCP-03-06-02447  \n",
       "percentage                                                    \n",
       "HIV_status(102%)                  Positive          Positive  \n",
       "Putative_histology(102%)          Squamous          Squamous  \n",
       "CASP8(7%)                              NaN               NaN  \n",
       "FAT1(19%)                              NaN               NaN  \n",
       "FBXW7(10%)                             NaN               NaN  \n",
       "MAPK1(5%)                              NaN               NaN  \n",
       "MLL2(15%)                              NaN               NaN  \n",
       "PCDHA9(3%)                             NaN               NaN  \n",
       "PCDHGA12(5%)                           NaN               NaN  \n",
       "PIK3CA(35%)                            NaN    Non-synonymous  \n",
       "PSPC1(5%)                              NaN               NaN  \n",
       "SLC35G5(7%)                            NaN               NaN  \n",
       "ZC3H6(3%)                              NaN               NaN  \n",
       "ZNF750(4%)                             NaN               NaN  \n",
       "\n",
       "[14 rows x 118 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients/smgs_reviewed_details_20190326.txt'\n",
    "# sdf.to_csv(of, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02417</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>percentage</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status(102%)</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology(102%)</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "\n",
       "                         HTMCP-03-06-02003 HTMCP-03-06-02006  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "\n",
       "                         HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Negative          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "\n",
       "                         HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Negative          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "\n",
       "                         HTMCP-03-06-02036 HTMCP-03-06-02037  ...  \\\n",
       "percentage                                                    ...   \n",
       "HIV_status(102%)                  Negative          Negative  ...   \n",
       "Putative_histology(102%)          Squamous          Squamous  ...   \n",
       "\n",
       "                         HTMCP-03-06-02417 HTMCP-03-06-02424  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "\n",
       "                         HTMCP-03-06-02427 HTMCP-03-06-02428  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "\n",
       "                         HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "\n",
       "                         HTMCP-03-06-02437 HTMCP-03-06-02441  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous             Adeno   \n",
       "\n",
       "                         HTMCP-03-06-02442 HTMCP-03-06-02447  \n",
       "percentage                                                    \n",
       "HIV_status(102%)                  Positive          Positive  \n",
       "Putative_histology(102%)          Squamous          Squamous  \n",
       "\n",
       "[2 rows x 118 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Positive', 'Negative', 'Squamous', 'Neuroendocrine', 'Adeno',\n",
       "       'Adenosquamous', 'Frameshift', nan, 'Stop gained',\n",
       "       'Non-synonymous', 'Synonymous', 'Multi_hit', 'Start gained'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.unique(sdf.values.ravel())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = sdf.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "dftmp = pd.DataFrame([list(np.array(re.split('\\(|\\)', i))[[0,1]]) for i in df.index.tolist()], columns=['gene', 'mfreq'])\n",
    "\n",
    "df['gene'] = dftmp.gene.tolist()\n",
    "df['mfreq'] = dftmp.mfreq.tolist()\n",
    "\n",
    "df['mfreq'] = df['mfreq'].apply(lambda x: int(x.replace('%', '')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "      <th>mfreq</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA</th>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAT1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>Multi_hit</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 119 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "gene                                                                       \n",
       "HIV_status                  Positive          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "PIK3CA                     Multi_hit    Non-synonymous               NaN   \n",
       "FAT1                             NaN               NaN               NaN   \n",
       "MLL2                             NaN       Stop gained               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02006 HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "gene                                                                       \n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "PIK3CA                           NaN               NaN         Multi_hit   \n",
       "FAT1                             NaN               NaN               NaN   \n",
       "MLL2                             NaN               NaN               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02013 HTMCP-03-06-02020 HTMCP-03-06-02036  \\\n",
       "gene                                                                       \n",
       "HIV_status                  Negative          Positive          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "PIK3CA                Non-synonymous               NaN    Non-synonymous   \n",
       "FAT1                             NaN               NaN       Stop gained   \n",
       "MLL2                             NaN               NaN               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02037  ... HTMCP-03-06-02424 HTMCP-03-06-02427  \\\n",
       "gene                                  ...                                       \n",
       "HIV_status                  Negative  ...          Positive          Positive   \n",
       "Putative_histology          Squamous  ...          Squamous          Squamous   \n",
       "PIK3CA                           NaN  ...               NaN               NaN   \n",
       "FAT1                             NaN  ...               NaN        Frameshift   \n",
       "MLL2                             NaN  ...               NaN               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02428 HTMCP-03-06-02434 HTMCP-03-06-02435  \\\n",
       "gene                                                                       \n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "PIK3CA                Non-synonymous               NaN               NaN   \n",
       "FAT1                       Multi_hit               NaN    Non-synonymous   \n",
       "MLL2                             NaN               NaN        Frameshift   \n",
       "\n",
       "                   HTMCP-03-06-02437 HTMCP-03-06-02441 HTMCP-03-06-02442  \\\n",
       "gene                                                                       \n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous             Adeno          Squamous   \n",
       "PIK3CA                           NaN               NaN               NaN   \n",
       "FAT1                             NaN               NaN               NaN   \n",
       "MLL2                             NaN               NaN               NaN   \n",
       "\n",
       "                   HTMCP-03-06-02447 mfreq  \n",
       "gene                                        \n",
       "HIV_status                  Positive   102  \n",
       "Putative_histology          Squamous   102  \n",
       "PIK3CA                Non-synonymous    35  \n",
       "FAT1                             NaN    19  \n",
       "MLL2                             NaN    15  \n",
       "\n",
       "[5 rows x 119 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.sort_values('mfreq', ascending=False)\n",
    "\n",
    "df.set_index('gene', drop=True, inplace=True)\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfa = (df.loc['HIV_status',] == 'Negative') \n",
    "dfa = df[dfa[dfa].index]\n",
    "df['hiv_neg_mfreq'] = (dfa.notnull().sum(axis=1)/dfa.shape[1]*100).astype(int)\n",
    "dfb = (df.loc['HIV_status',] == 'Positive') \n",
    "dfb = df[dfb[dfb].index]\n",
    "df['hiv_pos_mfreq'] = (dfb.notnull().sum(axis=1)/dfb.shape[1]*100).astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.sort_values(df.index.tolist(), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['percentage'] = ['{0}({1}%)'.format(i[0], i[1]) for i in [(i[0], i[1]) for i in zip(df.index, df.mfreq)] ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.set_index('percentage', drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [],
   "source": [
    "f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients/smgs_reviewed_details_with_frequency_20190326.txt'\n",
    "df.to_csv(f2)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
