{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import matplotlib\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import scipy.stats as ss\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib import style, colors\n",
    "from matplotlib import gridspec\n",
    "import random\n",
    "import string\n",
    "import math\n",
    "from tqdm import tqdm\n",
    "# matplotlib.style.use('ggplot')\n",
    "%matplotlib inline\n",
    "matplotlib.rcParams['figure.figsize'] = (12, 6)\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Plot imports\n",
    "import plotly.plotly as py\n",
    "import plotly.graph_objs as go\n",
    "\n",
    "## Offline mode\n",
    "from plotly.offline import init_notebook_mode, iplot\n",
    "init_notebook_mode(connected=True)\n",
    "\n",
    "import cufflinks as cf\n",
    "cf.go_offline()\n",
    "# Set the global theme\n",
    "cf.set_config_file(world_readable=True, theme='pearl', offline=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "a good post about match\n",
    "\n",
    "https://stackoverflow.com/questions/24958358/use-regular-expression-in-python-to-find-two-strings-in-line"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "a table details the equivalence of ontology an classic effect description\n",
    "\n",
    "http://snpeff.sourceforge.net/SnpEff_manual.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. parsing mutationseq vcf files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_mutationseq_vcf(vcf):\n",
    "    '''\n",
    "    include all variants high, moderate, low, and modifiers\n",
    "    '''\n",
    "    df = pd.read_csv(vcf, comment='#', sep='\\t', header=None, low_memory=False)\n",
    "    patient = vcf.split('/')[4]\n",
    "    df = df[[0,1,3,4,7]]\n",
    "    df.columns = ['chr', 'pos', 'ref', 'alt', 'effect']\n",
    "    df = df[(df['effect'].str.contains(\"HIGH\"))|(df['effect'].str.contains(\"MODERATE\"))|(df['effect'].str.contains(\"LOW\"))|(df['effect'].str.contains(\"MODIFIER\"))]\n",
    "    if not df.empty:\n",
    "        df['impact'], df['impact_type'], df['gene'] = df['effect'].apply(lambda x: parse_effect(x)).str.split('@').str\n",
    "        df['patient'] = patient\n",
    "    df = df.drop('effect', axis=1)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep it easy for now pick HIGH, MODERATE and then LOW\n",
    "def parse_effect(line):\n",
    "    effs = line.split('EFF=')[1].split(',')\n",
    "    #     extract impact, impact_type and gene\n",
    "    effs = ['@'.join(list(np.array(re.split('\\(|\\|',ef))[[0,1,6]])) for ef in effs if ('HIGH' in  ef) or ('MODERATE' in ef) or ('LOW' in ef) or ('MODIFIER' in ef)]\n",
    "    effs = list(set(effs))\n",
    "    high = [ef for ef in effs if 'HIGH' in ef]\n",
    "    moderate = [ef for ef in effs if 'MODERATE' in ef]\n",
    "    low = [ef for ef in effs if 'LOW' in ef]\n",
    "    modifier = [ef for ef in effs if 'MODIFIER' in ef]\n",
    "    if high:\n",
    "        anno = high[0]\n",
    "    elif moderate:\n",
    "        anno = moderate[0]\n",
    "    elif low:\n",
    "        anno = low[0]\n",
    "    elif modifier:\n",
    "        anno = modifier[0]\n",
    "    else:\n",
    "        print('ERROR!')\n",
    "#     make sure the genes have the same name\n",
    "    genes = [ef.split('@')[2] for ef in effs]\n",
    "    \n",
    "    return anno\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # example run\n",
    "# ft = '/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-03-06-02020/hg19a/GSH/A37247_A56468/strelka/23223/bwa/results/passed.somatic.snvs.eff.dbSNP_v137.cosmic_v64.annotations.vcf'\n",
    "# # assert count_coding_mutations(ft) == 266\n",
    "# df = parse_strelka_vcf(ft)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Scope: genes and cohort size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 118 patients\n",
    "f = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients.txt'\n",
    "with open(f) as file:\n",
    "    patients = [line.strip() for line in file]\n",
    "assert len(patients) == 118"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 12 SMGS\n",
    "f2 = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients/smgs_reviewed.txt'\n",
    "with open (f2) as file:\n",
    "    genes = [line.strip() for line in file]\n",
    "assert len(genes) == 12"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. mutationseq vcf files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HIV_status</th>\n",
       "      <th>DNA_lib</th>\n",
       "      <th>source</th>\n",
       "      <th>status</th>\n",
       "      <th>RNA_lib</th>\n",
       "      <th>ribodepleted_lib</th>\n",
       "      <th>DNA_bam</th>\n",
       "      <th>RNA_bam</th>\n",
       "      <th>DNA_single_vcf</th>\n",
       "      <th>paired_mpileup_vcf</th>\n",
       "      <th>...</th>\n",
       "      <th>other_vcf</th>\n",
       "      <th>DNA_tc</th>\n",
       "      <th>RNA_tc</th>\n",
       "      <th>cnv</th>\n",
       "      <th>bbt_transcriptome</th>\n",
       "      <th>bbt_genome</th>\n",
       "      <th>bbt_transcriptome_other_bacterial</th>\n",
       "      <th>bbt_genome_other_bacterial</th>\n",
       "      <th>bbt_transcriptome_other_viral</th>\n",
       "      <th>bbt_genome_other_viral</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <td>Positive</td>\n",
       "      <td>A37234</td>\n",
       "      <td>HTMCP_124</td>\n",
       "      <td>Malignant</td>\n",
       "      <td>A37700</td>\n",
       "      <td>A56295</td>\n",
       "      <td>/projects/analysis/analysis22/A37234/merge_bwa...</td>\n",
       "      <td>/projects/analysis/analysis22/IX3433/C67GDANXX...</td>\n",
       "      <td>/projects/analysis/analysis22/A37234/merge_bwa...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>55.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <td>Negative</td>\n",
       "      <td>A37235</td>\n",
       "      <td>HTMCP_125</td>\n",
       "      <td>Malignant</td>\n",
       "      <td>A37701</td>\n",
       "      <td>A56296</td>\n",
       "      <td>/projects/analysis/analysis22/A37235/merge_bwa...</td>\n",
       "      <td>/projects/analysis/analysis22/IX3432/C67GDANXX...</td>\n",
       "      <td>/projects/analysis/analysis22/A37235/merge_bwa...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "      <td>/projects/NCI_validation2_assembly/NCI_SAIC_HI...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  HIV_status DNA_lib     source     status RNA_lib  \\\n",
       "patient                                                              \n",
       "HTMCP-03-06-02001   Positive  A37234  HTMCP_124  Malignant  A37700   \n",
       "HTMCP-03-06-02002   Negative  A37235  HTMCP_125  Malignant  A37701   \n",
       "\n",
       "                  ribodepleted_lib  \\\n",
       "patient                              \n",
       "HTMCP-03-06-02001           A56295   \n",
       "HTMCP-03-06-02002           A56296   \n",
       "\n",
       "                                                             DNA_bam  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/analysis/analysis22/A37234/merge_bwa...   \n",
       "HTMCP-03-06-02002  /projects/analysis/analysis22/A37235/merge_bwa...   \n",
       "\n",
       "                                                             RNA_bam  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/analysis/analysis22/IX3433/C67GDANXX...   \n",
       "HTMCP-03-06-02002  /projects/analysis/analysis22/IX3432/C67GDANXX...   \n",
       "\n",
       "                                                      DNA_single_vcf  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/analysis/analysis22/A37234/merge_bwa...   \n",
       "HTMCP-03-06-02002  /projects/analysis/analysis22/A37235/merge_bwa...   \n",
       "\n",
       "                   paired_mpileup_vcf  \\\n",
       "patient                                 \n",
       "HTMCP-03-06-02001                 NaN   \n",
       "HTMCP-03-06-02002                 NaN   \n",
       "\n",
       "                                         ...                          \\\n",
       "patient                                  ...                           \n",
       "HTMCP-03-06-02001                        ...                           \n",
       "HTMCP-03-06-02002                        ...                           \n",
       "\n",
       "                  other_vcf DNA_tc RNA_tc  \\\n",
       "patient                                     \n",
       "HTMCP-03-06-02001       NaN   55.0   55.0   \n",
       "HTMCP-03-06-02002       NaN   70.0   70.0   \n",
       "\n",
       "                                                                 cnv  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "HTMCP-03-06-02002  /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...   \n",
       "\n",
       "                                                   bbt_transcriptome  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                                          bbt_genome  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                   bbt_transcriptome_other_bacterial  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                          bbt_genome_other_bacterial  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                       bbt_transcriptome_other_viral  \\\n",
       "patient                                                                \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...   \n",
       "\n",
       "                                              bbt_genome_other_viral  \n",
       "patient                                                               \n",
       "HTMCP-03-06-02001  /projects/NCI_validation2_assembly/NCI_SAIC_HI...  \n",
       "HTMCP-03-06-02002  /projects/NCI_validation2_assembly/NCI_SAIC_HI...  \n",
       "\n",
       "[2 rows x 25 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "Index(['HIV_status', 'DNA_lib', 'source', 'status', 'RNA_lib',\n",
       "       'ribodepleted_lib', 'DNA_bam', 'RNA_bam', 'DNA_single_vcf',\n",
       "       'paired_mpileup_vcf', 'strelka_snv_vcf', 'strelka_indel_vcf',\n",
       "       'mutseq_snv_vcf', 'mutseq_snv_vcf_old', 'RNA_single_vcf', 'other_vcf',\n",
       "       'DNA_tc', 'RNA_tc', 'cnv', 'bbt_transcriptome', 'bbt_genome',\n",
       "       'bbt_transcriptome_other_bacterial', 'bbt_genome_other_bacterial',\n",
       "       'bbt_transcriptome_other_viral', 'bbt_genome_other_viral'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "patient\n",
       "HTMCP-03-06-02001    /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...\n",
       "HTMCP-03-06-02002    /projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-...\n",
       "Name: mutseq_snv_vcf, dtype: object"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(118,)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "'/projects/somatic/NCI_SAIC_HIV_Cervical/HTMCP-03-06-02261/hg19a/GSH/A68235_A66443/mutationseq/63195/A68235_A66443.eff.dbSNP_v149.cosmic_v82.clinvar_20170801.annotations.classic.vcf'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f1 = '/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients_new/124_patients_bam_vcf_20190115.txt'\n",
    "df = pd.read_csv(f1, sep='\\t', index_col='patient')\n",
    "df.head(2)\n",
    "\n",
    "df.columns\n",
    "\n",
    "ndf = df['mutseq_snv_vcf'].dropna()\n",
    "ndf.head(2)\n",
    "\n",
    "ndf = ndf.reindex(patients)\n",
    "ndf.shape\n",
    "ndf.loc['HTMCP-03-06-02261',]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. concatenate snvs, mutationseq does not report indels"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## look into ~800 pog strelka snv calls to see if systematic noise exist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/projects/POG/POG_data/POG001/wgs/arch1_t_A10509_blood1_n_A10501/strelka/bwa/results/passed.somatic.snvs.eff.dbSNP_v137.cosmic_v64.annotations.vcf',\n",
       " '/projects/POG/POG_data/POG001/wgs/biop1_t_A10504_blood1_n_A10501/strelka/bwa/results/passed.somatic.snvs.eff.dbSNP_v137.cosmic_v64.annotations.vcf']"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "working on POG001\n",
      "working on POG001\n",
      "working on POG002\n",
      "working on POG002\n",
      "working on POG003\n",
      "working on POG003\n",
      "working on POG003\n",
      "working on POG004\n",
      "working on POG005\n",
      "working on POG005\n",
      "working on POG006\n",
      "working on POG007\n",
      "working on POG008\n",
      "working on POG008\n",
      "working on POG009\n",
      "working on POG010\n",
      "working on POG011\n",
      "working on POG011\n",
      "working on POG012\n",
      "working on POG012\n",
      "working on POG012\n",
      "working on POG014\n",
      "working on POG015\n",
      "working on POG015\n",
      "working on POG016\n",
      "working on POG017\n",
      "working on POG017\n",
      "working on POG018\n",
      "working on POG018\n",
      "working on POG019\n",
      "working on POG019\n",
      "working on POG020\n",
      "working on POG020\n",
      "working on POG021\n",
      "working on POG021\n",
      "working on POG021\n",
      "working on POG022\n",
      "working on POG022\n",
      "working on POG023\n",
      "working on POG023\n",
      "working on POG024\n",
      "working on POG025\n",
      "working on POG025\n",
      "working on POG026\n",
      "working on POG026\n",
      "working on POG027\n",
      "working on POG027\n",
      "working on POG029\n",
      "working on POG030\n",
      "working on POG030\n",
      "working on POG031\n",
      "working on POG032\n",
      "working on POG033\n",
      "working on POG034\n",
      "working on POG034\n",
      "working on POG035\n",
      "working on POG035\n",
      "working on POG036\n",
      "working on POG036\n",
      "working on POG036\n",
      "working on POG038\n",
      "working on POG038\n",
      "working on POG040\n",
      "working on POG040\n",
      "working on POG041\n",
      "working on POG041\n",
      "working on POG042\n",
      "working on POG042\n",
      "working on POG044\n",
      "working on POG044\n",
      "working on POG045\n",
      "working on POG046\n",
      "working on POG046\n",
      "working on POG047\n",
      "working on POG047\n",
      "working on POG047\n",
      "working on POG048\n",
      "working on POG048\n",
      "working on POG049\n",
      "working on POG050\n",
      "working on POG050\n",
      "working on POG051\n",
      "working on POG052\n",
      "working on POG052\n",
      "working on POG054\n",
      "working on POG054\n",
      "working on POG056\n",
      "working on POG056\n",
      "working on POG057\n",
      "working on POG058\n",
      "working on POG058\n",
      "working on POG059\n",
      "working on POG059\n",
      "working on POG061\n",
      "working on POG061\n",
      "working on POG062\n",
      "working on POG062\n",
      "working on POG062\n",
      "working on POG063\n",
      "working on POG063\n",
      "working on POG064\n",
      "working on POG064\n",
      "working on POG066\n",
      "working on POG067\n",
      "working on POG067\n",
      "working on POG068\n",
      "working on POG068\n",
      "working on POG069\n",
      "working on POG071\n",
      "working on POG071\n",
      "working on POG073\n",
      "working on POG074\n",
      "working on POG074\n",
      "working on POG075\n",
      "working on POG075\n",
      "working on POG076\n",
      "working on POG077\n",
      "working on POG077\n",
      "working on POG078\n",
      "working on POG078\n",
      "working on POG079\n",
      "working on POG079\n",
      "working on POG080\n",
      "working on POG083\n",
      "working on POG084\n",
      "working on POG084\n",
      "working on POG085\n",
      "working on POG086\n",
      "working on POG086\n",
      "working on POG086\n",
      "working on POG088\n",
      "working on POG089\n",
      "working on POG089\n",
      "working on POG090\n",
      "working on POG091\n",
      "working on POG092\n",
      "working on POG093\n",
      "working on POG093\n",
      "working on POG094\n",
      "working on POG094\n",
      "working on POG095\n",
      "working on POG097\n",
      "working on POG098\n",
      "working on POG099\n",
      "working on POG099\n",
      "working on POG1000\n",
      "working on POG1001\n",
      "working on POG1002\n",
      "working on POG1003\n",
      "working on POG1006\n",
      "working on POG1007\n",
      "working on POG1008\n",
      "working on POG1009\n",
      "working on POG1010\n",
      "working on POG1011\n",
      "working on POG1013\n",
      "working on POG1014\n",
      "working on POG1016\n",
      "working on POG1017\n",
      "working on POG1018\n",
      "working on POG1020\n",
      "working on POG1021\n",
      "working on POG1022\n",
      "working on POG1023\n",
      "working on POG1024\n",
      "working on POG1025\n",
      "working on POG1026\n",
      "working on POG1027\n",
      "working on POG1028\n",
      "working on POG1029\n",
      "working on POG1030\n",
      "working on POG1031\n",
      "working on POG1032\n",
      "working on POG1033\n",
      "working on POG1034\n",
      "working on POG1035\n",
      "working on POG1036\n",
      "working on POG1037\n",
      "working on POG1038\n",
      "working on POG103\n",
      "working on POG1040\n",
      "working on POG1041\n",
      "working on POG1044\n",
      "working on POG104\n",
      "working on POG106\n",
      "working on POG106\n",
      "working on POG106\n",
      "working on POG106\n",
      "working on POG107\n",
      "working on POG108\n",
      "working on POG109\n",
      "working on POG110\n",
      "working on POG111\n",
      "working on POG114\n",
      "working on POG116\n",
      "working on POG116\n",
      "working on POG117\n",
      "working on POG117\n",
      "working on POG118\n",
      "working on POG120\n",
      "working on POG122\n",
      "working on POG123\n",
      "working on POG124\n",
      "working on POG125\n",
      "working on POG127\n",
      "working on POG128\n",
      "working on POG129\n",
      "working on POG130\n",
      "working on POG130\n",
      "working on POG130\n",
      "working on POG130\n",
      "working on POG131\n",
      "working on POG132\n",
      "working on POG133\n",
      "working on POG134\n",
      "working on POG137\n",
      "working on POG138\n",
      "working on POG139\n",
      "working on POG140\n",
      "working on POG141\n",
      "working on POG143\n",
      "working on POG144\n",
      "working on POG144\n",
      "working on POG145\n",
      "working on POG146\n",
      "working on POG147\n",
      "working on POG148\n",
      "working on POG149\n",
      "working on POG151\n",
      "working on POG153\n",
      "working on POG157\n",
      "working on POG158\n",
      "working on POG159\n",
      "working on POG161\n",
      "working on POG162\n",
      "working on POG162\n",
      "working on POG163\n",
      "working on POG165\n",
      "working on POG166\n",
      "working on POG167\n",
      "working on POG168\n",
      "working on POG169\n",
      "working on POG170\n",
      "working on POG171\n",
      "working on POG172\n",
      "working on POG173\n",
      "working on POG174\n",
      "working on POG175\n",
      "working on POG176\n",
      "working on POG178\n",
      "working on POG181\n",
      "working on POG183\n",
      "working on POG184\n",
      "working on POG185\n",
      "working on POG186\n",
      "working on POG187\n",
      "working on POG189\n",
      "working on POG190\n",
      "working on POG192\n",
      "working on POG194\n",
      "working on POG194\n",
      "working on POG195\n",
      "working on POG196\n",
      "working on POG197\n",
      "working on POG198\n",
      "working on POG199\n",
      "working on POG200\n",
      "working on POG201\n",
      "working on POG202\n",
      "working on POG203\n",
      "working on POG204\n",
      "working on POG205\n",
      "working on POG206\n",
      "working on POG208\n",
      "working on POG209\n",
      "working on POG211\n",
      "working on POG212\n",
      "working on POG213\n",
      "working on POG215\n",
      "working on POG216\n",
      "working on POG216\n",
      "working on POG217\n",
      "working on POG217\n",
      "working on POG219\n",
      "working on POG221\n",
      "working on POG222\n",
      "working on POG222\n",
      "working on POG223\n",
      "working on POG225\n",
      "working on POG227\n",
      "working on POG228\n",
      "working on POG230\n",
      "working on POG233\n",
      "working on POG235\n",
      "working on POG236\n",
      "working on POG237\n",
      "working on POG237\n",
      "working on POG238\n",
      "working on POG239\n",
      "working on POG240\n",
      "working on POG243\n",
      "working on POG244\n",
      "working on POG246\n",
      "working on POG247\n",
      "working on POG248\n",
      "working on POG251\n",
      "working on POG252\n",
      "working on POG253\n",
      "working on POG254\n",
      "working on POG255\n",
      "working on POG256\n",
      "working on POG258\n",
      "working on POG261\n",
      "working on POG262\n",
      "working on POG263\n",
      "working on POG264\n",
      "working on POG265\n",
      "working on POG266\n",
      "working on POG268\n",
      "working on POG269\n",
      "working on POG270\n",
      "working on POG271\n",
      "working on POG272\n",
      "working on POG275\n",
      "working on POG277\n",
      "working on POG279\n",
      "working on POG280\n",
      "working on POG282\n",
      "working on POG284\n",
      "working on POG286\n",
      "working on POG287\n",
      "working on POG287\n",
      "working on POG288\n",
      "working on POG289\n",
      "working on POG294\n",
      "working on POG295\n",
      "working on POG295\n",
      "working on POG296\n",
      "working on POG296\n",
      "working on POG297\n",
      "working on POG298\n",
      "working on POG302\n",
      "working on POG302\n",
      "working on POG303\n",
      "working on POG304\n",
      "working on POG307\n",
      "working on POG309\n",
      "working on POG311\n",
      "working on POG312\n",
      "working on POG313\n",
      "working on POG314\n",
      "working on POG314\n",
      "working on POG315\n",
      "working on POG316\n",
      "working on POG316\n",
      "working on POG317\n",
      "working on POG318\n",
      "working on POG320\n",
      "working on POG323\n",
      "working on POG324\n",
      "working on POG326\n",
      "working on POG328\n",
      "working on POG329\n",
      "working on POG330\n",
      "working on POG330\n",
      "working on POG332\n",
      "working on POG336\n",
      "working on POG338\n",
      "working on POG339\n",
      "working on POG341\n",
      "working on POG342\n",
      "working on POG343\n",
      "working on POG345\n",
      "working on POG346\n",
      "working on POG347\n",
      "working on POG348\n",
      "working on POG350\n",
      "working on POG352\n",
      "working on POG353\n",
      "working on POG354\n",
      "working on POG355\n",
      "working on POG356\n",
      "working on POG357\n",
      "working on POG357\n",
      "working on POG358\n",
      "working on POG359\n",
      "working on POG360\n",
      "working on POG365\n",
      "working on POG367\n",
      "working on POG369\n",
      "working on POG370\n",
      "working on POG371\n",
      "working on POG372\n",
      "working on POG372\n",
      "working on POG373\n",
      "working on POG374\n",
      "working on POG375\n",
      "working on POG377\n",
      "working on POG379\n",
      "working on POG380\n",
      "working on POG381\n",
      "working on POG382\n",
      "working on POG383\n",
      "working on POG384\n",
      "working on POG385\n",
      "working on POG386\n",
      "working on POG387\n",
      "working on POG388\n",
      "working on POG395\n",
      "working on POG396\n",
      "working on POG396\n",
      "working on POG397\n",
      "working on POG398\n",
      "working on POG399\n",
      "working on POG400\n",
      "working on POG402\n",
      "working on POG403\n",
      "working on POG404\n",
      "working on POG405\n",
      "working on POG406\n",
      "working on POG407\n",
      "working on POG408\n",
      "working on POG409\n",
      "working on POG410\n",
      "working on POG411\n",
      "working on POG413\n",
      "working on POG414\n",
      "working on POG415\n",
      "working on POG415\n",
      "working on POG416\n",
      "working on POG417\n",
      "working on POG418\n",
      "working on POG419\n",
      "working on POG420\n",
      "working on POG421\n",
      "working on POG425\n",
      "working on POG426\n",
      "working on POG436\n",
      "working on POG438\n",
      "working on POG439\n",
      "working on POG441\n",
      "working on POG442\n",
      "working on POG443\n",
      "working on POG446\n",
      "working on POG448\n",
      "working on POG449\n",
      "working on POG450\n",
      "working on POG453\n",
      "working on POG454\n",
      "working on POG455\n",
      "working on POG456\n",
      "working on POG456\n",
      "working on POG457\n",
      "working on POG459\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "working on POG460\n",
      "working on POG461\n",
      "working on POG462\n",
      "working on POG463\n",
      "working on POG465\n",
      "working on POG468\n",
      "working on POG469\n",
      "working on POG469\n",
      "working on POG470\n",
      "working on POG472\n",
      "working on POG473\n",
      "working on POG474\n",
      "working on POG475\n",
      "working on POG476\n",
      "working on POG477\n",
      "working on POG478\n",
      "working on POG479\n",
      "working on POG480\n",
      "working on POG482\n",
      "working on POG485\n",
      "working on POG486\n",
      "working on POG487\n",
      "working on POG487\n",
      "working on POG488\n",
      "working on POG489\n",
      "working on POG490\n",
      "working on POG491\n",
      "working on POG495\n",
      "working on POG495\n",
      "working on POG496\n",
      "working on POG497\n",
      "working on POG498\n",
      "working on POG499\n",
      "working on POG502\n",
      "working on POG505\n",
      "working on POG506\n",
      "working on POG507\n",
      "working on POG509\n",
      "working on POG511\n",
      "working on POG512\n",
      "working on POG513\n",
      "working on POG514\n",
      "working on POG515\n",
      "working on POG517\n",
      "working on POG518\n",
      "working on POG519\n",
      "working on POG521\n",
      "working on POG524\n",
      "working on POG526\n",
      "working on POG530\n",
      "working on POG531\n",
      "working on POG532\n",
      "working on POG533\n",
      "working on POG534\n",
      "working on POG539\n",
      "working on POG540\n",
      "working on POG541\n",
      "working on POG541\n",
      "working on POG542\n",
      "working on POG546\n",
      "working on POG549\n",
      "working on POG550\n",
      "working on POG553\n",
      "working on POG554\n",
      "working on POG555\n",
      "working on POG558\n",
      "working on POG558\n",
      "working on POG559\n",
      "working on POG561\n",
      "working on POG565\n",
      "working on POG566\n",
      "working on POG569\n",
      "working on POG571\n",
      "working on POG574\n",
      "working on POG576\n",
      "working on POG577\n",
      "working on POG579\n",
      "working on POG580\n",
      "working on POG581\n",
      "working on POG583\n",
      "working on POG584\n",
      "working on POG586\n",
      "working on POG588\n",
      "working on POG590\n",
      "working on POG590\n",
      "working on POG591\n",
      "working on POG594\n",
      "working on POG596\n",
      "working on POG598\n",
      "working on POG600\n",
      "working on POG601\n",
      "working on POG602\n",
      "working on POG604\n",
      "working on POG604\n",
      "working on POG605\n",
      "working on POG605\n",
      "working on POG608\n",
      "working on POG609\n",
      "working on POG610\n",
      "working on POG611\n",
      "working on POG613\n",
      "working on POG615\n",
      "working on POG616\n",
      "working on POG617\n",
      "working on POG622\n",
      "working on POG623\n",
      "working on POG624\n",
      "working on POG625\n",
      "working on POG627\n",
      "working on POG629\n",
      "working on POG630\n",
      "working on POG631\n",
      "working on POG633\n",
      "working on POG634\n",
      "working on POG635\n",
      "working on POG636\n",
      "working on POG636\n",
      "working on POG637\n",
      "working on POG638\n",
      "working on POG640\n",
      "working on POG642\n",
      "working on POG643\n",
      "working on POG644\n",
      "working on POG646\n",
      "working on POG649\n",
      "working on POG650\n",
      "working on POG651\n",
      "working on POG652\n",
      "working on POG654\n",
      "working on POG655\n",
      "working on POG656\n",
      "working on POG658\n",
      "working on POG659\n",
      "working on POG661\n",
      "working on POG662\n",
      "working on POG663\n",
      "working on POG664\n",
      "working on POG665\n",
      "working on POG666\n",
      "working on POG667\n",
      "working on POG669\n",
      "working on POG670\n",
      "working on POG673\n",
      "working on POG674\n",
      "working on POG675\n",
      "working on POG676\n",
      "working on POG677\n",
      "working on POG678\n",
      "working on POG680\n",
      "working on POG681\n",
      "working on POG682\n",
      "working on POG683\n",
      "working on POG683\n",
      "working on POG684\n",
      "working on POG687\n",
      "working on POG688\n",
      "working on POG689\n",
      "working on POG690\n",
      "working on POG691\n",
      "working on POG692\n",
      "working on POG693\n",
      "working on POG694\n",
      "working on POG695\n",
      "working on POG698\n",
      "working on POG699\n",
      "working on POG700\n",
      "working on POG701\n",
      "working on POG702\n",
      "working on POG703\n",
      "working on POG704\n",
      "working on POG705\n",
      "working on POG706\n",
      "working on POG707\n",
      "working on POG710\n",
      "working on POG711\n",
      "working on POG712\n",
      "working on POG712\n",
      "working on POG713\n",
      "working on POG714\n",
      "working on POG716\n",
      "working on POG717\n",
      "working on POG719\n",
      "working on POG720\n",
      "working on POG720\n",
      "working on POG721\n",
      "working on POG723\n",
      "working on POG724\n",
      "working on POG725\n",
      "working on POG727\n",
      "working on POG728\n",
      "working on POG730\n",
      "working on POG732\n",
      "working on POG734\n",
      "working on POG736\n",
      "working on POG737\n",
      "working on POG738\n",
      "working on POG739\n",
      "working on POG740\n",
      "working on POG741\n",
      "working on POG742\n",
      "working on POG744\n",
      "working on POG745\n",
      "working on POG748\n",
      "working on POG751\n",
      "working on POG752\n",
      "working on POG753\n",
      "working on POG754\n",
      "working on POG755\n",
      "working on POG756\n",
      "working on POG757\n",
      "working on POG757\n",
      "working on POG759\n",
      "working on POG759\n",
      "working on POG760\n",
      "working on POG761\n",
      "working on POG762\n",
      "working on POG763\n",
      "working on POG764\n",
      "working on POG765\n",
      "working on POG766\n",
      "working on POG768\n",
      "working on POG771\n",
      "working on POG772\n",
      "working on POG773\n",
      "working on POG774\n",
      "working on POG775\n",
      "working on POG776\n",
      "working on POG777\n",
      "working on POG778\n",
      "working on POG780\n",
      "working on POG782\n",
      "working on POG783\n",
      "working on POG784\n",
      "working on POG785\n",
      "working on POG786\n",
      "working on POG788\n",
      "working on POG789\n",
      "working on POG790\n",
      "working on POG791\n",
      "working on POG792\n",
      "working on POG793\n",
      "working on POG794\n",
      "working on POG795\n",
      "working on POG797\n",
      "working on POG798\n",
      "working on POG799\n",
      "working on POG800\n",
      "working on POG802\n",
      "working on POG803\n",
      "working on POG804\n",
      "working on POG806\n",
      "working on POG807\n",
      "working on POG810\n",
      "working on POG811\n",
      "working on POG812\n",
      "working on POG813\n",
      "working on POG814\n",
      "working on POG815\n",
      "working on POG816\n",
      "working on POG818\n",
      "working on POG819\n",
      "working on POG820\n",
      "working on POG821\n",
      "working on POG822\n",
      "working on POG824\n",
      "working on POG826\n",
      "working on POG827\n",
      "working on POG828\n",
      "working on POG829\n",
      "working on POG830\n",
      "working on POG834\n",
      "working on POG835\n",
      "working on POG836\n",
      "working on POG837\n",
      "working on POG841\n",
      "working on POG844\n",
      "working on POG845\n",
      "working on POG846\n",
      "working on POG847\n",
      "working on POG848\n",
      "working on POG850\n",
      "working on POG852\n",
      "working on POG854\n",
      "working on POG855\n",
      "working on POG856\n",
      "working on POG857\n",
      "working on POG859\n",
      "working on POG861\n",
      "working on POG862\n",
      "working on POG864\n",
      "working on POG865\n",
      "working on POG867\n",
      "working on POG868\n",
      "working on POG870\n",
      "working on POG872\n",
      "working on POG873\n",
      "working on POG874\n",
      "working on POG875\n",
      "working on POG876\n",
      "working on POG877\n",
      "working on POG878\n",
      "working on POG879\n",
      "working on POG880\n",
      "working on POG881\n",
      "working on POG884\n",
      "working on POG885\n",
      "working on POG886\n",
      "working on POG887\n",
      "working on POG889\n",
      "working on POG890\n",
      "working on POG892\n",
      "working on POG894\n",
      "working on POG895\n",
      "working on POG896\n",
      "working on POG896\n",
      "working on POG898\n",
      "working on POG900\n",
      "working on POG901\n",
      "working on POG905\n",
      "working on POG906\n",
      "working on POG910\n",
      "working on POG912\n",
      "working on POG913\n",
      "working on POG914\n",
      "working on POG915\n",
      "working on POG918\n",
      "working on POG918\n",
      "working on POG920\n",
      "working on POG921\n",
      "working on POG922\n",
      "working on POG924\n",
      "working on POG925\n",
      "working on POG926\n",
      "working on POG927\n",
      "working on POG928\n",
      "working on POG929\n",
      "working on POG931\n",
      "working on POG932\n",
      "working on POG934\n",
      "working on POG935\n",
      "working on POG936\n",
      "working on POG937\n",
      "working on POG938\n",
      "working on POG939\n",
      "working on POG940\n",
      "working on POG941\n",
      "working on POG942\n"
     ]
    },
    {
     "ename": "EmptyDataError",
     "evalue": "No columns to parse from file",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mEmptyDataError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-d5ccfe728ef4>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0mpatient\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf'working on {patient}'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m     \u001b[0msnv_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparse_mutationseq_vcf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m     \u001b[0mmdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msnv_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mdfmg\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mempty\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-2-5bc5f93e7eb1>\u001b[0m in \u001b[0;36mparse_mutationseq_vcf\u001b[0;34m(vcf)\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0minclude\u001b[0m \u001b[0mall\u001b[0m \u001b[0mvariants\u001b[0m \u001b[0mhigh\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmoderate\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlow\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mmodifiers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     '''\n\u001b[0;32m----> 5\u001b[0;31m     \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvcf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcomment\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'#'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlow_memory\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m     \u001b[0mpatient\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvcf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'/'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m7\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, escapechar, comment, encoding, dialect, tupleize_cols, error_bad_lines, warn_bad_lines, skipfooter, doublequote, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    676\u001b[0m                     skip_blank_lines=skip_blank_lines)\n\u001b[1;32m    677\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 678\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    679\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    680\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    439\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 440\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    441\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    442\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    785\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    786\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 787\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    788\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    789\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1012\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'c'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1013\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'c'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1014\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1015\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1016\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'python'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/projects/da_workspace/software/miniconda/envs/dlpy3/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1706\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'usecols'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1707\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1708\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1709\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1710\u001b[0m         \u001b[0mpassed_names\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnames\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mEmptyDataError\u001b[0m: No columns to parse from file"
     ]
    }
   ],
   "source": [
    "f = '/projects/trans_scratch/validations/workspace/szong/Cervical/hotspots/pogs/pog_strelka_snv_calls.txt'\n",
    "df1 = pd.read_csv(f, header=None)[0].tolist()\n",
    "df1[:2]\n",
    "dfmg = pd.DataFrame()\n",
    "for p in df1:\n",
    "    patient = p.split('/')[4]\n",
    "    print(f'working on {patient}')\n",
    "    snv_df = parse_mutationseq_vcf(p)\n",
    "    mdf = snv_df\n",
    "    if dfmg.empty:\n",
    "        dfmg = mdf\n",
    "    else:\n",
    "        dfmg = pd.concat([dfmg, mdf], sort=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(24817057, 8)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>gene</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>825972</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>POG001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1063055</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>POG001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1277672</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRON</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>DVL1</td>\n",
       "      <td>POG001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1922492</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRON</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>C1orf222</td>\n",
       "      <td>POG001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2130700</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>EXON</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>C1orf86</td>\n",
       "      <td>POG001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr      pos ref alt      impact impact_type      gene patient\n",
       "0   1   825972   G   A  INTERGENIC    MODIFIER            POG001\n",
       "1   1  1063055   G   A  INTERGENIC    MODIFIER            POG001\n",
       "2   1  1277672   C   T      INTRON    MODIFIER      DVL1  POG001\n",
       "3   1  1922492   G   C      INTRON    MODIFIER  C1orf222  POG001\n",
       "4   1  2130700   A   C        EXON    MODIFIER   C1orf86  POG001"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "696"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfmg.shape\n",
    "dfmg.head()\n",
    "# dfmg.patient.nunique() # 696 pogs run through at this time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "snv_counts = dfmg.groupby(['chr', 'pos'])['patient'].nunique().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chr         pos     \n",
       "12          25398284    70\n",
       "GL000195.1  7094        60\n",
       "X           58562237    60\n",
       "1           565286      57\n",
       "X           58560599    57\n",
       "GL000205.1  109615      56\n",
       "GL000220.1  64524       56\n",
       "GL000195.1  7594        56\n",
       "MT          72          55\n",
       "15          84054881    55\n",
       "GL000205.1  109647      53\n",
       "8           43794763    53\n",
       "GL000220.1  65083       53\n",
       "GL000214.1  20627       52\n",
       "GL000195.1  61438       52\n",
       "8           43824536    51\n",
       "GL000220.1  65708       51\n",
       "GL000195.1  7559        51\n",
       "GL000220.1  85672       51\n",
       "GL000205.1  101824      51\n",
       "Name: patient, dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv_counts[:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "df11 = pd.DataFrame(snv_counts[snv_counts>14]).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "f11 = '/projects/trans_scratch/validations/workspace/szong/Cervical/hotspots/pogs/pog_strelka_snv_calls_696_pogs_3_or_more.txt'\n",
    "df11.to_csv(f11, sep='\\t', index=False )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.core.series.Series"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(snv_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "snv_counts.iplot(kind='hist',\n",
    "                xTitle='#patients',\n",
    "                yTitle='hist#patients',\n",
    "                title='recurrent snvs of 696 pogs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "working on HTMCP-03-06-02001\n",
      "working on HTMCP-03-06-02002\n",
      "working on HTMCP-03-06-02003\n",
      "working on HTMCP-03-06-02006\n",
      "working on HTMCP-03-06-02008\n",
      "working on HTMCP-03-06-02012\n",
      "working on HTMCP-03-06-02013\n",
      "working on HTMCP-03-06-02020\n",
      "working on HTMCP-03-06-02036\n",
      "working on HTMCP-03-06-02037\n",
      "working on HTMCP-03-06-02040\n",
      "working on HTMCP-03-06-02042\n",
      "working on HTMCP-03-06-02046\n",
      "working on HTMCP-03-06-02047\n",
      "working on HTMCP-03-06-02054\n",
      "working on HTMCP-03-06-02057\n",
      "working on HTMCP-03-06-02058\n",
      "working on HTMCP-03-06-02063\n",
      "working on HTMCP-03-06-02068\n",
      "working on HTMCP-03-06-02070\n",
      "working on HTMCP-03-06-02071\n",
      "working on HTMCP-03-06-02074\n",
      "working on HTMCP-03-06-02075\n",
      "working on HTMCP-03-06-02076\n",
      "working on HTMCP-03-06-02089\n",
      "working on HTMCP-03-06-02092\n",
      "working on HTMCP-03-06-02097\n",
      "working on HTMCP-03-06-02098\n",
      "working on HTMCP-03-06-02099\n",
      "working on HTMCP-03-06-02103\n",
      "working on HTMCP-03-06-02108\n",
      "working on HTMCP-03-06-02109\n",
      "working on HTMCP-03-06-02110\n",
      "working on HTMCP-03-06-02113\n",
      "working on HTMCP-03-06-02120\n",
      "working on HTMCP-03-06-02125\n",
      "working on HTMCP-03-06-02128\n",
      "working on HTMCP-03-06-02139\n",
      "working on HTMCP-03-06-02144\n",
      "working on HTMCP-03-06-02145\n",
      "working on HTMCP-03-06-02147\n",
      "working on HTMCP-03-06-02148\n",
      "working on HTMCP-03-06-02149\n",
      "working on HTMCP-03-06-02150\n",
      "working on HTMCP-03-06-02151\n",
      "working on HTMCP-03-06-02152\n",
      "working on HTMCP-03-06-02155\n",
      "working on HTMCP-03-06-02156\n",
      "working on HTMCP-03-06-02170\n",
      "working on HTMCP-03-06-02174\n",
      "working on HTMCP-03-06-02175\n",
      "working on HTMCP-03-06-02176\n",
      "working on HTMCP-03-06-02179\n",
      "working on HTMCP-03-06-02180\n",
      "working on HTMCP-03-06-02182\n",
      "working on HTMCP-03-06-02185\n",
      "working on HTMCP-03-06-02194\n",
      "working on HTMCP-03-06-02195\n",
      "working on HTMCP-03-06-02196\n",
      "working on HTMCP-03-06-02197\n",
      "working on HTMCP-03-06-02202\n",
      "working on HTMCP-03-06-02203\n",
      "working on HTMCP-03-06-02204\n",
      "working on HTMCP-03-06-02205\n",
      "working on HTMCP-03-06-02209\n",
      "working on HTMCP-03-06-02210\n",
      "working on HTMCP-03-06-02213\n",
      "working on HTMCP-03-06-02214\n",
      "working on HTMCP-03-06-02215\n",
      "working on HTMCP-03-06-02216\n",
      "working on HTMCP-03-06-02217\n",
      "working on HTMCP-03-06-02219\n",
      "working on HTMCP-03-06-02222\n",
      "working on HTMCP-03-06-02225\n",
      "working on HTMCP-03-06-02230\n",
      "working on HTMCP-03-06-02235\n",
      "working on HTMCP-03-06-02238\n",
      "working on HTMCP-03-06-02239\n",
      "working on HTMCP-03-06-02240\n",
      "working on HTMCP-03-06-02242\n",
      "working on HTMCP-03-06-02256\n",
      "working on HTMCP-03-06-02259\n",
      "working on HTMCP-03-06-02260\n",
      "working on HTMCP-03-06-02261\n",
      "working on HTMCP-03-06-02262\n",
      "working on HTMCP-03-06-02264\n",
      "working on HTMCP-03-06-02266\n",
      "working on HTMCP-03-06-02267\n",
      "working on HTMCP-03-06-02268\n",
      "working on HTMCP-03-06-02320\n",
      "working on HTMCP-03-06-02322\n",
      "working on HTMCP-03-06-02326\n",
      "working on HTMCP-03-06-02328\n",
      "working on HTMCP-03-06-02330\n",
      "working on HTMCP-03-06-02332\n",
      "working on HTMCP-03-06-02333\n",
      "working on HTMCP-03-06-02335\n",
      "working on HTMCP-03-06-02340\n",
      "working on HTMCP-03-06-02341\n",
      "working on HTMCP-03-06-02344\n",
      "working on HTMCP-03-06-02346\n",
      "working on HTMCP-03-06-02354\n",
      "working on HTMCP-03-06-02377\n",
      "working on HTMCP-03-06-02392\n",
      "working on HTMCP-03-06-02393\n",
      "working on HTMCP-03-06-02396\n",
      "working on HTMCP-03-06-02411\n",
      "working on HTMCP-03-06-02414\n",
      "working on HTMCP-03-06-02417\n",
      "working on HTMCP-03-06-02424\n",
      "working on HTMCP-03-06-02427\n",
      "working on HTMCP-03-06-02428\n",
      "working on HTMCP-03-06-02434\n",
      "working on HTMCP-03-06-02435\n",
      "working on HTMCP-03-06-02437\n",
      "working on HTMCP-03-06-02441\n",
      "working on HTMCP-03-06-02442\n",
      "working on HTMCP-03-06-02447\n"
     ]
    }
   ],
   "source": [
    "dfmg = pd.DataFrame()\n",
    "for ix, row in pd.DataFrame(ndf).iterrows():\n",
    "    patient = ix\n",
    "    print(f'working on {patient}')\n",
    "    snv_vcf = row['mutseq_snv_vcf']\n",
    "#     indel_vcf = row['strelka_indel_vcf']\n",
    "    snv_df = parse_mutationseq_vcf(snv_vcf)\n",
    "#     indel_df = parse_strelka_vcf(indel_vcf)\n",
    "#     mdf = pd.concat([snv_df, indel_df], sort=False)\n",
    "#     mdf = mdf.drop_duplicates(keep='first')\n",
    "    mdf = snv_df\n",
    "    if dfmg.empty:\n",
    "        dfmg = mdf\n",
    "    else:\n",
    "        dfmg = pd.concat([dfmg, mdf], sort=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>gene</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>54586</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>59734</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4G11P</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>66327</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>OR4G11P</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>99072</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRON</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>238990</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>AP006222.2</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr     pos ref alt      impact impact_type          gene            patient\n",
       "0   1   54586   T   C  INTERGENIC    MODIFIER                HTMCP-03-06-02001\n",
       "1   1   59734   A   G    UPSTREAM    MODIFIER       OR4G11P  HTMCP-03-06-02001\n",
       "2   1   66327   A   T  DOWNSTREAM    MODIFIER       OR4G11P  HTMCP-03-06-02001\n",
       "3   1   99072   C   T      INTRON    MODIFIER  RP11-34P13.7  HTMCP-03-06-02001\n",
       "4   1  238990   T   A    UPSTREAM    MODIFIER    AP006222.2  HTMCP-03-06-02001"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(7738105, 8)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "array(['MODIFIER', 'LOW', 'HIGH', 'MODERATE'], dtype=object)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfmg.head()\n",
    "dfmg.shape\n",
    "dfmg.impact_type.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "snv_counts = dfmg.groupby(['chr', 'pos'])['patient'].nunique().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "560480"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(snv_counts>=3).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "snv_counts[snv_counts>=3].to_csv('/projects/trans_scratch/validations/workspace/szong/Cervical/hotspots/rainstorm/118_cervical_mutationseq_snvs_3_patients.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>impact</th>\n",
       "      <th>impact_type</th>\n",
       "      <th>gene</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5769</th>\n",
       "      <td>3</td>\n",
       "      <td>178936091</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5771</th>\n",
       "      <td>3</td>\n",
       "      <td>178952117</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr        pos ref alt                 impact impact_type    gene  \\\n",
       "5769   3  178936091   G   A  NON_SYNONYMOUS_CODING    MODERATE  PIK3CA   \n",
       "5771   3  178952117   A   T  NON_SYNONYMOUS_CODING    MODERATE  PIK3CA   \n",
       "\n",
       "                patient  \n",
       "5769  HTMCP-03-06-02001  \n",
       "5771  HTMCP-03-06-02001  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(168, 8)"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset 12 genes of interest\n",
    "dfs = dfmg[dfmg.gene.isin(genes)&dfmg.patient.isin(patients)]\n",
    "dfs.head(2)\n",
    "dfs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(168, 3)"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(148, 3)"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = dfs[['gene', 'impact', 'patient']]\n",
    "dfs.shape\n",
    "\n",
    "dfs['impact_new'] = dfs['impact'].apply(lambda x: x.upper())\n",
    "dfs.drop('impact', axis=1, inplace=True)\n",
    "dfs.drop_duplicates(keep='first', inplace=True)\n",
    "dfs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NON_SYNONYMOUS_CODING', 'FRAME_SHIFT', 'STOP_GAINED',\n",
       "       'MISSENSE_VARIANT', 'SYNONYMOUS_VARIANT', 'SYNONYMOUS_CODING',\n",
       "       'SPLICE_SITE_ACCEPTOR', 'START_GAINED', 'CODON_DELETION',\n",
       "       'CODON_CHANGE_PLUS_CODON_DELETION'], dtype=object)"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['impact_new'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>patient</th>\n",
       "      <th>impact_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02001</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02012</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02036</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02239</td>\n",
       "      <td>MISSENSE_VARIANT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>HTMCP-03-06-02260</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene            patient             impact_new\n",
       "0  CASP8  HTMCP-03-06-02001            FRAME_SHIFT\n",
       "1  CASP8  HTMCP-03-06-02012            STOP_GAINED\n",
       "2  CASP8  HTMCP-03-06-02036  NON_SYNONYMOUS_CODING\n",
       "3  CASP8  HTMCP-03-06-02239       MISSENSE_VARIANT\n",
       "4  CASP8  HTMCP-03-06-02260            FRAME_SHIFT"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = dfs.drop_duplicates().groupby(['gene', 'patient']).agg({'impact_new': ','.join}).reset_index()\n",
    "dfs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "def impact_type(x):\n",
    "#     print(x)\n",
    "    xsplit = list(set(x.split(',')))\n",
    "    type = xsplit[0].upper()\n",
    "    if len(xsplit) > 1:\n",
    "        impact = 'Multiple'\n",
    "    elif len(xsplit) == 1:\n",
    "         if '+' in xsplit[0]:\n",
    "             impact = 'Multiple'\n",
    "         elif type == 'MISSENSE_VARIANT' or type == 'NON_SYNONYMOUS_CODING':\n",
    "             impact = 'Non-synonymous'\n",
    "         elif type == 'SYNONYMOUS_VARIANT' or type == 'SYNONYMOUS_CODING':\n",
    "             impact = 'Synonymous'\n",
    "         elif  type == 'SPLICE_SITE_ACCEPTOR' or  type == 'SPLICE_SITE_DONOR':\n",
    "             impact = 'Splite site'\n",
    "         elif type == 'STOP_LOST':\n",
    "             impact = 'Stop lost'\n",
    "         elif type == 'STOP_GAINED':\n",
    "             impact = 'Stop gained'\n",
    "         elif  type == 'START_LOST':\n",
    "             impact = 'Start lost'\n",
    "         elif type == 'START_GAINED':\n",
    "             impact = 'Start gained'\n",
    "         elif type == 'FRAME_SHIFT' or type == 'FRAMESHIFT_VARIANT':\n",
    "             impact = 'Frameshift'\n",
    "         elif type == 'INFRAME_DELETION' or type == 'CODON_INSERTION' or type == 'CODON_DELETION' or type == 'CODON_CHANGE_PLUS_CODON_INSERTION' or type == 'CODON_CHANGE_PLUS_CODON_DELETION' or type.lower() == 'disruptive_inframe_deletion':\n",
    "             impact = 'Codon indel'                \n",
    "         else: exit(1);print('ERROR');print(x)\n",
    "    return impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02040</th>\n",
       "      <th>HTMCP-03-06-02042</th>\n",
       "      <th>HTMCP-03-06-02047</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02344</th>\n",
       "      <th>HTMCP-03-06-02346</th>\n",
       "      <th>HTMCP-03-06-02354</th>\n",
       "      <th>HTMCP-03-06-02392</th>\n",
       "      <th>HTMCP-03-06-02411</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CASP8</th>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAT1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>Synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 80 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02006  \\\n",
       "gene                                                            \n",
       "CASP8          Frameshift               NaN               NaN   \n",
       "FAT1                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02012 HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "gene                                                            \n",
       "CASP8         Stop gained               NaN               NaN   \n",
       "FAT1                  NaN               NaN               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02036 HTMCP-03-06-02040 HTMCP-03-06-02042  \\\n",
       "gene                                                            \n",
       "CASP8      Non-synonymous               NaN               NaN   \n",
       "FAT1          Stop gained        Synonymous               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02047        ...        HTMCP-03-06-02344  \\\n",
       "gene                             ...                            \n",
       "CASP8                 NaN        ...                      NaN   \n",
       "FAT1                  NaN        ...                      NaN   \n",
       "\n",
       "patient HTMCP-03-06-02346 HTMCP-03-06-02354 HTMCP-03-06-02392  \\\n",
       "gene                                                            \n",
       "CASP8                 NaN               NaN               NaN   \n",
       "FAT1                  NaN       Stop gained               NaN   \n",
       "\n",
       "patient HTMCP-03-06-02411 HTMCP-03-06-02427 HTMCP-03-06-02428  \\\n",
       "gene                                                            \n",
       "CASP8                 NaN               NaN               NaN   \n",
       "FAT1                  NaN        Frameshift          Multiple   \n",
       "\n",
       "patient HTMCP-03-06-02435 HTMCP-03-06-02441 HTMCP-03-06-02447  \n",
       "gene                                                           \n",
       "CASP8                 NaN               NaN               NaN  \n",
       "FAT1       Non-synonymous               NaN               NaN  \n",
       "\n",
       "[2 rows x 80 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(12, 80)"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['impact'] = dfs['impact_new'].apply(lambda x: impact_type(x))\n",
    "dfs = dfs[['gene', 'patient', 'impact']].drop_duplicates()\n",
    "dfs = dfs.set_index(['gene', 'patient'])['impact'].unstack()\n",
    "dfs.head(2)\n",
    "dfs.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5. add in covariates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Squamous', 'Adenosquamous', 'Adeno', 'Neuroendocrine'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>patient</th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02417</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "patient            HTMCP-03-06-02001 HTMCP-03-06-02002 HTMCP-03-06-02003  \\\n",
       "HIV_status                  Positive          Negative          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02006 HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "HIV_status                  Negative          Negative          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02013 HTMCP-03-06-02020 HTMCP-03-06-02036  \\\n",
       "HIV_status                  Negative          Positive          Negative   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02037        ...        HTMCP-03-06-02417  \\\n",
       "HIV_status                  Negative        ...                 Positive   \n",
       "Putative_histology          Squamous        ...                 Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02424 HTMCP-03-06-02427 HTMCP-03-06-02428  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02434 HTMCP-03-06-02435 HTMCP-03-06-02437  \\\n",
       "HIV_status                  Positive          Positive          Positive   \n",
       "Putative_histology          Squamous          Squamous          Squamous   \n",
       "\n",
       "patient            HTMCP-03-06-02441 HTMCP-03-06-02442 HTMCP-03-06-02447  \n",
       "HIV_status                  Positive          Positive          Positive  \n",
       "Putative_histology             Adeno          Squamous          Squamous  \n",
       "\n",
       "[2 rows x 118 columns]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "(2, 118)"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get meta track for example histology\n",
    "f3 ='/projects/trans_scratch/validations/workspace/szong/Cervical/variant_bwamem/124_patients/mutation_load_clinic.txt'\n",
    "ddf = pd.read_csv(f3, sep='\\t')\n",
    "ddf = ddf[['patient', 'HIV_status', 'Putative_histology']]\n",
    "edf = ddf.set_index('patient').T\n",
    "edf.loc['Putative_histology',].unique()\n",
    "edf = edf[patients]\n",
    "edf.head(2)\n",
    "edf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs = pd.concat([edf, dfs], sort=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(14, 118)"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs = dfs[patients]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 6. add in mutation rate (fraction of patients)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs['occurrence'] = (dfs.notnull().sum(axis=1)/(dfs.shape[1] -2))\n",
    "dfs['percentage'] = ['{0}({1}%)'.format(i[0], int(round(i[1]*100))) for i in zip(dfs.index.tolist(), dfs.occurrence)]\n",
    "# dfs.loc['HIV_status', 'percentage'] = 'HIV_Status'\n",
    "# dfs.loc['Putative_histology', 'percentage' ] = 'Puatative_histology'\n",
    "sdf = dfs.drop('occurrence', axis=1).set_index('percentage', drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HTMCP-03-06-02001</th>\n",
       "      <th>HTMCP-03-06-02002</th>\n",
       "      <th>HTMCP-03-06-02003</th>\n",
       "      <th>HTMCP-03-06-02006</th>\n",
       "      <th>HTMCP-03-06-02008</th>\n",
       "      <th>HTMCP-03-06-02012</th>\n",
       "      <th>HTMCP-03-06-02013</th>\n",
       "      <th>HTMCP-03-06-02020</th>\n",
       "      <th>HTMCP-03-06-02036</th>\n",
       "      <th>HTMCP-03-06-02037</th>\n",
       "      <th>...</th>\n",
       "      <th>HTMCP-03-06-02417</th>\n",
       "      <th>HTMCP-03-06-02424</th>\n",
       "      <th>HTMCP-03-06-02427</th>\n",
       "      <th>HTMCP-03-06-02428</th>\n",
       "      <th>HTMCP-03-06-02434</th>\n",
       "      <th>HTMCP-03-06-02435</th>\n",
       "      <th>HTMCP-03-06-02437</th>\n",
       "      <th>HTMCP-03-06-02441</th>\n",
       "      <th>HTMCP-03-06-02442</th>\n",
       "      <th>HTMCP-03-06-02447</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>percentage</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HIV_status(102%)</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Putative_histology(102%)</th>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>...</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Adeno</td>\n",
       "      <td>Squamous</td>\n",
       "      <td>Squamous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CASP8(7%)</th>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FAT1(19%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FBXW7(10%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAPK1(5%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLL2(15%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Stop gained</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCDHA9(3%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PCDHGA12(5%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3CA(35%)</th>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSPC1(5%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-synonymous</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC35G5(7%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZC3H6(3%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF750(4%)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14 rows Ã— 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         HTMCP-03-06-02001 HTMCP-03-06-02002  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                       Frameshift               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN       Stop gained   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                 Non-synonymous    Non-synonymous   \n",
       "PSPC1(5%)                              NaN    Non-synonymous   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN        Frameshift   \n",
       "\n",
       "                         HTMCP-03-06-02003 HTMCP-03-06-02006  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN    Non-synonymous   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN    Non-synonymous   \n",
       "PIK3CA(35%)                            NaN               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02008 HTMCP-03-06-02012  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Negative          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN       Stop gained   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                            NaN    Non-synonymous   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02013 HTMCP-03-06-02020  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Negative          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                 Non-synonymous               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN          Multiple   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02036 HTMCP-03-06-02037  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Negative          Negative   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                   Non-synonymous               NaN   \n",
       "FAT1(19%)                      Stop gained               NaN   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                 Non-synonymous               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                                ...        HTMCP-03-06-02417  \\\n",
       "percentage                      ...                            \n",
       "HIV_status(102%)                ...                 Positive   \n",
       "Putative_histology(102%)        ...                 Squamous   \n",
       "CASP8(7%)                       ...                      NaN   \n",
       "FAT1(19%)                       ...                      NaN   \n",
       "FBXW7(10%)                      ...                      NaN   \n",
       "MAPK1(5%)                       ...                      NaN   \n",
       "MLL2(15%)                       ...                      NaN   \n",
       "PCDHA9(3%)                      ...                      NaN   \n",
       "PCDHGA12(5%)                    ...                      NaN   \n",
       "PIK3CA(35%)                     ...                      NaN   \n",
       "PSPC1(5%)                       ...                      NaN   \n",
       "SLC35G5(7%)                     ...                      NaN   \n",
       "ZC3H6(3%)                       ...                      NaN   \n",
       "ZNF750(4%)                      ...                      NaN   \n",
       "\n",
       "                         HTMCP-03-06-02424 HTMCP-03-06-02427  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                              NaN        Frameshift   \n",
       "FBXW7(10%)                             NaN               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                            NaN               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02428 HTMCP-03-06-02434  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                         Multiple               NaN   \n",
       "FBXW7(10%)                  Non-synonymous               NaN   \n",
       "MAPK1(5%)                   Non-synonymous               NaN   \n",
       "MLL2(15%)                              NaN               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                           NaN               NaN   \n",
       "PIK3CA(35%)                 Non-synonymous               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                         Multiple               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02435 HTMCP-03-06-02437  \\\n",
       "percentage                                                     \n",
       "HIV_status(102%)                  Positive          Positive   \n",
       "Putative_histology(102%)          Squamous          Squamous   \n",
       "CASP8(7%)                              NaN               NaN   \n",
       "FAT1(19%)                   Non-synonymous               NaN   \n",
       "FBXW7(10%)                      Frameshift               NaN   \n",
       "MAPK1(5%)                              NaN               NaN   \n",
       "MLL2(15%)                       Frameshift               NaN   \n",
       "PCDHA9(3%)                             NaN               NaN   \n",
       "PCDHGA12(5%)                Non-synonymous               NaN   \n",
       "PIK3CA(35%)                            NaN               NaN   \n",
       "PSPC1(5%)                              NaN               NaN   \n",
       "SLC35G5(7%)                            NaN               NaN   \n",
       "ZC3H6(3%)                              NaN               NaN   \n",
       "ZNF750(4%)                             NaN               NaN   \n",
       "\n",
       "                         HTMCP-03-06-02441 HTMCP-03-06-02442 HTMCP-03-06-02447  \n",
       "percentage                                                                      \n",
       "HIV_status(102%)                  Positive          Positive          Positive  \n",
       "Putative_histology(102%)             Adeno          Squamous          Squamous  \n",
       "CASP8(7%)                              NaN               NaN               NaN  \n",
       "FAT1(19%)                              NaN               NaN               NaN  \n",
       "FBXW7(10%)                             NaN               NaN               NaN  \n",
       "MAPK1(5%)                              NaN               NaN               NaN  \n",
       "MLL2(15%)                              NaN               NaN               NaN  \n",
       "PCDHA9(3%)                             NaN               NaN               NaN  \n",
       "PCDHGA12(5%)                           NaN               NaN               NaN  \n",
       "PIK3CA(35%)                            NaN               NaN    Non-synonymous  \n",
       "PSPC1(5%)                   Non-synonymous               NaN               NaN  \n",
       "SLC35G5(7%)                            NaN               NaN               NaN  \n",
       "ZC3H6(3%)                              NaN               NaN               NaN  \n",
       "ZNF750(4%)                             NaN               NaN               NaN  \n",
       "\n",
       "[14 rows x 118 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "of = '/projects/trans_scratch/validations/workspace/szong/Cervical/mutsig2cv/118_patients/smgs_reviewed_details.txt'\n",
    "sdf.to_csv(of, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
